Index: kpdf/xpdf/xpdf/Lexer.cc
===================================================================
--- kpdf/xpdf/xpdf/Lexer.cc	(.../tags/KDE/3.5.9/kdegraphics)	(wersja 781012)
+++ kpdf/xpdf/xpdf/Lexer.cc	(.../branches/KDE/3.5/kdegraphics)	(wersja 781012)
@@ -298,7 +298,7 @@
 	  n = 0;
 	  
 	  // we are growing see if the document is not malformed and we are growing too much
-	  if (objNum > 0)
+	  if (objNum > 0 && xref != NULL)
 	  {
 	    int newObjNum = xref->getNumEntry(curStr.streamGetPos());
 	    if (newObjNum != objNum)
Index: kpdf/xpdf/splash/SplashFont.cc
===================================================================
--- kpdf/xpdf/splash/SplashFont.cc	(.../tags/KDE/3.5.9/kdegraphics)	(wersja 781012)
+++ kpdf/xpdf/splash/SplashFont.cc	(.../branches/KDE/3.5/kdegraphics)	(wersja 781012)
@@ -167,27 +167,35 @@
     size = ((bitmap2.w + 7) >> 3) * bitmap2.h;
   }
   p = NULL; // make gcc happy
-  for (j = 0; j < cacheAssoc; ++j) {
-    if ((cacheTags[i+j].mru & 0x7fffffff) == cacheAssoc - 1) {
-      cacheTags[i+j].mru = 0x80000000;
-      cacheTags[i+j].c = c;
-      cacheTags[i+j].xFrac = (short)xFrac;
-      cacheTags[i+j].yFrac = (short)yFrac;
-      cacheTags[i+j].x = bitmap2.x;
-      cacheTags[i+j].y = bitmap2.y;
-      cacheTags[i+j].w = bitmap2.w;
-      cacheTags[i+j].h = bitmap2.h;
-      p = cache + (i+j) * glyphSize;
-      memcpy(p, bitmap2.data, size);
-    } else {
-      ++cacheTags[i+j].mru;
+  if (cacheAssoc == 0)
+  {
+    // we had problems on the malloc of the cache, so ignore it
+    *bitmap = bitmap2;
+  }
+  else
+  {
+    for (j = 0; j < cacheAssoc; ++j) {
+      if ((cacheTags[i+j].mru & 0x7fffffff) == cacheAssoc - 1) {
+        cacheTags[i+j].mru = 0x80000000;
+        cacheTags[i+j].c = c;
+        cacheTags[i+j].xFrac = (short)xFrac;
+        cacheTags[i+j].yFrac = (short)yFrac;
+        cacheTags[i+j].x = bitmap2.x;
+        cacheTags[i+j].y = bitmap2.y;
+        cacheTags[i+j].w = bitmap2.w;
+        cacheTags[i+j].h = bitmap2.h;
+        p = cache + (i+j) * glyphSize;
+        memcpy(p, bitmap2.data, size);
+      } else {
+        ++cacheTags[i+j].mru;
+      }
     }
+    *bitmap = bitmap2;
+    bitmap->data = p;
+    bitmap->freeData = gFalse;
+    if (bitmap2.freeData) {
+      gfree(bitmap2.data);
+    }
   }
-  *bitmap = bitmap2;
-  bitmap->data = p;
-  bitmap->freeData = gFalse;
-  if (bitmap2.freeData) {
-    gfree(bitmap2.data);
-  }
   return gTrue;
 }
Index: kdvi/fontpool.cpp
===================================================================
--- kdvi/fontpool.cpp	(.../tags/KDE/3.5.9/kdegraphics)	(wersja 781012)
+++ kdvi/fontpool.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 781012)
@@ -111,6 +111,9 @@
   kdDebug(4300) << "fontPool::~fontPool() called" << endl;
 #endif
 
+  // need to manually clear the fonts _before_ freetype gets unloaded
+  fontList.clear();
+
 #ifdef HAVE_FREETYPE
   if (FreeType_could_be_loaded == true)
     FT_Done_FreeType( FreeType_library );

Zmiany atrybutów dla: .
___________________________________________________________________
Nazwa: svn:mergeinfo
   - 
Nazwa: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


