--- kdegraphics-3.4.2/ksnapshot/ksnapshotwidget.ui.orig	2005-05-23 14:06:04.000000000 +0200
+++ kdegraphics-3.4.2/ksnapshot/ksnapshotwidget.ui	2005-09-22 13:28:15.000000000 +0200
@@ -342,7 +342,6 @@
 <layoutdefaults spacing="6" margin="11"/>
 <layoutfunctions spacing="KDialog::spacingHint" margin="KDialog::marginHint"/>
 <includehints>
-    <includehint>ksnapshotpreview.h</includehint>
     <includehint>kpushbutton.h</includehint>
     <includehint>kpushbutton.h</includehint>
     <includehint>kpushbutton.h</includehint>
--- kdegraphics-3.4.2/kviewshell/pageSizeWidget_base.ui.orig	2005-05-23 14:06:06.000000000 +0200
+++ kdegraphics-3.4.2/kviewshell/pageSizeWidget_base.ui	2005-09-22 14:18:45.000000000 +0200
@@ -284,6 +284,5 @@
     <includehint>kcombobox.h</includehint>
     <includehint>kcombobox.h</includehint>
     <includehint>kcombobox.h</includehint>
-    <includehint>sizepreview.h</includehint>
 </includehints>
 </UI>
