--- kdegraphics-2.2/kamera/kcontrol/kamera.cpp--	Tue Aug 14 15:34:17 2001
+++ kdegraphics-2.2/kamera/kcontrol/kamera.cpp	Tue Aug 14 15:34:23 2001
@@ -368,7 +368,7 @@ void KKameraConfig::setCameraType(QListV
 {
 	CameraAbilities abilities;
 
-	char *name = item->text(0).latin1();
+	char *name = (char*)item->text(0).latin1();
 
 	// retrieve camera abilities structure
 	if(gp_camera_abilities_by_name(name, &abilities) == GP_OK) {
