diff -urN -x CVS kdegraphics.orig/kamera/kcontrol/kamera.desktop kdegraphics/kamera/kcontrol/kamera.desktop
--- kdegraphics.orig/kamera/kcontrol/kamera.desktop	Fri Mar  5 13:51:03 2004
+++ kdegraphics/kamera/kcontrol/kamera.desktop	Mon May  3 15:03:28 2004
@@ -20,7 +20,7 @@
 Comment[ga]=Cumraigh Kamera
 Comment[gl]=Configurar Kamera
 Comment[he]=שינוי הגדרות Kamera
-Comment[hi]=कॉन्फिगर केमरा
+Comment[hi]=कॉन्फ़िगर केमरा
 Comment[hr]=Podesi Kameru
 Comment[hu]=a Kamera beállítása
 Comment[id]=Konfigurasi kamera
@@ -58,7 +58,7 @@
 Comment[zu]=Hlanganisa ikhamera
 Exec=kcmshell Peripherals/kamera
 Keywords=gphoto,camera,digicam,webcam,kamera
-Keywords[ar]=gphoto,كاميرا,كاميرا رقمية,كلميرا ويب,kamera
+Keywords[ar]=gphoto,كاميرا,كاميرا رقمية,كاميرا ويب,kamera
 Keywords[az]=gphoto,kamera,digicam,veb kamera,Kamera,webcam
 Keywords[bg]=фото,апарат,фотоапарат,камера,цифров,gphoto,camera,digicam,webcam,kamera
 Keywords[ca]=gphoto,càmera,digicam,webcam,kamera
diff -urN -x CVS kdegraphics.orig/kcoloredit/kcolorchooser.desktop kdegraphics/kcoloredit/kcolorchooser.desktop
--- kdegraphics.orig/kcoloredit/kcolorchooser.desktop	Mon Mar 29 15:01:11 2004
+++ kdegraphics/kcoloredit/kcolorchooser.desktop	Mon May  3 15:03:30 2004
@@ -65,6 +65,7 @@
 GenericName[zu]=Umkhethi Wombala
 Name=KColorChooser
 Name[af]=K-kleur-kieser
+Name[ar]=برنامج KColorChooser
 Name[ca]=Elecció de colors
 Name[cs]=Výběr barev
 Name[cy]=KDewisLliw
diff -urN -x CVS kdegraphics.orig/kcoloredit/kcoloredit.desktop kdegraphics/kcoloredit/kcoloredit.desktop
--- kdegraphics.orig/kcoloredit/kcoloredit.desktop	Mon Mar 29 15:01:13 2004
+++ kdegraphics/kcoloredit/kcoloredit.desktop	Mon May  3 15:03:32 2004
@@ -10,7 +10,7 @@
 DocPath=kcoloredit/index.html
 GenericName=Color Palette Editor
 GenericName[af]=Kleur Palet Redigeerder
-GenericName[ar]=محرر لوحة رسم الألوان
+GenericName[ar]=محرر لوحة الألوان
 GenericName[bg]=Редактор на цветове
 GenericName[bs]=Editor palete boja
 GenericName[ca]=Editor de la paleta de colors
@@ -63,6 +63,7 @@
 GenericName[zu]=Umhleli Wombala we Palette
 Name=KColorEdit
 Name[af]=K-kleur-redigeer
+Name[ar]=برنامج KColorEdit
 Name[cs]=Editor barev
 Name[cy]=KGolyguLliw
 Name[eo]=Kolorredaktilo
diff -urN -x CVS kdegraphics.orig/kdvi/dviwin.cpp kdegraphics/kdvi/dviwin.cpp
--- kdegraphics.orig/kdvi/dviwin.cpp	Sat Mar 27 11:55:38 2004
+++ kdegraphics/kdvi/dviwin.cpp	Sun Apr 11 11:12:08 2004
@@ -309,7 +309,7 @@
 
       if (showMsg) {
 	KDialogBase *dialog= new KDialogBase(i18n("KDVI: Information"), KDialogBase::Yes, KDialogBase::Yes, KDialogBase::Yes,
-					     this, "information", true, true, i18n("&OK"));
+					     this, "information", true, true, KStdGuiItem::ok());
 
 	QVBox *topcontents = new QVBox (dialog);
 	topcontents->setSpacing(KDialog::spacingHint()*2);
@@ -739,7 +739,7 @@
   // which is NOT the default format, that means that the document
   // will be drawn twice.
   if (dviFile->suggestedPageSize != 0)
-    emit( documentSpecifiedPageSize(*(dviFile->suggestedPageSize)) );    
+    emit( documentSpecifiedPageSize(*(dviFile->suggestedPageSize)) );
   drawPage();
 
 
@@ -799,21 +799,21 @@
     // the LaTeX file. In that case, we jump to the beginning of the
     // document.
     bool anchorForRefFileFound = false; // Flag that is set if source file anchors for the refFileName could be found at all
-    
+
     QValueVector<DVI_SourceFileAnchor>::iterator bestMatch = sourceHyperLinkAnchors.end();
     QValueVector<DVI_SourceFileAnchor>::iterator it;
-    for( it = sourceHyperLinkAnchors.begin(); it != sourceHyperLinkAnchors.end(); ++it ) 
+    for( it = sourceHyperLinkAnchors.begin(); it != sourceHyperLinkAnchors.end(); ++it )
       if (refFileName.stripWhiteSpace() == it->fileName.stripWhiteSpace()) {
 	anchorForRefFileFound = true;
-	
+
 	if ( (it->line <= refLineNumber) &&
 	     ( (bestMatch == sourceHyperLinkAnchors.end()) || (it->line > bestMatch->line) ) )
 	  bestMatch = it;
       }
-    
+
     reference = QString::null;
-    
-    if (bestMatch != sourceHyperLinkAnchors.end()) 
+
+    if (bestMatch != sourceHyperLinkAnchors.end())
       emit(request_goto_page(bestMatch->page, (Q_INT32)(bestMatch->vertical_coordinate/shrinkfactor+0.5)));
     else
       if (anchorForRefFileFound == false)
@@ -822,7 +822,7 @@
 			   i18n( "Could Not Find Reference" ));
       else
 	emit(request_goto_page(0, 0));
-    
+
     return;
   }
   reference = QString::null;
@@ -836,7 +836,7 @@
 #ifdef DEBUG_DVIWIN
   kdDebug(4300) << "dviWindow::gotoPage( new_page=" << new_page << " )" << endl;
 #endif
-  
+
   if (dviFile == NULL) {
 #ifdef DEBUG_DVIWIN
     kdDebug(4300) << "gotoPage fails because no DVI file loaded" << endl;
@@ -1075,7 +1075,7 @@
     if (currentlyDrawnPage.hyperLinkList.size() > 0)
       for(int i=0; i<currentlyDrawnPage.hyperLinkList.size(); i++) {
 	if (currentlyDrawnPage.hyperLinkList[i].box.contains(e->pos())) {
-	  
+
 #ifdef DEBUG_SPECIAL
 	  kdDebug(4300) << "hit: local link to " << currentlyDrawnPage.hyperLinkList[i].linkText << endl;
 #endif
@@ -1102,7 +1102,7 @@
 	      // the kfmclient seems to be MUCH safer.
 	      QUrl DVI_Url(dviFile->filename);
 	      QUrl Link_Url(DVI_Url, currentlyDrawnPage.hyperLinkList[i].linkText, TRUE );
-	      
+
 	      QStringList args;
 	      args << "openURL";
 	      args << Link_Url.toString();
@@ -1142,7 +1142,7 @@
 	  QFileInfo fi3(fi1.dir(),cp.mid(i));
 	  TeXfile = fi3.absFilePath();
 	  if ( !fi3.exists() ) {
-	    KMessageBox::sorry(this, QString("<qt>") + 
+	    KMessageBox::sorry(this, QString("<qt>") +
 			       i18n("The DVI-file refers to the TeX-file "
 				    "<strong>%1</strong> which could not be found.").arg(KShellProcess::quote(TeXfile)) +
 			       QString("</qt>"),
@@ -1157,7 +1157,7 @@
 						     i18n("You have not yet specified an editor for inverse search. "
 							  "Please choose your favorite editor in the "
 							  "<strong>DVI options dialog</strong> "
-							  "which you will find in the <strong>Settings</strong>-menu.") + 
+							  "which you will find in the <strong>Settings</strong>-menu.") +
 							  QString("</qt>"),
 						     i18n("Need to Specify Editor"),
 		                                     i18n("Use KDE's Editor Kate for Now"));
diff -urN -x CVS kdegraphics.orig/kdvi/dviwin_textsearch.cpp kdegraphics/kdvi/dviwin_textsearch.cpp
--- kdegraphics.orig/kdvi/dviwin_textsearch.cpp	Sun Sep 14 13:18:09 2003
+++ kdegraphics/kdvi/dviwin_textsearch.cpp	Wed Apr 14 19:47:55 2004
@@ -25,9 +25,9 @@
 // of kdvi.
 
 
-#include <qapplication.h> 
+#include <qapplication.h>
 #include <qpainter.h>
-#include <qprogressdialog.h> 
+#include <qprogressdialog.h>
 #include <kmessagebox.h>
 #include <kdebug.h>
 #include <keditcl.h>
@@ -43,13 +43,13 @@
   if (findDialog == 0) {
     // WARNING: This text appears several times in the code. Change
     // everywhere, or nowhere!
-    if (KMessageBox::warningContinueCancel( this, 
+    if (KMessageBox::warningContinueCancel( this,
 					    i18n("<qt>This function searches the DVI file for plain text. Unfortunately, this version of "
 						 "KDVI treats only plain ASCII characters properly. Symbols, ligatures, mathematical "
 						 "formulae, accented characters, and non-english text, such as Russian or Korean, will "
 						 "most likely be messed up completely. Continue anyway?</qt>"),
 					    i18n("Function May Not Work as Expected"),
-					    i18n("Continue"),
+					    KStdGuiItem::cont(),
 					    "warning_search_text_may_not_work") == KMessageBox::Cancel)
       return;
     findDialog = new KEdFind(this, "Text find dialog", true);
@@ -97,9 +97,9 @@
   unsigned int lastPageOfSearch  = dviFile->total_pages-1;
 
   _postscript = FALSE; // Switch off postscript to speed up things...
-  QPixmap pixie(1,1); // Dummy pixmap for the method draw_page which wants to have a valid painter. 
+  QPixmap pixie(1,1); // Dummy pixmap for the method draw_page which wants to have a valid painter.
 
-  QProgressDialog progress( i18n("Searching for '%1'...").arg(searchText), i18n("Abort"), 
+  QProgressDialog progress( i18n("Searching for '%1'...").arg(searchText), i18n("Abort"),
 			    lastPageOfSearch - firstPageOfSearch, this, "searchForwardTextProgress", TRUE );
   progress.setMinimumDuration ( 1000 );
   while(current_page <= lastPageOfSearch) {
@@ -110,7 +110,7 @@
 
     // Go trough the text of the current page and search for the
     // string.
-    for(unsigned int i=DVIselection.selectedTextStart+1; i<currentlyDrawnPage.textLinkList.size(); i++) 
+    for(unsigned int i=DVIselection.selectedTextStart+1; i<currentlyDrawnPage.textLinkList.size(); i++)
       if (currentlyDrawnPage.textLinkList[i].linkText.find(searchText, 0, case_sensitive) >= 0) {
 	// Restore the previous settings, including the current
 	// page. Otherwise, the program is "smart enough" not to
@@ -134,7 +134,7 @@
       if (current_page_sav != 0) {
 	int answ = KMessageBox::questionYesNo(this, i18n("<qt>The search string <strong>%1</strong> could not be found till the "
 							 "end of the document. Should the search be restarted from the beginning "
-							 "of the document?</qt>").arg(searchText), 
+							 "of the document?</qt>").arg(searchText),
 					      i18n("Text Not Found"));
 	if (answ == KMessageBox::Yes) {
 	  firstPageOfSearch = 0;
@@ -149,12 +149,12 @@
     draw_page(); // We don't really care for errors in draw_page(), no error handling here.
     foreGroundPaint.end();
   }// of while (all pages)
-  
+
   KMessageBox::sorry( this, i18n("<qt>The search string <strong>%1</strong> could not be found.</qt>").arg(searchText) );
 
-  // Restore the PostScript setting 
+  // Restore the PostScript setting
   _postscript = _postscript_sav;
-  
+
   // Restore the current page.
   current_page = current_page_sav;
   foreGroundPaint.begin( &pixie );
@@ -185,7 +185,7 @@
   unsigned int lastPageOfSearch  = current_page;
 
   _postscript = FALSE; // Switch off postscript to speed up things...
-  QPixmap pixie(1,1); // Dummy pixmap for the method draw_page which wants to have a valid painter. 
+  QPixmap pixie(1,1); // Dummy pixmap for the method draw_page which wants to have a valid painter.
 
   QProgressDialog progress( i18n("Searching for '%1'...").arg(searchText), i18n("Abort"),
 			    lastPageOfSearch - firstPageOfSearch, this, "searchForwardTextProgress", TRUE );
@@ -216,8 +216,8 @@
       if (current_page_sav != 0) {
 	int answ = KMessageBox::questionYesNo(this, i18n("<qt>The search string <strong>%1</strong> could not be found till the "
 							 "beginning of the document. Should the search be restarted from the end "
-							 "of the document?</qt>").arg(searchText), 
-					      i18n("Text Not Found")); 
+							 "of the document?</qt>").arg(searchText),
+					      i18n("Text Not Found"));
 	if (answ == KMessageBox::Yes) {
 	  firstPageOfSearch = current_page_sav;
 	  current_page      = dviFile->total_pages-1;
@@ -232,7 +232,7 @@
     qApp->processEvents();
     if ( progress.wasCancelled() )
       break;
-    
+
     // Go trough the text of the current page and search for the
     // string.
     int i=DVIselection.selectedTextStart-1;
@@ -259,12 +259,12 @@
     draw_page(); // We don't really care for errors in draw_page(), no error handling here.
     foreGroundPaint.end();
   } while( current_page < dviFile->total_pages ); // Note: current_page is unsigned. It will not become negative, but very big
-  
+
   KMessageBox::sorry( this, i18n("<qt>The search string <strong>%1</strong> could not be found.</qt>").arg(searchText) );
 
-  // Restore the PostScript setting 
+  // Restore the PostScript setting
   _postscript = _postscript_sav;
-  
+
   // Restore the current page.
   current_page = current_page_sav;
   foreGroundPaint.begin( &pixie );
diff -urN -x CVS kdegraphics.orig/kdvi/kdvi.desktop kdegraphics/kdvi/kdvi.desktop
--- kdegraphics.orig/kdvi/kdvi.desktop	Fri Mar  5 13:51:05 2004
+++ kdegraphics/kdvi/kdvi.desktop	Mon May  3 15:03:34 2004
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 GenericName=DVI Viewer
 GenericName[af]=Dvi Aansig
-GenericName[ar]=عرض ملفات DVI
+GenericName[ar]=عارض ملفات DVI
 GenericName[az]=DVI Nümayişçisi
 GenericName[bg]=Преглед на документи DVI
 GenericName[bs]=Preglednik DVI dokumenata
@@ -65,6 +65,7 @@
 BinaryPattern=kdvi;
 Name=KDVI
 Name[af]=Kdvi
+Name[ar]=برنامج KDVI
 Name[eo]=DVI-rigardilo
 Name[hi]=के-डीवीआई
 Name[xx]=xxKDVIxx
diff -urN -x CVS kdegraphics.orig/kdvi/squeeze.c kdegraphics/kdvi/squeeze.c
--- kdegraphics.orig/kdvi/squeeze.c	Sat Jul 24 23:05:11 1999
+++ kdegraphics/kdvi/squeeze.c	Mon Apr 19 15:42:05 2004
@@ -181,6 +181,6 @@
    }
    (void)fprintf(out, "\\n\";\n\n\
 int\tpsheaderlen\t= %d;\n", stringlen + linepos + 1);
-   exit(0) ;
+   return 0;
    /*NOTREACHED*/
 }
diff -urN -x CVS kdegraphics.orig/kfax/kfax.desktop kdegraphics/kfax/kfax.desktop
--- kdegraphics.orig/kfax/kfax.desktop	Sun Feb 29 16:26:55 2004
+++ kdegraphics/kfax/kfax.desktop	Mon May  3 15:03:36 2004
@@ -20,7 +20,7 @@
 GenericName[fr]=Afficheur de fax
 GenericName[gl]=Visor de fax
 GenericName[he]=מציג פקסים
-GenericName[hi]=फैक्स प्रदर्शक
+GenericName[hi]=फ़ैक्स प्रदर्शक
 GenericName[hr]=Preglednik faksova
 GenericName[hu]=Faxnézegető
 GenericName[is]=Fax sjá
@@ -57,9 +57,10 @@
 GenericName[zu]=Umbonisi wefax
 Name=KFax
 Name[af]=K-faks
+Name[ar]=برنامج KFax
 Name[cy]=KFfacs
 Name[eo]=Faksrigardilo
-Name[hi]=के-फैक्स
+Name[hi]=के-फ़ैक्स
 Name[lv]=KFakss
 Name[sv]=Kfax
 Name[ta]=கேஃபாக்ஸ்
diff -urN -x CVS kdegraphics.orig/kfile-plugins/bmp/kfile_bmp.desktop kdegraphics/kfile-plugins/bmp/kfile_bmp.desktop
--- kdegraphics.orig/kfile-plugins/bmp/kfile_bmp.desktop	Mon Mar 29 15:01:15 2004
+++ kdegraphics/kfile-plugins/bmp/kfile_bmp.desktop	Mon May  3 15:03:38 2004
@@ -3,6 +3,7 @@
 Type=Service
 Name=BMP Info
 Name[af]=Bmp Inligting
+Name[ar]=معلومات BMP
 Name[ca]=Informació de BMP
 Name[cs]=BMP info
 Name[cy]=Gwybodaeth BMP
diff -urN -x CVS kdegraphics.orig/kfile-plugins/dvi/kfile_dvi.desktop kdegraphics/kfile-plugins/dvi/kfile_dvi.desktop
--- kdegraphics.orig/kfile-plugins/dvi/kfile_dvi.desktop	Mon Mar 29 15:01:17 2004
+++ kdegraphics/kfile-plugins/dvi/kfile_dvi.desktop	Mon May  3 15:03:40 2004
@@ -3,6 +3,7 @@
 Icon=
 MimeType=application/x-dvi
 Name=DVI Info
+Name[ar]=معلومات DVI
 Name[ca]=Informació de DVI
 Name[cs]=DVI info
 Name[cy]=Gwybodaeth DVI
diff -urN -x CVS kdegraphics.orig/kfile-plugins/gif/kfile_gif.desktop kdegraphics/kfile-plugins/gif/kfile_gif.desktop
--- kdegraphics.orig/kfile-plugins/gif/kfile_gif.desktop	Mon Mar 29 15:01:19 2004
+++ kdegraphics/kfile-plugins/gif/kfile_gif.desktop	Mon May  3 15:03:42 2004
@@ -3,6 +3,7 @@
 Type=Service
 Name=GIF Info
 Name[af]=Gif Inligting
+Name[ar]=معلومات GIF
 Name[ca]=Informació de GIF
 Name[cs]=GIF info
 Name[cy]=Gwybodaeth GIF
diff -urN -x CVS kdegraphics.orig/kfile-plugins/ico/kfile_ico.desktop kdegraphics/kfile-plugins/ico/kfile_ico.desktop
--- kdegraphics.orig/kfile-plugins/ico/kfile_ico.desktop	Mon Mar 29 15:01:21 2004
+++ kdegraphics/kfile-plugins/ico/kfile_ico.desktop	Mon May  3 15:03:44 2004
@@ -3,6 +3,7 @@
 Type=Service
 Name=ICO Info
 Name[af]=Ico Inligting
+Name[ar]=معلومات ICO
 Name[ca]=Informació d'ICO
 Name[cs]=ICO info
 Name[cy]=Gwybodaeth ICO
diff -urN -x CVS kdegraphics.orig/kfile-plugins/jpeg/exif.cpp kdegraphics/kfile-plugins/jpeg/exif.cpp
--- kdegraphics.orig/kfile-plugins/jpeg/exif.cpp	Mon Feb  2 00:21:20 2004
+++ kdegraphics/kfile-plugins/jpeg/exif.cpp	Mon Apr 12 19:59:17 2004
@@ -569,7 +569,7 @@
                 // have appropriate aperture information yet.
                 if (ExifData::ApertureFNumber == 0){
                     ExifData::ApertureFNumber
-                        = (float)exp(ConvertAnyFormat(ValuePtr, Format)*log(2)*0.5);
+                        = (float)exp(ConvertAnyFormat(ValuePtr, Format)*log(2.0)*0.5);
                 }
                 break;
 
@@ -596,7 +596,7 @@
                 // this value if we don't already have it from somewhere else.
                 if (ExifData::ExposureTime == 0){
                     ExifData::ExposureTime
-                        = (float)(1/exp(ConvertAnyFormat(ValuePtr, Format)*log(2)));
+                        = (float)(1/exp(ConvertAnyFormat(ValuePtr, Format)*log(2.0)));
                 }
                 break;
 
diff -urN -x CVS kdegraphics.orig/kfile-plugins/jpeg/kfile_jpeg.desktop kdegraphics/kfile-plugins/jpeg/kfile_jpeg.desktop
--- kdegraphics.orig/kfile-plugins/jpeg/kfile_jpeg.desktop	Mon Mar 29 15:01:23 2004
+++ kdegraphics/kfile-plugins/jpeg/kfile_jpeg.desktop	Mon May  3 15:03:46 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Name=JPEG EXIF Info
+Name[ar]=معلومات JPEG EXIF
 Name[ca]=Informació de JPEG EXIF
 Name[cs]=JPEG EXIF info
 Name[cy]=Gwybodaeth JPEG EXIF
diff -urN -x CVS kdegraphics.orig/kfile-plugins/pcx/kfile_pcx.desktop kdegraphics/kfile-plugins/pcx/kfile_pcx.desktop
--- kdegraphics.orig/kfile-plugins/pcx/kfile_pcx.desktop	Mon Mar 29 15:01:25 2004
+++ kdegraphics/kfile-plugins/pcx/kfile_pcx.desktop	Mon May  3 15:03:48 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Name=PCX File Meta Info
+Name[ar]=معلومات ملف PCX
 Name[bs]=PCX meta-podaci
 Name[ca]=Metainformació de fitxer PCX
 Name[cs]=Metainformace obrázku typu PCX
@@ -14,7 +15,7 @@
 Name[fi]=PCX-tiedoston metatiedot
 Name[fr]=Meta Informations sur les fichiers PCX
 Name[gl]=Inf. metaficheiro PCX
-Name[hi]=PCX फाइल मेटा इन्फो
+Name[hi]=PCX फ़ाइल मेटा इन्फो
 Name[hu]=PCX-jellemzők
 Name[is]=PCX File Meta upplýsingar
 Name[it]=Meta-informazioni sui file PCX
diff -urN -x CVS kdegraphics.orig/kfile-plugins/pnm/kfile_pnm.desktop kdegraphics/kfile-plugins/pnm/kfile_pnm.desktop
--- kdegraphics.orig/kfile-plugins/pnm/kfile_pnm.desktop	Mon Mar 29 15:01:31 2004
+++ kdegraphics/kfile-plugins/pnm/kfile_pnm.desktop	Mon May  3 15:03:50 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Name=PNM Info
+Name[ar]=معلومات PNM
 Name[ca]=Informació de PNM
 Name[cs]=PNM info
 Name[cy]=Gybodaeth PNM
diff -urN -x CVS kdegraphics.orig/kfile-plugins/tga/kfile_tga.desktop kdegraphics/kfile-plugins/tga/kfile_tga.desktop
--- kdegraphics.orig/kfile-plugins/tga/kfile_tga.desktop	Mon Mar 29 15:01:35 2004
+++ kdegraphics/kfile-plugins/tga/kfile_tga.desktop	Mon May  3 15:03:52 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Name=Truevision Targa Info
+Name[ar]=معلومات Truevision Targa
 Name[ca]=Informació de Targa Truevision
 Name[cs]=Truevision Targa info
 Name[cy]=Gwybodaeth Truevision Targa
diff -urN -x CVS kdegraphics.orig/kfile-plugins/tiff/kfile_tiff.desktop kdegraphics/kfile-plugins/tiff/kfile_tiff.desktop
--- kdegraphics.orig/kfile-plugins/tiff/kfile_tiff.desktop	Mon Mar 29 15:01:37 2004
+++ kdegraphics/kfile-plugins/tiff/kfile_tiff.desktop	Mon May  3 15:03:54 2004
@@ -3,6 +3,7 @@
 Type=Service
 Name=TIFF File Meta Info
 Name[af]=Tiff Lêer Meta Inligting
+Name[ar]=معلومات ملف TIFF
 Name[ca]=Metainformació de fitxer TIFF
 Name[cs]=Metainformace obrázku typu TIFF
 Name[cy]=Meta-wybodaeth Ffeil TIFF
@@ -16,7 +17,7 @@
 Name[fr]=Meta Informations sur les fichiers TIFF
 Name[gl]=Inf. metaficheiro TIFF
 Name[he]=מידע TIFF
-Name[hi]=TIFF फाइल मेटा इन्फो
+Name[hi]=TIFF फ़ाइल मेटा इन्फो
 Name[hr]=TIFF meta informacije
 Name[hu]=TIFF-metajellemzők
 Name[is]=TIFF File Meta upplýsingar
diff -urN -x CVS kdegraphics.orig/kfile-plugins/xbm/kfile_xbm.desktop kdegraphics/kfile-plugins/xbm/kfile_xbm.desktop
--- kdegraphics.orig/kfile-plugins/xbm/kfile_xbm.desktop	Mon Mar 29 15:01:39 2004
+++ kdegraphics/kfile-plugins/xbm/kfile_xbm.desktop	Mon May  3 15:03:57 2004
@@ -3,6 +3,7 @@
 Type=Service
 Name=XBM Info
 Name[af]=Xbm Inligting
+Name[ar]=معلومات XBM
 Name[ca]=Informació d'XBM
 Name[cs]=XBM info
 Name[cy]=Gwybodaeth XBM
diff -urN -x CVS kdegraphics.orig/kgamma/kcmkgamma/kgamma.desktop kdegraphics/kgamma/kcmkgamma/kgamma.desktop
--- kdegraphics.orig/kgamma/kcmkgamma/kgamma.desktop	Mon Mar 29 15:01:41 2004
+++ kdegraphics/kgamma/kcmkgamma/kgamma.desktop	Mon May  3 15:03:59 2004
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Comment=A monitor calibration tool
+Comment[ar]=أداة مراقبة وتعيير
 Comment[bg]=Калибриране на монитора
 Comment[bs]=Alat za kalibraciju monitora
 Comment[ca]=Una eina de calibració del monitor
@@ -38,6 +39,7 @@
 Comment[zh_CN]=监视器校准工具
 
 Name=Monitor Gamma
+Name[ar]=مراقبة جاما
 Name[bs]=Monitor gamma
 Name[ca]=Gamma del monitor
 Name[cs]=Gamma monitoru
diff -urN -x CVS kdegraphics.orig/kghostview/infodialog.cpp kdegraphics/kghostview/infodialog.cpp
--- kdegraphics.orig/kghostview/infodialog.cpp	Sun Jun 15 23:02:30 2003
+++ kdegraphics/kghostview/infodialog.cpp	Sun Apr 11 11:12:08 2004
@@ -33,7 +33,7 @@
 //
 InfoDialog::InfoDialog( QWidget *parent, const char *name, bool modal )
   :KDialogBase( i18n("Document Information"), Yes, Yes, Yes, parent,
-		name, modal, true, i18n("&OK") )
+		name, modal, true, KStdGuiItem::ok() )
 {
   QFrame *page = makeMainWidget();
   QVBoxLayout *topLayout = new QVBoxLayout( page, 0, spacingHint() );
@@ -70,13 +70,13 @@
      *      mm minute
      *      SS second
      *      O is "+" or "-"
-     *      HH is hour 
+     *      HH is hour
      *      mm is minute
-     *      
+     *
      *      OHH'mm' form together the desviation to UCT time ( the timezone ).
      *      Everything after the YYYY is optional.
      *      The D: is "highly recommended", but legally optional
-     *      
+     *
      * For PS files, there is no such standard and dates appear
      * in any format they desire.
      */
diff -urN -x CVS kdegraphics.orig/kghostview/kdscerrordialog.cpp kdegraphics/kghostview/kdscerrordialog.cpp
--- kdegraphics.orig/kghostview/kdscerrordialog.cpp	Sun Jul 27 02:25:15 2003
+++ kdegraphics/kghostview/kdscerrordialog.cpp	Wed Apr 14 19:47:55 2004
@@ -1,6 +1,6 @@
 /**
  * Copyright (C) 2001 the KGhostView authors. See file AUTHORS.
- * 	
+ *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation; either version 2 of the License, or
@@ -18,17 +18,18 @@
 
 #include <qlabel.h>
 #include <qlayout.h>
-#include <qpushbutton.h>
 #include <qtextedit.h>
 
 #include <kdebug.h>
 #include <klocale.h>
 #include <kseparator.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include "kdscerrordialog.h"
 #include "kdscerrordialog.moc"
 
-KDSCErrorThreshold::KDSCErrorThreshold( int threshold, 
+KDSCErrorThreshold::KDSCErrorThreshold( int threshold,
                                         KDSCErrorHandler* errorHandler ) :
     _threshold( threshold ),
     _errorHandler( errorHandler )
@@ -39,25 +40,25 @@
     if( _errorHandler != 0 && err.severity() >= _threshold )
 	return _errorHandler->error( err );
     else
-	// Cancel is the default handling strategy for dsc_error_fn, so 
-	// we keep it. This cancels error handling and *not* document 
-	// parsing! 
+	// Cancel is the default handling strategy for dsc_error_fn, so
+	// we keep it. This cancels error handling and *not* document
+	// parsing!
 	return Cancel;
 }
-    
+
 KDSCErrorDialog::KDSCErrorDialog( QWidget* parent ) :
     KDialog( parent, "dscerrordialog", true ),
     _response( Ok )
 {
     QVBoxLayout* vbox = new QVBoxLayout( this, marginHint(), spacingHint() );
-    
+
     _lineNumberLabel = new QLabel( this );
     vbox->addWidget( _lineNumberLabel );
-    
+
     _lineLabel = new QTextEdit( this );
     _lineLabel->setReadOnly( true );
     vbox->addWidget( _lineLabel );
-    
+
     _descriptionLabel = new QLabel( this );
     vbox->addWidget( _descriptionLabel );
 
@@ -67,17 +68,17 @@
     QHBoxLayout* hbox = new QHBoxLayout( vbox );
 
     hbox->addStretch();
-    
-    _okButton = new QPushButton( "Ok", this );
+
+    _okButton = new KPushButton( KStdGuiItem::ok(), this );
     hbox->addWidget( _okButton );
-    _cancelButton = new QPushButton( "Cancel", this );
+    _cancelButton = new KPushButton( KStdGuiItem::cancel(), this );
     hbox->addWidget( _cancelButton );
-    _ignoreAllButton = new QPushButton( "IgnoreAll", this );
+    _ignoreAllButton = new QPushButton( i18n("Ignore All"), this );
     hbox->addWidget( _ignoreAllButton );
 
     connect( _okButton,     SIGNAL( clicked() ), this, SLOT( slotOk()     ) );
     connect( _cancelButton, SIGNAL( clicked() ), this, SLOT( slotCancel() ) );
-    connect( _ignoreAllButton, SIGNAL( clicked() ), 
+    connect( _ignoreAllButton, SIGNAL( clicked() ),
              this, SLOT( slotIgnoreAll() ) );
 }
 
@@ -86,24 +87,24 @@
     switch( err.severity() )
     {
     case KDSCError::Information:
-	setCaption( i18n( "DSC Information" ) ); 
+	setCaption( i18n( "DSC Information" ) );
 	break;
     case KDSCError::Warning:
-	setCaption( i18n( "DSC Warning" ) ); 
+	setCaption( i18n( "DSC Warning" ) );
 	break;
     case KDSCError::Error:
-	setCaption( i18n( "DSC Error" ) ); 
+	setCaption( i18n( "DSC Error" ) );
 	break;
     }
 
     _lineNumberLabel->setText( i18n( "On line %1:" ).arg( err.lineNumber() ) );
     _lineLabel->setText( err.line() );
     _descriptionLabel->setText( description( err.type() ) );
-    
+
     exec();
 
     kdDebug(4500) << "KDSCErrorDialog: returning " << _response << endl;
-	
+
     return _response;
 }
 
@@ -111,40 +112,40 @@
 {
     switch( type )
     {
-    case KDSCError::BBox: 
-	return "TODO"; 
-    case KDSCError::EarlyTrailer: 
-	return "TODO"; 
-    case KDSCError::EarlyEOF: 
-	return "TODO"; 
-    case KDSCError::PageInTrailer: 
-	return "TODO"; 
-    case KDSCError::PageOrdinal: 
-	return "TODO"; 
-    case KDSCError::PagesWrong: 
-	return "TODO"; 
-    case KDSCError::EPSNoBBox: 
-	return "TODO"; 
-    case KDSCError::EPSPages: 
-	return "TODO"; 
-    case KDSCError::NoMedia: 
-	return "TODO"; 
-    case KDSCError::AtEnd: 
-	return "TODO"; 
-    case KDSCError::DuplicateComment: 
-	return "TODO"; 
-    case KDSCError::DuplicateTrailer: 
-	return "TODO"; 
-    case KDSCError::BeginEnd: 
-	return "TODO"; 
-    case KDSCError::BadSection: 
-	return "TODO"; 
-    case KDSCError::LongLine: 
+    case KDSCError::BBox:
+	return "TODO";
+    case KDSCError::EarlyTrailer:
+	return "TODO";
+    case KDSCError::EarlyEOF:
+	return "TODO";
+    case KDSCError::PageInTrailer:
+	return "TODO";
+    case KDSCError::PageOrdinal:
+	return "TODO";
+    case KDSCError::PagesWrong:
+	return "TODO";
+    case KDSCError::EPSNoBBox:
+	return "TODO";
+    case KDSCError::EPSPages:
+	return "TODO";
+    case KDSCError::NoMedia:
+	return "TODO";
+    case KDSCError::AtEnd:
+	return "TODO";
+    case KDSCError::DuplicateComment:
+	return "TODO";
+    case KDSCError::DuplicateTrailer:
+	return "TODO";
+    case KDSCError::BeginEnd:
+	return "TODO";
+    case KDSCError::BadSection:
+	return "TODO";
+    case KDSCError::LongLine:
 	return i18n( "Lines in DSC documents must be shorter than 255 "
-	             "characters." ); 
-    case KDSCError::IncorrectUsage: 
-	return "TODO"; 
-    default: return "TODO"; 
+	             "characters." );
+    case KDSCError::IncorrectUsage:
+	return "TODO";
+    default: return "TODO";
     }
 }
 
@@ -156,13 +157,13 @@
 
 void KDSCErrorDialog::slotCancel()
 {
-    _response = Cancel;   
+    _response = Cancel;
     accept();
 }
 
 void KDSCErrorDialog::slotIgnoreAll()
 {
-    _response = IgnoreAll;   
+    _response = IgnoreAll;
     accept();
 }
 
diff -urN -x CVS kdegraphics.orig/kghostview/kghostview.desktop kdegraphics/kghostview/kghostview.desktop
--- kdegraphics.orig/kghostview/kghostview.desktop	Fri Mar  5 13:51:09 2004
+++ kdegraphics/kghostview/kghostview.desktop	Mon May  3 15:04:01 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=KGhostView
 Name[af]=Kghostview
+Name[ar]=برنامج KGhostView
 Name[eo]=Postskriptrigardilo
 Name[hi]=के-घोस्ट-व्यू
 Name[hu]=KGhostview
diff -urN -x CVS kdegraphics.orig/kiconedit/kiconedit.desktop kdegraphics/kiconedit/kiconedit.desktop
--- kdegraphics.orig/kiconedit/kiconedit.desktop	Mon Mar 29 15:01:43 2004
+++ kdegraphics/kiconedit/kiconedit.desktop	Mon May  3 15:04:03 2004
@@ -8,7 +8,7 @@
 GenericName[ar]=محرر الإيقونات
 GenericName[bg]=Редактор на икони
 GenericName[bs]=Editor ikona
-GenericName[ca]=Editor de icones
+GenericName[ca]=Editor d'icones
 GenericName[cs]=Editor ikon
 GenericName[cy]=Golygydd Eiconau
 GenericName[da]=Ikoneditor
@@ -60,6 +60,7 @@
 GenericName[zu]=Umhleli we Icon
 Name=KIconEdit
 Name[af]=K-ikoon-redigeer
+Name[ar]=برنامج KIconEdit
 Name[cy]=KGolyguEicon
 Name[eo]=Piktogramredaktilo
 Name[hi]=के-आइकॉन-एडिट
diff -urN -x CVS kdegraphics.orig/kmrml/kmrml/kcontrol/kcmkmrml.cpp kdegraphics/kmrml/kmrml/kcontrol/kcmkmrml.cpp
--- kdegraphics.orig/kmrml/kmrml/kcontrol/kcmkmrml.cpp	Sun Jul 27 02:25:15 2003
+++ kdegraphics/kmrml/kmrml/kcontrol/kcmkmrml.cpp	Wed Apr 14 19:47:55 2004
@@ -63,7 +63,7 @@
 {
     if (KMessageBox::warningContinueCancel(this,
          i18n("Do you really want the configuration to be reset "
-              "to the defaults?"), i18n("Are you sure?"), i18n("Continue"))
+              "to the defaults?"), i18n("Are you sure?"), KStdGuiItem::cont())
         != KMessageBox::Continue)
         return;
 
diff -urN -x CVS kdegraphics.orig/kmrml/kmrml/kcontrol/kcmkmrml.desktop kdegraphics/kmrml/kmrml/kcontrol/kcmkmrml.desktop
--- kdegraphics.orig/kmrml/kmrml/kcontrol/kcmkmrml.desktop	Sun Feb 29 16:27:29 2004
+++ kdegraphics/kmrml/kmrml/kcontrol/kcmkmrml.desktop	Mon May  3 15:04:05 2004
@@ -8,9 +8,10 @@
 X-KDE-Library=kmrml
 
 Name=Image Index
+Name[ar]=فهرس الصور
 Name[bg]=Графичен индекс
 Name[bs]=Indeks slika
-Name[ca]=Índex de imatge
+Name[ca]=Índex d'imatge
 Name[cs]=Rejstřík obrázků
 Name[cy]=Mynegai Delweddau
 Name[da]=Billedindeks
@@ -22,7 +23,7 @@
 Name[fr]=Index des images
 Name[gl]=Índice imaxe
 Name[he]=אינדקס תמונות
-Name[hi]=चित्र निर्देशिका
+Name[hi]=छवि निर्देशिका
 Name[hu]=Képindexelő
 Name[is]=Myndayfirlit
 Name[it]=Indice di immagini
@@ -56,9 +57,10 @@
 Name[zu]=Isiqalo Sesithombe
 
 Comment=Configuration for using the GNU Image Finding Tool
+Comment[ar]=اعدادات لاستخدام أداة GNU للبحث عن الصور
 Comment[bg]=Програма за индексиране и търсене на изображения
 Comment[bs]=Podešavanje za upotrebu GNU Alata za pronalaženje slika
-Comment[ca]=Configuració per a l'ús de l'eina de cerca de imatges GNU
+Comment[ca]=Configuració per a l'ús de l'eina de cerca d'imatges GNU
 Comment[cs]=Konfigurace používání nástroje GNU Image Finding Tool
 Comment[cy]=Ffurfweddiad am ddefnyddio'r Erfyn Canfod Delweddau GNU
 Comment[da]=Indstilling for brug af GNU Image Finding Tool
@@ -70,7 +72,7 @@
 Comment[fr]=Configuration pour l'utilisation du GNU Image Finding Tool
 Comment[gl]=Configuración para empregar "GNU Image Finding Tool"
 Comment[he]=שינוי הגדרות כלי חיפוש התמונות של GNU
-Comment[hi]=ग्नू चित्र खोज उपकरण को उपयोग करने हेतु कॉन्फिगरेशन
+Comment[hi]=ग्नू छवि खोज उपकरण को उपयोग करने हेतु कॉन्फ़िगरेशन
 Comment[hu]=A GNU képkereső program beállításai
 Comment[is]=Stillingar til þess að nota GNU myndleitartólið
 Comment[it]=Configurazione dello strumento per la ricerca delle immagini
@@ -102,6 +104,7 @@
 Comment[zu]=Inhlanganiselo yokusebenzisa Ithuluzi Lokuthola Isithombe se-GNU
 
 Keywords=Images,Search,Query,Find,Gift,kmrml,mrml,CBIR
+Keywords[ar]=صور,بحث,استعلام,Find,Gift,kmrml,mrml,CBIR
 Keywords[bg]=изображения,търсене,заявка,картинка,картинки,снимки,Images,Search,Query,Find,Gift,kmrml,mrml,CBIR
 Keywords[bs]=Images,Search,Query,Find,Gift,kmrml,mrml,CBIR,slike,pretraga,upit,nađi
 Keywords[ca]=Imatges,Cerca,Consulta,Busca,Gift,kmrml,mrml,CBIR
@@ -116,7 +119,7 @@
 Keywords[fr]=Images,Recherche,Requête,Chercher,Gift,kmrml,mrml,CBIR
 Keywords[gl]=Images,Search,Query,Find,Gift,kmrml,mrml,CBIR, Buscar, Procurar
 Keywords[he]=תמונות,חיפוש,שאילתה,Gift,kmrml,mrml,CBIR
-Keywords[hi]=चित्र, खोज,ढूंढ,तलाश,उपहार,केएमआरएमएल,एमआरएमएल,सीबीआईआर
+Keywords[hi]=छवि, खोज,ढूंढ,तलाश,उपहार,केएमआरएमएल,एमआरएमएल,सीबीआईआर
 Keywords[hu]=képek,keresés,lekérdezés,találat,Gift,kmrml,mrml,CBIR
 Keywords[it]=Immagini,Ricerca,Trovare,Gift,kmrml,mrml,CBIR
 Keywords[ja]=画像,検索,クエリ,検索,Gift,kmrml,mrml,CBIR
diff -urN -x CVS kdegraphics.orig/kmrml/kmrml/mrml-servicemenu.desktop kdegraphics/kmrml/kmrml/mrml-servicemenu.desktop
--- kdegraphics.orig/kmrml/kmrml/mrml-servicemenu.desktop	Mon Mar 29 15:01:45 2004
+++ kdegraphics/kmrml/kmrml/mrml-servicemenu.desktop	Mon May  3 15:04:07 2004
@@ -5,6 +5,7 @@
 [Desktop Action search]
 Name=Search for Similar Images...
 Name[af]=Soektog vir Soortgelyk Beelde...
+Name[ar]=بحث عن الصور المتشابهه...
 Name[bg]=Търсене за подобни изображения...
 Name[bs]=Traženje sličnih slika...
 Name[ca]=Cerca imatges similars...
@@ -19,7 +20,7 @@
 Name[fr]=Recherche d'images semblables...
 Name[gl]=Procurar imaxe semellantes...
 Name[he]=חיפוש תמונות דומות...
-Name[hi]=एक जैसे चित्रों हेतु ढूंढें...
+Name[hi]=एक जैसे छविों हेतु ढूंढें...
 Name[hu]=Ehhez hasonló képek keresése...
 Name[is]=Leita að svipuðum myndum...
 Name[it]=Cerca immagini simili...
diff -urN -x CVS kdegraphics.orig/kmrml/kmrml/mrml.desktop kdegraphics/kmrml/kmrml/mrml.desktop
--- kdegraphics.orig/kmrml/kmrml/mrml.desktop	Sun Feb 29 16:27:33 2004
+++ kdegraphics/kmrml/kmrml/mrml.desktop	Mon May  3 15:04:09 2004
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Comment=Multimedia Retrieval Markup Language Document
 Comment[af]=Multimedia Onttrekking Opmerk Taal Dokument
+Comment[ar]=مستند لغة ترميز استرجاع الوسائط المتعددة
 Comment[bs]=Multimedia Retrieval Markup Language dokument
 Comment[ca]=Document de llenguatge de marcatge de recuperació multimèdia
 Comment[cs]=Multimedia Retrieval Markup Language dokument
@@ -13,7 +14,7 @@
 Comment[fi]=Multimedianhakuasiakirja
 Comment[fr]=Document en langage Multimedia Retrieval Markup
 Comment[he]=מסמך שפת סימון לאיחזור מולטימדיה
-Comment[hi]=मल्टीमीडिया रिट्राइवल मार्कअप लैंग्वेज दस्तावेज
+Comment[hi]=मल्टीमीडिया रिट्राइवल मार्कअप लैंग्वेज दस्तावेज़
 Comment[hu]=MRML (Multimedia Retrieval Markup Language) dokumentum
 Comment[is]=Multimedia Retrieval Markup Language skjal
 Comment[it]=Documento MRML
diff -urN -x CVS kdegraphics.orig/kmrml/kmrml/mrml_part.desktop kdegraphics/kmrml/kmrml/mrml_part.desktop
--- kdegraphics.orig/kmrml/kmrml/mrml_part.desktop	Sun Feb 29 16:27:35 2004
+++ kdegraphics/kmrml/kmrml/mrml_part.desktop	Mon May  3 15:04:11 2004
@@ -2,6 +2,7 @@
 Type=Service
 Exec=blahfoo
 Name=MRML View
+Name[ar]=برنامج MRML View
 Name[ca]=Vista MRML
 Name[cs]=MRML pohled
 Name[cy]=Gwelydd MRML
diff -urN -x CVS kdegraphics.orig/kmrml/kmrml/server/daemonwatcher.desktop kdegraphics/kmrml/kmrml/server/daemonwatcher.desktop
--- kdegraphics.orig/kmrml/kmrml/server/daemonwatcher.desktop	Thu Mar 18 14:49:30 2004
+++ kdegraphics/kmrml/kmrml/server/daemonwatcher.desktop	Mon May  3 15:04:13 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Type=Service
 Name=KDED KMRML Daemon Watcher
+Name[ar]=مراقب KDED KMRML Daemon
 Name[bs]=KDED KMRML nadzor demona
 Name[ca]=Dimoni vigilant KDED KMRL
 Name[cs]=Sledovač KMRML démonů
@@ -36,6 +37,7 @@
 Name[uk]=Спостерігач демону KDED KMRML
 Name[zh_CN]=KDED KMRML 守护程序监视器
 Comment=Starts daemons on demand and restarts them on failure.
+Comment[ar]=يشغَل daemons عند الطلب ويعيد تشغيلهم عند الفشل.
 Comment[bg]=Стартиране на демоните при заявка и рестартиране на демони при грешка.
 Comment[bs]=Pokreće demone po potrebi i restartuje ih ako se sruše.
 Comment[ca]=Engega els dimonis sota petició i els torna a engegar si fallen.
diff -urN -x CVS kdegraphics.orig/kooka/kooka.desktop kdegraphics/kooka/kooka.desktop
--- kdegraphics.orig/kooka/kooka.desktop	Mon Mar 29 15:01:47 2004
+++ kdegraphics/kooka/kooka.desktop	Mon May  3 15:04:15 2004
@@ -60,7 +60,7 @@
 GenericName[zh_TW]=掃描和文字辦識程式
 GenericName[zu]=Scan & OCR Iprogremu
 Name=Kooka
-Name[ar]=كووكا
+Name[ar]=برنامج Kooka
 Name[eo]=Kokao
 Name[hi]=कूका
 Name[is]=Skanni
diff -urN -x CVS kdegraphics.orig/kpaint/kpaint.desktop kdegraphics/kpaint/kpaint.desktop
--- kdegraphics.orig/kpaint/kpaint.desktop	Mon Mar 29 15:01:49 2004
+++ kdegraphics/kpaint/kpaint.desktop	Mon May  3 15:04:17 2004
@@ -22,8 +22,9 @@
 GenericName[fr]=Petit programme de dessin
 GenericName[gl]=Programa de debuxo
 GenericName[he]=תוכנית ציור
-GenericName[hi]=चित्र बनाने का प्रोग्राम
+GenericName[hi]=छवि बनाने का प्रोग्राम
 GenericName[hr]=Program za slikanje
+GenericName[hsb]=Molowanski program
 GenericName[hu]=Rajzolóprogram
 GenericName[is]=Teikniforrit
 GenericName[it]=Programma di disegno
@@ -64,7 +65,7 @@
 Type=Application
 Name=KPaint
 Name[af]=K-verf
-Name[ar]=الرسام
+Name[ar]=برنامج KPaint
 Name[cy]=KPeintio
 Name[el]=Ζωγραφική
 Name[eo]=Pentrilo
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/QOutputDev.cpp kdegraphics/kpdf/kpdf/QOutputDev.cpp
--- kdegraphics.orig/kpdf/kpdf/QOutputDev.cpp	Thu Feb 19 16:14:03 2004
+++ kdegraphics/kpdf/kpdf/QOutputDev.cpp	Fri Apr 23 22:48:02 2004
@@ -59,8 +59,8 @@
 
 #include "QOutputDev.moc"
 
-#define QPDFDBG(x) x		// special debug mode
-//#define QPDFDBG(x)   		// normal compilation
+//#define QPDFDBG(x) x		// special debug mode
+#define QPDFDBG(x)   		// normal compilation
 
 //------------------------------------------------------------------------
 // Constants and macros
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/QOutputDevPixmap.cpp kdegraphics/kpdf/kpdf/QOutputDevPixmap.cpp
--- kdegraphics.orig/kpdf/kpdf/QOutputDevPixmap.cpp	Fri Jan  2 22:10:39 2004
+++ kdegraphics/kpdf/kpdf/QOutputDevPixmap.cpp	Fri Apr 23 22:48:02 2004
@@ -48,8 +48,8 @@
 #include "QOutputDevPixmap.h"
 #include "QOutputDevPixmap.moc"
 
-#define QPDFDBG(x) x		// special debug mode
-//#define QPDFDBG(x)   		// normal compilation
+//#define QPDFDBG(x) x		// special debug mode
+#define QPDFDBG(x)   		// normal compilation
 
 //------------------------------------------------------------------------
 // Constants and macros
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf.desktop kdegraphics/kpdf/kpdf/kpdf.desktop
--- kdegraphics.orig/kpdf/kpdf/kpdf.desktop	Mon Mar 29 15:01:51 2004
+++ kdegraphics/kpdf/kpdf/kpdf.desktop	Mon May  3 15:04:19 2004
@@ -3,6 +3,7 @@
 MimeType=application/pdf
 Terminal=0
 Name=KPDF
+Name[ar]=برنامج KPDF
 Name[hi]=के-पीडीएफ
 Name[xx]=xxKPDFxx
 GenericName=PDF Viewer
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf_pagewidget.cc kdegraphics/kpdf/kpdf/kpdf_pagewidget.cc
--- kdegraphics.orig/kpdf/kpdf/kpdf_pagewidget.cc	Mon Dec 29 05:39:40 2003
+++ kdegraphics/kpdf/kpdf/kpdf_pagewidget.cc	Tue Apr 27 00:44:50 2004
@@ -38,6 +38,7 @@
     PageWidget::setPDFDocument(PDFDoc* doc)
     {
         m_doc = doc;
+        m_currentPage = 1;
         updatePixmap();
     }
 
@@ -121,16 +122,6 @@
             p->fillRect ( clipx, clipy, clipw, cliph, bc );
     }
 
-    void PageWidget::nextPage()
-    {
-        setPage( getPage() + 1);
-    }
-
-    void PageWidget::previousPage()
-    {
-        setPage( getPage() - 1 );
-    }
-
     void PageWidget::zoomTo( double _value )
     {
         if ( m_zoomFactor != _value)
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf_pagewidget.h kdegraphics/kpdf/kpdf/kpdf_pagewidget.h
--- kdegraphics.orig/kpdf/kpdf/kpdf_pagewidget.h	Sun Sep 28 15:44:25 2003
+++ kdegraphics/kpdf/kpdf/kpdf_pagewidget.h	Tue Apr 27 00:44:50 2004
@@ -41,8 +41,6 @@
         void zoomTo( double _value );
 
     public slots:
-        void nextPage();
-        void previousPage();
         void zoomIn();
         void zoomOut();
 
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf_part.cpp kdegraphics/kpdf/kpdf/kpdf_part.cpp
--- kdegraphics.orig/kpdf/kpdf/kpdf_part.cpp	Tue Mar 30 16:55:05 2004
+++ kdegraphics/kpdf/kpdf/kpdf_part.cpp	Wed Apr 28 00:31:10 2004
@@ -53,7 +53,7 @@
 
   pdfpartview = new PDFPartView(parentWidget, widgetName);
 
-  connect(pdfpartview->pagesListBox, SIGNAL( clicked ( QListBoxItem * ) ),
+  connect(pdfpartview->pagesListBox, SIGNAL( currentChanged ( QListBoxItem * ) ),
           this, SLOT( pageClicked ( QListBoxItem * ) ));
 
   m_outputDev = pdfpartview->outputdev;
@@ -93,7 +93,6 @@
                        actionCollection(), "forward");
 #endif
 
-  KStdAction::print( this, SLOT( print() ), actionCollection() );
   KStdAction::printPreview( this, SLOT( printPreview() ), actionCollection() );
 
   m_prevPage = KStdAction::prior(this, SLOT(slotPreviousPage()),
@@ -170,7 +169,7 @@
     if ( m_doc )
     {
         bool ok = false;
-        int num = KInputDialog::getInteger(i18n("Go to Page"), i18n("Page:"), m_currentPage,
+        int num = KInputDialog::getInteger(i18n("Go to Page"), i18n("Page:"), m_currentPage+1,
                                            1, m_doc->getNumPages(), 1, 10, &ok/*, _part->widget()*/);
         if (ok)
             goToPage( num );
@@ -179,9 +178,9 @@
 
 void Part::goToPage( int page )
 {
-    m_currentPage = page;
-    pdfpartview->pagesListBox->setCurrentItem(m_currentPage-1);
-    m_outputDev->setPage(m_currentPage);
+    m_currentPage = page-1;
+    pdfpartview->pagesListBox->setCurrentItem(m_currentPage);
+    m_outputDev->setPage(m_currentPage+1);
     updateActionPage();
 }
 
@@ -204,9 +203,9 @@
     if ( m_doc )
     {
         m_firstPage->setEnabled(m_currentPage!=0);
-        m_lastPage->setEnabled(m_currentPage<m_doc->getNumPages());
+        m_lastPage->setEnabled(m_currentPage<m_doc->getNumPages()-1);
         m_prevPage->setEnabled(m_currentPage!=0);
-        m_nextPage->setEnabled(m_currentPage<m_doc->getNumPages());
+        m_nextPage->setEnabled(m_currentPage<m_doc->getNumPages()-1);
     }
     else
     {
@@ -273,9 +272,9 @@
 {
     if ( m_doc && m_doc->getNumPages() > 0 );
     {
-        m_currentPage = m_doc->getNumPages();
-        m_outputDev->setPage(m_currentPage);
-        pdfpartview->pagesListBox->setCurrentItem(m_currentPage-1);
+        m_currentPage = m_doc->getNumPages()-1;
+        m_outputDev->setPage(m_currentPage+1);
+        pdfpartview->pagesListBox->setCurrentItem(m_currentPage);
         updateActionPage();
     }
 }
@@ -294,12 +293,16 @@
 
 bool Part::nextPage()
 {
-    m_currentPage = pdfpartview->pagesListBox->currentItem() + 1;
-    if ( m_doc && m_currentPage >= m_doc->getNumPages())
+    m_currentPage++;
+    
+    if (m_doc && m_currentPage >= m_doc->getNumPages())
+    {
+        m_currentPage--;
         return false;
+    }
 
+    m_outputDev->setPage(m_currentPage+1);
     pdfpartview->pagesListBox->setCurrentItem(m_currentPage);
-    m_outputDev->nextPage();
     updateActionPage();
     return true;
 }
@@ -316,12 +319,16 @@
 
 bool Part::previousPage()
 {
-    m_currentPage = pdfpartview->pagesListBox->currentItem() - 1;
-    if ( m_currentPage  < 0)
+    m_currentPage--;
+    
+    if (m_currentPage < 0)
+    {
+        m_currentPage++;
         return false;
+    }
 
-    pdfpartview->pagesListBox->setCurrentItem(m_currentPage );
-    m_outputDev->previousPage();
+    m_outputDev->setPage(m_currentPage+1);
+    pdfpartview->pagesListBox->setCurrentItem(m_currentPage);
     updateActionPage();
     return true;
 }
@@ -641,8 +648,9 @@
 {
     if ( !qbi )
         return;
-    m_currentPage = pdfpartview->pagesListBox->index(qbi)+1;
-    m_outputDev->setPage(m_currentPage);
+    m_currentPage = pdfpartview->pagesListBox->index(qbi);
+    
+    m_outputDev->setPage(m_currentPage+1);
     updateActionPage();
 }
 
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf_part.desktop kdegraphics/kpdf/kpdf/kpdf_part.desktop
--- kdegraphics.orig/kpdf/kpdf/kpdf_part.desktop	Sun Feb 29 16:27:45 2004
+++ kdegraphics/kpdf/kpdf/kpdf_part.desktop	Mon May  3 15:04:21 2004
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=KPDFPart
+Name[ar]=برنامج KPDFPart
 Name[cs]=Komponenta KPDF
 Name[hi]=के-पीडीएफ-पार्ट
 Name[pt_BR]=Componente do KPDF
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf_part.rc kdegraphics/kpdf/kpdf/kpdf_part.rc
--- kdegraphics.orig/kpdf/kpdf/kpdf_part.rc	Tue Mar 30 16:55:05 2004
+++ kdegraphics/kpdf/kpdf/kpdf_part.rc	Wed Apr 28 00:31:10 2004
@@ -1,10 +1,9 @@
 <!DOCTYPE kpartgui SYSTEM "kpartgui.dtd">
-<kpartgui name="kpdf_part" version="13">
+<kpartgui name="kpdf_part" version="14">
 <MenuBar>
   <Menu name="file"><text>&amp;File</text>
     <Action name="file_print" group="file_print"/>
     <Action name="file_print_preview" group="file_print"/>
-    <Separator group="file_print"/>
   </Menu>
   <Menu name="view"><text>&amp;View</text>
     <Action name="fit_to_width"/>
diff -urN -x CVS kdegraphics.orig/kpdf/kpdf/kpdf_shell.cpp kdegraphics/kpdf/kpdf/kpdf_shell.cpp
--- kdegraphics.orig/kpdf/kpdf/kpdf_shell.cpp	Sun Feb  1 17:07:11 2004
+++ kdegraphics/kpdf/kpdf/kpdf_shell.cpp	Wed Apr 28 00:31:10 2004
@@ -50,9 +50,6 @@
   // set the shell's ui resource file
   setXMLFile("kpdf_shell.rc");
 
-  // then, setup our actions
-  setupActions();
-
   // this routine will find and load our Part.  it finds the Part by
   // name which is a bad idea usually.. but it's alright in this
   // case since our Part is made for this Shell
@@ -65,6 +62,8 @@
                 factory->create(this, "kpdf_part", "KParts::ReadOnlyPart"));
     if (m_part)
     {
+      // then, setup our actions
+      setupActions();
       // tell the KParts::MainWindow that this is indeed the main widget
       setCentralWidget(m_part->widget());
       // and integrate the part's GUI with the shell's
@@ -122,6 +121,7 @@
   recent = KStdAction::openRecent( this, SLOT( openURL( const KURL& ) ),
 				    actionCollection() );
   KStdAction::saveAs(this, SLOT(fileSaveAs()), actionCollection());
+  KStdAction::print(m_part, SLOT(print()), actionCollection());
   KStdAction::quit( this, SLOT(slotQuit()), actionCollection());
 
   //createStandardStatusBarAction();
diff -urN -x CVS kdegraphics.orig/kpovmodeler/kpovmodeler.desktop kdegraphics/kpovmodeler/kpovmodeler.desktop
--- kdegraphics.orig/kpovmodeler/kpovmodeler.desktop	Sun Feb 29 16:27:48 2004
+++ kdegraphics/kpovmodeler/kpovmodeler.desktop	Mon May  3 15:04:23 2004
@@ -6,7 +6,7 @@
 MimeType=application/x-kpovmodeler;
 Name=Povray Modeler
 Name[af]=Povray Moduleerder
-Name[ar]=مصمم Povray
+Name[ar]=مصمم ملفات Povray
 Name[az]=Povrey Modelyer
 Name[ca]=Modelador Povray
 Name[cs]=Modelář Povray
diff -urN -x CVS kdegraphics.orig/kpovmodeler/pmerrordialog.cpp kdegraphics/kpovmodeler/pmerrordialog.cpp
--- kdegraphics.orig/kpovmodeler/pmerrordialog.cpp	Wed Mar 19 20:50:19 2003
+++ kdegraphics/kpovmodeler/pmerrordialog.cpp	Wed Apr 14 19:47:55 2004
@@ -36,13 +36,13 @@
    QVBox* page = makeVBoxMainWidget( );
    QLabel* text = new QLabel( QString( "" ), ( QWidget* )page );
 
-   
+
    m_pTextView = new QTextEdit( ( QWidget* )page );
    m_pTextView->setReadOnly( true );
 
    m_messageDict.setAutoDelete( true );
    m_messages.setAutoDelete( true );
-   
+
    PMMessageList::ConstIterator it;
    for( it = messages.begin( ); it != messages.end( ); ++it )
       m_messages.append( new PMMessage( *it ) );
@@ -70,7 +70,7 @@
          }
       }
    }
-   
+
    displayMessages( );
 
    if( errorFlags & PMEWarning )
@@ -82,16 +82,16 @@
    }
    else
       text->setText( i18n( "There were errors:" ) );
-   
-   setButtonOKText( i18n( "OK" ),
+
+   setButtonOKText( KStdGuiItem::ok().text(),
                     i18n( "Proceed" ),
                     i18n( "When clicking <b>Proceed</b>, the program\n"
                           "will try to proceed with the current action." ) );
-   setButtonCancelText( i18n( "Cancel" ),
+   setButtonCancelText( KStdGuiItem::cancel().text(),
                         i18n( "Cancel" ),
                         i18n( "When clicking <b>Cancel<b>, the program\n"
                               "will cancel the current action." ) );
-      
+
    if( errorFlags & PMEFatal )
       showButtonOK( false );
    else
@@ -106,12 +106,12 @@
    QString text;
 
    text = "<qt>\n";
-   
+
    for( ; pit.current( ); ++pit )
       text += "<p>" + pit.current( )->text( ) + "</p>\n";
 
    text += "</qt>";
-   
+
    m_pTextView->setText( text );
 }
 
diff -urN -x CVS kdegraphics.orig/kpovmodeler/pmlinkedit.cpp kdegraphics/kpovmodeler/pmlinkedit.cpp
--- kdegraphics.orig/kpovmodeler/pmlinkedit.cpp	Mon Dec 30 20:55:15 2002
+++ kdegraphics/kpovmodeler/pmlinkedit.cpp	Wed Apr 14 19:47:55 2004
@@ -17,11 +17,12 @@
 
 #include "pmlinkedit.h"
 #include <qlineedit.h>
-#include <qpushbutton.h>
 #include <qlabel.h>
 #include <qlayout.h>
 #include <klocale.h>
 #include <kdialog.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include "pmdeclare.h"
 #include "pmobjectselect.h"
@@ -55,19 +56,19 @@
    m_bReadOnly = false;
 
    QGridLayout* grid = new QGridLayout( this, 2, 2, 0, KDialog::spacingHint( ) );
-   
+
    grid->addWidget( new QLabel( i18n( "Prototype:" ), this ), 0, 0 );
    grid->setColStretch( 0, 0 );
    grid->setColStretch( 1, 1 );
    m_pIDEdit = new QLineEdit( this );
    m_pIDEdit->setReadOnly( true );
    grid->addWidget( m_pIDEdit, 0, 1 );
-   
+
    QHBoxLayout* layout = new QHBoxLayout( );
    grid->addLayout( layout, 1, 1 );
    m_pSelectButton = new QPushButton( i18n( "Select" ), this );
    layout->addWidget( m_pSelectButton );
-   m_pClearButton = new QPushButton( i18n( "Clear" ), this );
+   m_pClearButton = new KPushButton( KStdGuiItem::clear(), this );
    layout->addWidget( m_pClearButton );
 
    connect( m_pSelectButton, SIGNAL( clicked( ) ), SLOT( slotSelectClicked( ) ) );
@@ -124,7 +125,7 @@
       else
          result = PMObjectSelect::selectDeclare(
             m_pDisplayedObject, m_declareTypes, obj, this );
-         
+
       if( ( result == QDialog::Accepted ) && obj )
       {
          m_pDeclare = ( PMDeclare* ) obj;
@@ -132,7 +133,7 @@
          m_pClearButton->setEnabled( true );
          emit dataChanged( );
       }
-   }                                                                 
+   }
 }
 
 void PMLinkEdit::slotClearClicked( )
diff -urN -x CVS kdegraphics.orig/kpovmodeler/pmpovrayoutputwidget.cpp kdegraphics/kpovmodeler/pmpovrayoutputwidget.cpp
--- kdegraphics.orig/kpovmodeler/pmpovrayoutputwidget.cpp	Sat Sep 20 18:27:52 2003
+++ kdegraphics/kpovmodeler/pmpovrayoutputwidget.cpp	Wed Apr 14 19:47:55 2004
@@ -19,11 +19,12 @@
 
 #include <qtextedit.h>
 #include <qlayout.h>
-#include <qpushbutton.h>
 
 #include <klocale.h>
 #include <kglobalsettings.h>
 #include <kconfig.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include "pmdebug.h"
 
@@ -42,7 +43,7 @@
 
    QHBoxLayout* buttonLayout = new QHBoxLayout( topLayout );
    buttonLayout->addStretch( 1 );
-   QPushButton* closeButton = new QPushButton( i18n( "Close" ), this );
+   QPushButton* closeButton = new KPushButton( KStdGuiItem::close(), this );
    buttonLayout->addWidget( closeButton );
    closeButton->setDefault( true );
    connect( closeButton, SIGNAL( clicked( ) ), SLOT( hide( ) ) );
diff -urN -x CVS kdegraphics.orig/kpovmodeler/pmpovraywidget.cpp kdegraphics/kpovmodeler/pmpovraywidget.cpp
--- kdegraphics.orig/kpovmodeler/pmpovraywidget.cpp	Sat Sep 20 18:27:52 2003
+++ kdegraphics/kpovmodeler/pmpovraywidget.cpp	Wed Apr 14 19:47:55 2004
@@ -27,10 +27,11 @@
 #include <ktempfile.h>
 #include <kio/netaccess.h>
 #include <kglobalsettings.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include <qlayout.h>
 #include <qprogressbar.h>
-#include <qpushbutton.h>
 #include <qlabel.h>
 #include <qscrollview.h>
 #include <qtimer.h>
@@ -77,11 +78,11 @@
    buttonLayout->addWidget( m_pPovrayOutputButton );
 
    buttonLayout = new QHBoxLayout( topLayout );
-   m_pSaveButton = new QPushButton( i18n( "Save..." ), this );
+   m_pSaveButton = new KPushButton( KStdGuiItem::saveAs(), this );
    m_pSaveButton->setEnabled( false );
    buttonLayout->addWidget( m_pSaveButton );
    buttonLayout->addStretch( 1 );
-   QPushButton* closeButton = new QPushButton( i18n( "Close" ), this );
+   QPushButton* closeButton = new KPushButton( KStdGuiItem::close(), this );
    buttonLayout->addWidget( closeButton );
 
    connect( m_pRenderWidget, SIGNAL( finished( int ) ),
diff -urN -x CVS kdegraphics.orig/kruler/eventsrc kdegraphics/kruler/eventsrc
--- kdegraphics.orig/kruler/eventsrc	Sun Feb 29 16:27:50 2004
+++ kdegraphics/kruler/eventsrc	Mon May  3 15:04:25 2004
@@ -2,7 +2,7 @@
 IconName=kruler
 Comment=KDE Screen Ruler
 Comment[af]=Kde Skerm Liniaal
-Comment[ar]=مسطرة الشاشة
+Comment[ar]=مسطرة شاشة KDE
 Comment[az]=KDE Ekran Xətkeşi
 Comment[bg]=Линийка за екрана
 Comment[bs]=KDE ekranski linijar
@@ -111,7 +111,7 @@
 Name[zu]=Inyakaziswe Izikhiye Ze Cursor
 Comment=The ruler has moved pixelwise using the cursor keys
 Comment[af]=Die liniaal het verskuif beeldelement-gewys te gebruik Die plekaanduier sleutels
-Comment[ar]=تم تحريك المسطرو بكسليا باستخدام مفاتيح الفأرة
+Comment[ar]=تم تحريك المسطرة بكسليا باستخدام مفاتيح الفأرة
 Comment[az]=Cədvəl piksel piksel istiqamət düymələri ilə hərəkət etdirildi
 Comment[bs]=Linijar se pomjera u pixelima koristeći kursorske tipke
 Comment[ca]=La regla s'ha mogut píxel a píxel usant les tecles de cursor
@@ -127,7 +127,7 @@
 Comment[fr]=La règle s'est déplacée sous l'action des touches fléchées
 Comment[gl]=A regra moverá o "pixelwise" empregando as teclas de cursor
 Comment[he]=הסרגל הוזז באמצעות מקשי החצים
-Comment[hi]=संकेतक कुंजियों की सहायता से रूसर पिक्सलवाइज़ खिसका
+Comment[hi]=संकेतक कुंजियों की मदद से रूसर पिक्सलवाइज़ खिसका
 Comment[hr]=Ravnalo je pomaknuto pomoću tipki za kretanje
 Comment[hu]=A vonalzó elmozgatva képpontonként a kurzorbillentyűkkel
 Comment[is]=Reglustikan hefur verið færð um e-a punkta með örvalyklunum
diff -urN -x CVS kdegraphics.orig/kruler/klineal.cpp kdegraphics/kruler/klineal.cpp
--- kdegraphics.orig/kruler/klineal.cpp	Thu Nov 13 16:49:26 2003
+++ kdegraphics/kruler/klineal.cpp	Mon Apr 12 19:04:25 2004
@@ -157,8 +157,8 @@
   lenMenu->insertItem(i18n("&Tall"), this, SLOT(setTallLength()), CTRL+Key_T);
   lenMenu->insertItem(i18n("&Full Screen Width"), this, SLOT(setFullLength()), CTRL+Key_F);
   mMenu->insertItem(i18n("&Length"), lenMenu);
-  mMenu->insertItem(i18n("&Choose Color..."), this, SLOT(choseColor()), CTRL+Key_C);
-  mMenu->insertItem(i18n("Choose &Font..."), this, SLOT(choseFont()), Key_F);
+  mMenu->insertItem(SmallIcon("colorscm"),i18n("&Choose Color..."), this, SLOT(choseColor()), CTRL+Key_C);
+  mMenu->insertItem(SmallIcon("font"), i18n("Choose &Font..."), this, SLOT(choseFont()), Key_F);
   mMenu->insertSeparator();
   mMenu->insertItem(SmallIcon( "help" ), KStdGuiItem::help().text(), helpMenu());
   mMenu->insertSeparator();
diff -urN -x CVS kdegraphics.orig/kruler/kruler.desktop kdegraphics/kruler/kruler.desktop
--- kdegraphics.orig/kruler/kruler.desktop	Mon Mar 29 15:01:53 2004
+++ kdegraphics/kruler/kruler.desktop	Mon May  3 15:04:27 2004
@@ -52,7 +52,7 @@
 BinaryPattern=
 Name=KRuler
 Name[af]=K-lineaal
-Name[ar]=المسطرة
+Name[ar]=المسطرة (KRuler)
 Name[cy]=KMesurydd
 Name[eo]=Liniilo
 Name[hi]=के-रूलर
diff -urN -x CVS kdegraphics.orig/ksnapshot/ksnapshot.cpp kdegraphics/ksnapshot/ksnapshot.cpp
--- kdegraphics.orig/ksnapshot/ksnapshot.cpp	Sat Sep 20 18:51:42 2003
+++ kdegraphics/ksnapshot/ksnapshot.cpp	Wed Apr 14 19:47:55 2004
@@ -61,9 +61,9 @@
 #define kApp KApplication::kApplication()
 
 KSnapshot::KSnapshot(QWidget *parent, const char *name)
-  : DCOPObject("interface"), 
-    KDialogBase(parent, name, true, QString::null, Help|User1, User1, 
-    true, KGuiItem( i18n( "&Quit" ), "exit" ) )
+  : DCOPObject("interface"),
+    KDialogBase(parent, name, true, QString::null, Help|User1, User1,
+    true, KStdGuiItem::quit() )
 {
     grabber = new QWidget( 0, 0, WStyle_Customize | WX11BypassWM );
     grabber->move( -1000, -1000 );
@@ -152,7 +152,7 @@
     if ( KIO::NetAccess::exists( url, false, this ) ) {
         const QString title = i18n( "File Exists" );
         const QString text = i18n( "<qt>Do you really want to overwrite <b>%1</b>?</qt>" ).arg(url.prettyURL());
-        if (KMessageBox::Yes != KMessageBox::warningYesNoCancel( this, text, title ) ) 
+        if (KMessageBox::Yes != KMessageBox::warningYesNoCancel( this, text, title ) )
         {
             return false;
         }
@@ -389,7 +389,7 @@
         }
     }
 
-    //Rebuild the path 
+    //Rebuild the path
     KURL newURL = filename;
     newURL.setFileName( name );
     setURL( newURL.url() );
@@ -519,7 +519,7 @@
 
 		//Create the bounding box.
 		QRegion bbox(0, 0, snapshot.width(), snapshot.height());
-                
+
                 if( border > 0 ) {
                     contents.translate( border, border );
                     contents += QRegion( 0, 0, border, h );
@@ -527,7 +527,7 @@
                     contents += QRegion( 0, h - border, w, border );
                     contents += QRegion( w - border, 0, border, h );
                 }
-                
+
 		//Get the masked away area.
 		QRegion maskedAway = bbox - contents;
 		QMemArray<QRect> maskedAwayRects = maskedAway.rects();
diff -urN -x CVS kdegraphics.orig/ksvg/core/ksvgrenderer.desktop kdegraphics/ksvg/core/ksvgrenderer.desktop
--- kdegraphics.orig/ksvg/core/ksvgrenderer.desktop	Sun Feb 29 16:28:01 2004
+++ kdegraphics/ksvg/core/ksvgrenderer.desktop	Mon May  3 15:04:29 2004
@@ -4,6 +4,7 @@
 X-KDE-ServiceType=KSVG/Renderer
 X-KDE-Derived=
 Comment=KSVG Rendering Backend
+Comment[ar]=خلفية رسم KSVG
 Comment[bs]=KSVG renderiranje
 Comment[ca]=Representació en segon pla de KSVG
 Comment[cs]=Vykreslovací nástroj KSVG
diff -urN -x CVS kdegraphics.orig/ksvg/impl/SVGWindowImpl.cc kdegraphics/ksvg/impl/SVGWindowImpl.cc
--- kdegraphics.orig/ksvg/impl/SVGWindowImpl.cc	Mon Oct 27 11:17:03 2003
+++ kdegraphics/ksvg/impl/SVGWindowImpl.cc	Sun Apr 11 11:12:08 2004
@@ -170,7 +170,7 @@
 
 bool SVGWindowImpl::confirm(const DOM::DOMString &message, const QString &title)
 {
-	return KMessageBox::warningYesNo(0L, QStyleSheet::convertFromPlainText(message.string()), title, i18n("&OK"), i18n("&Cancel")) == KMessageBox::Yes;
+	return KMessageBox::warningYesNo(0L, QStyleSheet::convertFromPlainText(message.string()), title, KStdGuiItem::ok(), KStdGuiItem::cancel()) == KMessageBox::Yes;
 }
 
 DOM::DOMString SVGWindowImpl::prompt(const DOM::DOMString &message, const DOM::DOMString &_default, const QString &)
diff -urN -x CVS kdegraphics.orig/ksvg/plugin/backends/agg/ksvgaggcanvas.desktop kdegraphics/ksvg/plugin/backends/agg/ksvgaggcanvas.desktop
--- kdegraphics.orig/ksvg/plugin/backends/agg/ksvgaggcanvas.desktop	Sun Feb 29 16:28:03 2004
+++ kdegraphics/ksvg/plugin/backends/agg/ksvgaggcanvas.desktop	Mon May  3 15:04:31 2004
@@ -4,6 +4,7 @@
 X-KDE-Library=libksvgrendereragg
 X-KSVG-InternalName=agg
 Name=KSVG Rendering Backend - Anti Grain Geometry
+Name[ar]=خلفية رسم ل KSVG - هندسة Anti Grain
 Name[bs]=KSVG renderiranje - Anti Grain Geometry
 Name[ca]=Representació en segon pla de KSVG - Geometria anti gra
 Name[cs]=Vykreslovací nástroj KSVG - Anti Grain Geometry
@@ -36,6 +37,7 @@
 Name[xx]=xxKSVG Rendering Backend - Anti Grain Geometryxx
 Name[zh_CN]=KSVG 渲染后端 - 反增长几何形状
 Comment=New, unstable ksvg rendering backend
+Comment[ar]=خلفية الرسم ل ksvg الجديدة والغير مستقرة
 Comment[bs]=Novi, nestabilni ksvg renderiranje backend
 Comment[ca]=Nou, inestable representador en segon pla de ksvg
 Comment[cs]=Nový, nestabilní vykreslovací nástroj KSVG
diff -urN -x CVS kdegraphics.orig/ksvg/plugin/backends/libart/ksvglibartcanvas.desktop kdegraphics/ksvg/plugin/backends/libart/ksvglibartcanvas.desktop
--- kdegraphics.orig/ksvg/plugin/backends/libart/ksvglibartcanvas.desktop	Sun Feb 29 16:28:05 2004
+++ kdegraphics/ksvg/plugin/backends/libart/ksvglibartcanvas.desktop	Mon May  3 15:04:33 2004
@@ -4,8 +4,9 @@
 X-KDE-Library=libksvgrendererlibart
 X-KSVG-InternalName=libart
 Name=KSVG Rendering Backend - Libart
+Name[ar]=خلفية الرسم ل KSVG - Libart
 Name[bs]=KSVG renderiranje - Libart
-Name[ca]=Representació en segon pla de KSVG - Biblioteca de imatges
+Name[ca]=Representació en segon pla de KSVG - Biblioteca d'imatges
 Name[cs]=Vykreslovací nástroj KSVG - Libart
 Name[cy]=Ôl-wyneb Llunio KSVG - Libart
 Name[da]=Bagende til KSVG-visning - Libart
@@ -36,6 +37,7 @@
 Name[xx]=xxKSVG Rendering Backend - Libartxx
 Name[zh_CN]=KSVG 渲染后端 - Libart
 Comment=Mature ksvg rendering backend
+Comment[ar]=خلفية الرسم لksvg البالغة
 Comment[bs]=Zreli ksvg rendering backend
 Comment[ca]=Representador madur en segon pla de ksvg
 Comment[cs]=Vyspělý vykreslovací nástroj KSVG
diff -urN -x CVS kdegraphics.orig/ksvg/plugin/ksvgplugin.desktop kdegraphics/ksvg/plugin/ksvgplugin.desktop
--- kdegraphics.orig/ksvg/plugin/ksvgplugin.desktop	Sun Feb 29 16:28:07 2004
+++ kdegraphics/ksvg/plugin/ksvgplugin.desktop	Mon May  3 15:04:35 2004
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Name=KSVGPlugin
+Name[ar]=برنامج KSVGPlugin
 Name[ca]=Connector KSVG
 Name[cy]=AtegynKSVG
 Name[el]=Πρόσθετο KSVG
diff -urN -x CVS kdegraphics.orig/kuickshow/src/imagewindow.cpp kdegraphics/kuickshow/src/imagewindow.cpp
--- kdegraphics.orig/kuickshow/src/imagewindow.cpp	Sun Jan 11 01:30:10 2004
+++ kdegraphics/kuickshow/src/imagewindow.cpp	Wed Apr 14 19:47:55 2004
@@ -177,12 +177,10 @@
     new KAction( i18n("Print Image..."), KStdAccel::print(),
                  this, SLOT( printImage() ),
                  m_actions, "print_image" );
-    new KAction( i18n("Save As..."), KStdAccel::save(),
-                 this, SLOT( saveImage() ),
+    KStdAction::saveAs( this, SLOT( saveImage() ),
                  m_actions, "save_image_as" );
 
-    new KAction( i18n("Close"), Key_Q,
-                 this, SLOT( close() ),
+    KStdAction::close( this, SLOT( close() ),
                  m_actions, "close_image" );
     // --------
     new KAction( i18n("More Brightness"), Key_B,
@@ -273,7 +271,7 @@
     }
 
     updateCursor();
-	
+
     QString caption = i18n( "Filename (Imagewidth x Imageheight)",
                             "%3 (%1 x %2)" );
     caption = caption.arg( m_kuim->originalWidth() ).
@@ -518,7 +516,7 @@
 
     if ( key == Key_Shift )
         updateCursor( ZoomCursor );
-	
+
     if ( key == Key_Escape || KStdAccel::close().contains( KKey( e ) ) )
         close( true );
     else if ( KStdAccel::save().contains( KKey( e ) ) )
@@ -603,7 +601,7 @@
     }
 
     if ( e->state() & ShiftButton ) {
-	
+
 	if ( !transWidget ) {
 	    transWidget = new QWidget( this );
 	    transWidget->setGeometry( 0, 0, width(), height() );
@@ -616,10 +614,10 @@
  	p.eraseRect( transWidget->rect() );
 	transWidget->show();
 	qApp->processOneEvent();
-	
+
 	int width  = e->x() - xposPress;
 	int height = e->y() - yposPress;
-	
+
 	if ( width < 0 ) {
 	    width = abs( width );
 	    xzoom = e->x();
@@ -682,7 +680,7 @@
     }
 
     if ( yposPress > y ) {
-	topY = y;	
+	topY = y;
 	botY = yposPress;
     }
     else {
diff -urN -x CVS kdegraphics.orig/kuickshow/src/kuickshow.desktop kdegraphics/kuickshow/src/kuickshow.desktop
--- kdegraphics.orig/kuickshow/src/kuickshow.desktop	Mon Mar 29 15:01:55 2004
+++ kdegraphics/kuickshow/src/kuickshow.desktop	Sun Apr 25 01:23:02 2004
@@ -25,7 +25,7 @@
 GenericName[ar]=عارض صور
 GenericName[bg]=Преглед на изображения
 GenericName[bs]=Preglednik slika
-GenericName[ca]=Visualitzador de imatges
+GenericName[ca]=Visualitzador d'imatges
 GenericName[cs]=Prohlížeč obrázků
 GenericName[cy]=Gwelydd Delweddau
 GenericName[da]=Billedfremviser
@@ -40,7 +40,7 @@
 GenericName[ga]=Amharcán Íomhánna
 GenericName[gl]=Visor de imaxes
 GenericName[he]=מציג תמונות
-GenericName[hi]=चित्र प्रदर्शक
+GenericName[hi]=छवि प्रदर्शक
 GenericName[hr]=Preglednik slika
 GenericName[hu]=Képnézegető
 GenericName[is]=Myndaskoðari
diff -urN -x CVS kdegraphics.orig/kview/config/kview.setdlg kdegraphics/kview/config/kview.setdlg
--- kdegraphics.orig/kview/config/kview.setdlg	Mon Mar 29 15:01:57 2004
+++ kdegraphics/kview/config/kview.setdlg	Mon May  3 15:04:37 2004
@@ -1,5 +1,6 @@
 [Viewer]
 Name=Viewer
+Name[ar]=العارض
 Name[bs]=Preglednik
 Name[ca]=Visualitzador
 Name[cs]=Prohlížeč
@@ -37,6 +38,7 @@
 Name[xx]=xxViewerxx
 Name[zh_CN]=查看器
 Comment=This is the configuration for the part of KView that can be reused by other applications (meaning that the settings will also affect the KView part that gets embedded into Konqueror or other applications).
+Comment[ar]=هذه الإعدادات هي لجزء برنامج KView الذي يدمج في داخل متصفح Konqueror والبرامج الأخرى
 Comment[bg]=Конфигуриране на модулите на KView, които могат да бъдат използвани в другите програми. Това означава, че настройките ще имат ефект навсякъде, където се използва съответния модул.
 Comment[bs]=Ovo je konfiguracija dijela KView-a kojeg mogu koristiti druge aplikacije (što znači da će se ove postavke takođe odraziti na KView dio koji se ugrađuje u Konqueror i druge aplikacije).
 Comment[ca]=Aquesta és la configuració de la part de KView que es pot reutilitzar per altres aplicacions (vol dir que els paràmetres també afectaran la part KView que està incrustada dins el Konqueror o altres aplicacions).
@@ -72,7 +74,7 @@
 Weight=1
 [Application]
 Name=Application
-Name[ar]=تطبيق
+Name[ar]=التطبيق
 Name[az]=Proqram
 Name[bg]=Програма
 Name[br]=Arload
@@ -123,6 +125,7 @@
 Name[zh_CN]=应用程序
 Name[zu]=Isicelo
 Comment=Here you can set options that are special for the KView application when running stand-alone. What you change in here will not affect other applications.
+Comment[ar]=هنا يمكنك اختار الاختيارات الخاصة بتطبيق KView المستقل. أي تغيير هنا لن يؤثر على التطبيقات الأخرى.
 Comment[bg]=Конфигуриране на самостоятелната програма KView. Това означава, че настройките ще имат ефект само за програмата KView и останалите програми, които ползват модулите и, няма да бъдат повлияни.
 Comment[bs]=Ovdje možete podesiti opcije koje su specifične za KView samostalnu aplikaciju. Ono što ovdje promijenite se neće odraziti na druge aplikacije.
 Comment[ca]=Aquí podeu establir les opcions específiques de l'aplicació KView quan s'executa autonomament. El que es canviï aquí no afectarà a les altres aplicacions.
diff -urN -x CVS kdegraphics.orig/kview/config/kviewgeneralconfig.desktop kdegraphics/kview/config/kviewgeneralconfig.desktop
--- kdegraphics.orig/kview/config/kviewgeneralconfig.desktop	Fri Mar  5 13:51:20 2004
+++ kdegraphics/kview/config/kviewgeneralconfig.desktop	Mon May  3 15:04:39 2004
@@ -13,7 +13,7 @@
 X-KDE-CfgDlgHierarchy=Application
 
 Name=Application
-Name[ar]=تطبيق
+Name[ar]=التطبيق
 Name[az]=Proqram
 Name[bg]=Програма
 Name[br]=Arload
@@ -64,6 +64,7 @@
 Name[zh_CN]=应用程序
 Name[zu]=Isicelo
 Comment=Configuration for the standalone Version of KView
+Comment[ar]=اعدادات الاصدار المستقل من KView
 Comment[bg]=Настройки на самостоятелната програма KView
 Comment[bs]=Postavke za samostalnu verziju KView
 Comment[ca]=Configuració per a la versió independent de KView
@@ -76,7 +77,7 @@
 Comment[fi]=Asetukset itsenäiselle versiolle KView ohjelmasta
 Comment[fr]=Configuration de la version isolée de KView
 Comment[gl]=Configuración para a versión "standalone" of KView
-Comment[hi]= के-व्यू के स्टैण्डएलोन संस्करण हेतु कॉन्फिगरेशन
+Comment[hi]= के-व्यू के स्टैण्डएलोन संस्करण हेतु कॉन्फ़िगरेशन
 Comment[hu]=A KView önállóan futtatható verziójának beállításai
 Comment[is]=Stillingar fyrir KView
 Comment[it]=Configurazione per KView
diff -urN -x CVS kdegraphics.orig/kview/config/plugins/kviewpluginsconfig.desktop kdegraphics/kview/config/plugins/kviewpluginsconfig.desktop
--- kdegraphics.orig/kview/config/plugins/kviewpluginsconfig.desktop	Fri Mar  5 13:51:22 2004
+++ kdegraphics/kview/config/plugins/kviewpluginsconfig.desktop	Mon May  3 15:04:41 2004
@@ -13,6 +13,7 @@
 X-KDE-CfgDlgHierarchy=Application
 
 Name=Plugins
+Name[ar]=البرامج المساعدة
 Name[be]=Дапаўненьні
 Name[bg]=Приставки
 Name[bs]=Dodaci
@@ -54,6 +55,7 @@
 Name[xx]=xxPluginsxx
 Name[zh_CN]=插件
 Comment=Choose and configure your plugins:
+Comment[ar]=اختيار وإعداد برامجك المساعدة
 Comment[bg]=Избор и конфигуриране на приставки:
 Comment[bs]=Izaberite i podesite vaše dodatke:
 Comment[ca]=Escolliu i configureu els connectors:
@@ -67,7 +69,7 @@
 Comment[fi]=Valitse ja muokkaa liitännäisten asetuksia:
 Comment[fr]=Choisir et configurer vos modules externes
 Comment[gl]=Elixe e configura os plugins:
-Comment[hi]=अपने प्लगइन्स को चुनें तथा कॉन्फिगर करें:
+Comment[hi]=अपने प्लगइन्स को चुनें तथा कॉन्फ़िगर करें:
 Comment[hu]=A bővítőmodulok beállítása:
 Comment[is]=Velja og stilla íforrit:
 Comment[it]=Scegli e configura i tuoi plugin:
diff -urN -x CVS kdegraphics.orig/kview/kimageviewer/kimageviewer.desktop kdegraphics/kview/kimageviewer/kimageviewer.desktop
--- kdegraphics.orig/kview/kimageviewer/kimageviewer.desktop	Sun Feb 29 16:28:19 2004
+++ kdegraphics/kview/kimageviewer/kimageviewer.desktop	Mon May  3 15:04:43 2004
@@ -4,9 +4,10 @@
 X-KDE-Derived=KParts/ReadWritePart
 Comment=Embeddable Image Viewer Component
 Comment[af]=Inlegbare Beeld Aansig Komponent
+Comment[ar]=مكون عارض الصور القابل للدمج
 Comment[bg]=Модул за преглед на изображения
 Comment[bs]=Ugradiva komponenta za pregled slika
-Comment[ca]=Component visualitzador de imatges encastable
+Comment[ca]=Component visualitzador d'imatges encastable
 Comment[cs]=Komponenta pro zobrazování obrázků
 Comment[cy]=Cydran Mewnadeiladadwy Gwelydd Delweddau
 Comment[da]=Indlejrbar billedviserkomponent
@@ -19,7 +20,7 @@
 Comment[fr]=Composant afficheur d'images incorporable
 Comment[gl]=Componente de visualización integrable
 Comment[he]=רכיב מציג תמונות בר-הטבעה
-Comment[hi]=एम्बेडेबल चित्र प्रदर्शक अवयव
+Comment[hi]=एम्बेडेबल छवि प्रदर्शक अवयव
 Comment[hr]=Umetljiva komponenta za gledanje slika
 Comment[hu]=Beágyazható képnézegető komponens
 Comment[is]=Ívefjanleg myndsjá
diff -urN -x CVS kdegraphics.orig/kview/kimageviewer/kimageviewercanvas.desktop kdegraphics/kview/kimageviewer/kimageviewercanvas.desktop
--- kdegraphics.orig/kview/kimageviewer/kimageviewercanvas.desktop	Sun Feb 29 16:28:21 2004
+++ kdegraphics/kview/kimageviewer/kimageviewercanvas.desktop	Mon May  3 15:04:45 2004
@@ -3,9 +3,10 @@
 X-KDE-ServiceType=KImageViewer/Canvas
 Comment=Embeddable Image Viewer Canvas (widget that shows an image)
 Comment[af]=Inlegbare Beeld Aansig Kanvas (gui-element wat vertoon 'n Beeld)
+Comment[ar]= مساحة رسم لعرض الصور قابلة للدمج (كائن يعرض صورة)
 Comment[bg]=Модул за преглед на изображения (полето, в което се показва самото изображение)
 Comment[bs]=Ugradiva pozadina za pregled slika (grafički element koji prikazuje slike)
-Comment[ca]=Llenç visualitzador de imatges encastable (estri que mostra una imatge)
+Comment[ca]=Llenç visualitzador d'imatges encastable (estri que mostra una imatge)
 Comment[cs]=Pohltitelná komponenta pro zobrazování obrázků
 Comment[cy]=Cynfas Mewnadeiladadwy Gwelydd Delweddau (celfigyn sy'n dangos delwedd)
 Comment[da]=Indlejrbar billedviserlærred (kontrol der viser et billede)
@@ -16,7 +17,7 @@
 Comment[fi]=Upotettava kuviennäyttökomponentti (komponentti joka näyttää kuvan)
 Comment[fr]=Composant afficheur d'images incorporable (widget qui affiche une image)
 Comment[he]=רכיב מציג תמונות בר-הטבעה (פריט המציג תמונה)
-Comment[hi]=एम्बेडेबल चित्र प्रदर्शक केनवास (विजेट जो चित्र दिखाता है)
+Comment[hi]=एम्बेडेबल छवि प्रदर्शक केनवास (विजेट जो छवि दिखाता है)
 Comment[hr]=Umetljivo platno za pregled slika (widget koji pokazuje sliku)
 Comment[hu]=Beágyazható képnézegető objektum (képmegjelenítő)
 Comment[is]=Ívefjanleg myndsjá (græja sem birtir mynd)
diff -urN -x CVS kdegraphics.orig/kview/kview.desktop kdegraphics/kview/kview.desktop
--- kdegraphics.orig/kview/kview.desktop	Mon Mar 29 15:01:59 2004
+++ kdegraphics/kview/kview.desktop	Mon May  3 15:04:47 2004
@@ -13,7 +13,7 @@
 GenericName[ar]=عارض صور
 GenericName[bg]=Преглед на изображения
 GenericName[bs]=Preglednik slika
-GenericName[ca]=Visualitzador de imatges
+GenericName[ca]=Visualitzador d'imatges
 GenericName[cs]=Prohlížeč obrázků
 GenericName[cy]=Gwelydd Delweddau
 GenericName[da]=Billedfremviser
@@ -28,7 +28,7 @@
 GenericName[ga]=Amharcán Íomhánna
 GenericName[gl]=Visor de imaxes
 GenericName[he]=מציג תמונות
-GenericName[hi]=चित्र प्रदर्शक
+GenericName[hi]=छवि प्रदर्शक
 GenericName[hr]=Preglednik slika
 GenericName[hu]=Képnézegető
 GenericName[is]=Myndaskoðari
@@ -69,7 +69,7 @@
 MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/x-bmp;image/png;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap;image/tiff;image/jp2
 Name=KView
 Name[af]=K-bekyk
-Name[ar]=كفيو
+Name[ar]=برنامج KView
 Name[cy]=KGweld
 Name[eo]=Rigardilo
 Name[hi]=के-व्यू
diff -urN -x CVS kdegraphics.orig/kview/kviewcanvas/config/kviewcanvasconfig.desktop kdegraphics/kview/kviewcanvas/config/kviewcanvasconfig.desktop
--- kdegraphics.orig/kview/kviewcanvas/config/kviewcanvasconfig.desktop	Sun Feb 29 16:28:25 2004
+++ kdegraphics/kview/kviewcanvas/config/kviewcanvasconfig.desktop	Mon May  3 15:04:49 2004
@@ -13,6 +13,7 @@
 X-KDE-CfgDlgHierarchy=Viewer
 
 Name=Viewer
+Name[ar]=العارض
 Name[bs]=Preglednik
 Name[ca]=Visualitzador
 Name[cs]=Prohlížeč
@@ -50,6 +51,7 @@
 Name[xx]=xxViewerxx
 Name[zh_CN]=查看器
 Comment=General KViewCanvas Configuration
+Comment[ar]=اعدادات KViewCanvas العامة
 Comment[bg]=Общи настройки на KViewCanvas
 Comment[bs]=Opšte KViewCanvas postavke
 Comment[ca]=Configuració general de KViewCanvas
@@ -63,7 +65,7 @@
 Comment[fi]=Yleiset KViewCanvas asetukset
 Comment[fr]=Configuration générale de KViewCanvas
 Comment[gl]=Configuración xeral dos lenzos de KView
-Comment[hi]=सामान्य के-व्यू-केनवास कॉन्फिगरेशन
+Comment[hi]=सामान्य के-व्यू-केनवास कॉन्फ़िगरेशन
 Comment[hu]=A KViewCanvas általános beállításai
 Comment[is]=Almennar stillingar KViewCanvas
 Comment[it]=Configurazione generale per KViewCanvas
diff -urN -x CVS kdegraphics.orig/kview/kviewcanvas/kviewcanvas.desktop kdegraphics/kview/kviewcanvas/kviewcanvas.desktop
--- kdegraphics.orig/kview/kviewcanvas/kviewcanvas.desktop	Sun Feb 29 16:28:27 2004
+++ kdegraphics/kview/kviewcanvas/kviewcanvas.desktop	Mon May  3 15:04:51 2004
@@ -2,8 +2,9 @@
 Type=Service
 Name=KView Image Viewer Widget
 Name[af]=K-bekyk Beeld Aansig Gui-element
+Name[ar]=كائن عارض الصور KView 
 Name[bs]=KView predglednik slika - grafički element
-Name[ca]=Estri visualitzador de imatges KView
+Name[ca]=Estri visualitzador d'imatges KView
 Name[cs]=Komponenta prohlížeče obrázků KView
 Name[cy]=Celfigyn Gwelydd Delweddau KGweld
 Name[da]=KView-billedviserkontrol
@@ -16,7 +17,7 @@
 Name[fr]=Widget visualisateur d'images KView
 Name[gl]=Widget do Visor de Imaxes KView
 Name[he]=פריט מציג תמונות ל-KView
-Name[hi]=के-व्यू चित्र प्रदर्शक विजेट
+Name[hi]=के-व्यू छवि प्रदर्शक विजेट
 Name[hr]=KView widget za pregled slika
 Name[hu]=KView képnézegető grafikus elem
 Name[is]=KView myndbirtigræja
diff -urN -x CVS kdegraphics.orig/kview/kviewviewer/config/kviewviewerpluginsconfig.desktop kdegraphics/kview/kviewviewer/config/kviewviewerpluginsconfig.desktop
--- kdegraphics.orig/kview/kviewviewer/config/kviewviewerpluginsconfig.desktop	Fri Mar  5 13:51:26 2004
+++ kdegraphics/kview/kviewviewer/config/kviewviewerpluginsconfig.desktop	Mon May  3 15:04:53 2004
@@ -13,6 +13,7 @@
 X-KDE-CfgDlgHierarchy=Viewer
 
 Name=Plugins
+Name[ar]=البرامج المساعدة
 Name[be]=Дапаўненьні
 Name[bg]=Приставки
 Name[bs]=Dodaci
@@ -54,6 +55,7 @@
 Name[xx]=xxPluginsxx
 Name[zh_CN]=插件
 Comment=Choose and configure your plugins:
+Comment[ar]=اختيار وإعداد برامجك المساعدة
 Comment[bg]=Избор и конфигуриране на приставки:
 Comment[bs]=Izaberite i podesite vaše dodatke:
 Comment[ca]=Escolliu i configureu els connectors:
@@ -67,7 +69,7 @@
 Comment[fi]=Valitse ja muokkaa liitännäisten asetuksia:
 Comment[fr]=Choisir et configurer vos modules externes
 Comment[gl]=Elixe e configura os plugins:
-Comment[hi]=अपने प्लगइन्स को चुनें तथा कॉन्फिगर करें:
+Comment[hi]=अपने प्लगइन्स को चुनें तथा कॉन्फ़िगर करें:
 Comment[hu]=A bővítőmodulok beállítása:
 Comment[is]=Velja og stilla íforrit:
 Comment[it]=Scegli e configura i tuoi plugin:
diff -urN -x CVS kdegraphics.orig/kview/kviewviewer/kviewviewer.desktop kdegraphics/kview/kviewviewer/kviewviewer.desktop
--- kdegraphics.orig/kview/kviewviewer/kviewviewer.desktop	Mon Mar 29 15:02:01 2004
+++ kdegraphics/kview/kviewviewer/kviewviewer.desktop	Mon May  3 15:04:55 2004
@@ -3,8 +3,9 @@
 Icon=kview
 Comment=KDE Image Viewer Part
 Comment[af]=Kde Beeld Aansig Deel
+Comment[ar]=عارض جزء الصور ل KDE
 Comment[bs]=KDE Part - Preglednik slika
-Comment[ca]=Peça de KDE visualitzadora de imatges
+Comment[ca]=Peça de KDE visualitzadora d'imatges
 Comment[cs]=Komponenta prohlížeče obrázků KDE
 Comment[cy]=Rhan Gwelydd Delweddau KDE 
 Comment[da]=KDE Billedviserdel
@@ -17,7 +18,7 @@
 Comment[fr]=Composant d'affichage d'images de KDE
 Comment[gl]=Visor de imaxes integrado de KDE
 Comment[he]=רכיב מציג תמונות ל-KDE
-Comment[hi]=केडीई चित्र प्रदर्शक पार्ट
+Comment[hi]=केडीई छवि प्रदर्शक पार्ट
 Comment[hr]=KDE komponenta za pregledavanje slika
 Comment[hu]=KDE képnézegető objektum
 Comment[is]=Myndbirtir KDE
@@ -51,8 +52,9 @@
 Comment[zu]=Ingxenye Yombukisi Wesithombe se-KDE
 MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/x-bmp;image/png;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap;image/tiff;image/jp2
 Name=KView Image Viewer
+Name[ar]=عارض KView للصور
 Name[bs]=KView predglednik slika
-Name[ca]=Estri visualitzador de imatges KView
+Name[ca]=Estri visualitzador d'imatges KView
 Name[cs]=Prohlížeč obrázků KView
 Name[cy]=Celfigyn Gwelydd Delweddau KGweld
 Name[da]=KView-billedfremviser
@@ -63,7 +65,7 @@
 Name[fi]=KView-kuvakatsoja
 Name[fr]=Visualisateur d'images KView
 Name[gl]=Visor de imaxes KView
-Name[hi]=के-व्यू चित्र प्रदर्शक
+Name[hi]=के-व्यू छवि प्रदर्शक
 Name[hu]=KView képnézegető
 Name[is]=KView myndbirtirinn
 Name[it]=Visualizzatore immagini KView
diff -urN -x CVS kdegraphics.orig/kview/modules/browser/kviewbrowser.desktop kdegraphics/kview/modules/browser/kviewbrowser.desktop
--- kdegraphics.orig/kview/modules/browser/kviewbrowser.desktop	Fri Mar  5 13:51:28 2004
+++ kdegraphics/kview/modules/browser/kviewbrowser.desktop	Mon May  3 15:04:57 2004
@@ -14,6 +14,7 @@
 
 Name=Browser
 Name[af]=Blaaier
+Name[ar]=المتصفح
 Name[bg]=Браузър
 Name[bs]=Preglednik
 Name[ca]=Fullejador
@@ -30,6 +31,7 @@
 Name[gl]=Explorador
 Name[he]=דפדפן
 Name[hi]=ब्राउज़र
+Name[hr]=Preglednik
 Name[hu]=Böngésző
 Name[is]=Flakkari
 Name[ja]=ブラウザー
@@ -62,6 +64,7 @@
 Name[zu]=Umcingi
 Comment=Enables you to browse through all of the images in the current directory.
 Comment[af]=Aktiveer jy na blaai deur alle van die beelde in die huidige gids.
+Comment[ar]=يمكنك من تصفَح كل الصور في الدليل الحالي.
 Comment[bg]=Преглед на изображенията в текущата директория
 Comment[bs]=Omogućuje vam da pregledate sve slike u trenutnom direktoriju.
 Comment[ca]=Us permet navegar entre totes les imatges del directori actual.
@@ -76,7 +79,7 @@
 Comment[fr]=Vous permet de naviguer parmi les images dans le dossier courant.
 Comment[gl]=Activa o erowse through all of the images in the current directory.
 Comment[he]=מאפשר לך לעיין בכל התמונות שבספריה הנוכחית
-Comment[hi]=वर्तमान डिरेक्ट्री के सभी चित्रों को ब्राउज़ करने में आपको सक्षम बनाता है.
+Comment[hi]=वर्तमान डिरेक्ट्री के सभी छविों को ब्राउज़ करने में आपको सक्षम बनाता है.
 Comment[hu]=Lehetővé teszi az aktuális könyvtárban található képek áttekintését, böngészését.
 Comment[is]=Gerir þér kleyft að flakka í öllum myndunum í þessari möppu.
 Comment[it]=Permette di navigare tra le immagini nella directory corrente.
diff -urN -x CVS kdegraphics.orig/kview/modules/effects/kvieweffects.desktop kdegraphics/kview/modules/effects/kvieweffects.desktop
--- kdegraphics.orig/kview/modules/effects/kvieweffects.desktop	Sun Feb 29 16:28:35 2004
+++ kdegraphics/kview/modules/effects/kvieweffects.desktop	Mon May  3 15:04:59 2004
@@ -12,6 +12,7 @@
 X-KDE-PluginInfo-EnabledByDefault=false
 
 Name=Effects
+Name[ar]=مؤثرات
 Name[bg]=Ефекти
 Name[bs]=Efekti
 Name[ca]=Efectes
@@ -51,9 +52,10 @@
 Name[xx]=xxEffectsxx
 Name[zh_CN]=特效
 Comment=Provides some image effects
+Comment[ar]=يقدم بعض مؤثرات الصور
 Comment[bg]=Визуални ефекти при зареждане на изображенията
 Comment[bs]=Pruža neke efekte za slike
-Comment[ca]=Proporciona alguns efectes de imatges
+Comment[ca]=Proporciona alguns efectes d'imatges
 Comment[cs]=Poskytuje několik efektů pro obrázky
 Comment[cy]=Darparu rhai effeithiau delwedd
 Comment[da]=Sørger for nogle billedeffekter
@@ -64,7 +66,7 @@
 Comment[fi]=Tarjoaa joitain kuvatehosteita
 Comment[fr]=Fournit des images sur les images
 Comment[gl]=Proporciona algúns efectos para as imaxes
-Comment[hi]=चित्र में प्रभाव उत्पन्न करता है
+Comment[hi]=छवि में प्रभाव उत्पन्न करता है
 Comment[hu]=Képeffektusok használata
 Comment[is]=Býður uppá ýmsar myndbrellur
 Comment[it]=Provvede alcuni effetti per le immagini
diff -urN -x CVS kdegraphics.orig/kview/modules/presenter/config/kviewpresenterconfig.desktop kdegraphics/kview/modules/presenter/config/kviewpresenterconfig.desktop
--- kdegraphics.orig/kview/modules/presenter/config/kviewpresenterconfig.desktop	Sun Feb 29 16:28:37 2004
+++ kdegraphics/kview/modules/presenter/config/kviewpresenterconfig.desktop	Mon May  3 15:05:01 2004
@@ -10,6 +10,7 @@
 X-KDE-ParentComponents=kviewpresenter
 
 Name=Name
+Name[ar]=اسم
 Name[bg]=Име
 Name[bs]=Ime
 Name[ca]=Nom
@@ -46,6 +47,7 @@
 Name[xx]=xxNamexx
 Name[zh_CN]=名称
 Comment=Comment
+Comment[ar]=تعليق
 Comment[bg]=Коментар
 Comment[bs]=Komentar
 Comment[ca]=Comentari
diff -urN -x CVS kdegraphics.orig/kview/modules/presenter/kviewpresenter.desktop kdegraphics/kview/modules/presenter/kviewpresenter.desktop
--- kdegraphics.orig/kview/modules/presenter/kviewpresenter.desktop	Sun Feb 29 16:28:39 2004
+++ kdegraphics/kview/modules/presenter/kviewpresenter.desktop	Mon May  3 15:05:03 2004
@@ -12,6 +12,7 @@
 X-KDE-PluginInfo-EnabledByDefault=true
 
 Name=Presenter
+Name[ar]=المقدّم
 Name[bg]=Презентация
 Name[bs]=Prezentator
 Name[ca]=Presentador
@@ -26,6 +27,7 @@
 Name[gl]=Escaparate
 Name[he]=מצגות
 Name[hi]=प्रस्तुतकर्ता
+Name[hr]=Prezentator
 Name[hu]=Bemutató
 Name[ja]=プレゼンター
 Name[lt]=Pristatytojas
@@ -51,9 +53,10 @@
 Name[zh_TW]=簡報
 Name[zu]=Umnikeli
 Comment=Creates an imagelist and enables you to create a slideshow
+Comment[ar]=ينشئ قائمة صور ويمكنك من استعراض الصور تلقائيا
 Comment[bg]=Създаване на списък с изображения и слайдшоу от тях
 Comment[bs]=Pravi listu slika i omogućuje vam da napravite slide show
-Comment[ca]=Crea una llista de imatges i us permet crear un passi de diapositives
+Comment[ca]=Crea una llista d'imatges i us permet crear un passi de diapositives
 Comment[cs]=Vytvoří seznam obrázků a umožní vám z nich vytvořit slideshow.
 Comment[cy]=Creu rhestr delweddau ac alluogi i chi greu sioe haenluniau
 Comment[da]=Laver en billedliste og giver dig muligheden for at lave et diasshow
@@ -65,7 +68,7 @@
 Comment[fr]=Crée une liste d'images et vous permet de créer un diaporama
 Comment[gl]=Crea unha lista de imaxes e te permite crear unha moviola
 Comment[he]=יוצר רשימת תמונות ומאפשר לך ליצור מצגת שקופיות
-Comment[hi]=एक चित्र-सूची बनाता है तथा आपको एक स्लाइड-शो तैयार करने में सक्षम बनाता है
+Comment[hi]=एक छवि-सूची बनाता है तथा आपको एक स्लाइड-शो तैयार करने में सक्षम बनाता है
 Comment[hu]=Képsorozat összeállítását és bemutatását teszi lehetővé
 Comment[is]=Býr til myndlista og gerir þér kleyft að búa til myndsýningu
 Comment[it]=Crea una lista di immagini e permette di creare una presentazione
diff -urN -x CVS kdegraphics.orig/kview/modules/scale/kview_scale.desktop kdegraphics/kview/modules/scale/kview_scale.desktop
--- kdegraphics.orig/kview/modules/scale/kview_scale.desktop	Sun Feb 29 16:28:41 2004
+++ kdegraphics/kview/modules/scale/kview_scale.desktop	Mon May  3 15:05:05 2004
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Scale
 Name[af]=Skaal
+Name[ar]=تكبير أو تصغير
 Name[bg]=Мащабиране
 Name[bs]=Skaliraj
 Name[ca]=Escala
@@ -50,6 +51,7 @@
 Name[zu]=Isikali
 Comment=Filter to scale the image
 Comment[af]=Filter na skaal die beeld
+Comment[ar]=فلتر لتكبير أو تصغير الصورة 
 Comment[bg]=Филтър за мащабиране на изображения
 Comment[bs]=Filter za skaliranje slike
 Comment[ca]=Filtre per escalar la imatge
@@ -65,7 +67,8 @@
 Comment[fr]=Filtre pour zoomer une image
 Comment[gl]=Filtro para redimensionar a imaxe
 Comment[he]=מסנן לשינוי קנה המידה של התמונה
-Comment[hi]=चित्र का आकार बदलने का फिल्टर
+Comment[hi]=छवि का आकार बदलने का फिल्टर
+Comment[hr]=Filter za promjenu veličine slike
 Comment[hu]=Képméretező szűrő
 Comment[is]=Sía sem skalar myndina
 Comment[it]=Filtro per ridimensionare l'immagine
diff -urN -x CVS kdegraphics.orig/kview/modules/scanner/kviewscanner.desktop kdegraphics/kview/modules/scanner/kviewscanner.desktop
--- kdegraphics.orig/kview/modules/scanner/kviewscanner.desktop	Mon Mar 29 15:02:03 2004
+++ kdegraphics/kview/modules/scanner/kviewscanner.desktop	Mon May  3 15:05:07 2004
@@ -13,6 +13,7 @@
 
 Name=Scanner
 Name[af]=Skandeerder
+Name[ar]=الماسح الضوئي
 Name[bg]=Скенер
 Name[bs]=Skener
 Name[ca]=Escànner
@@ -26,6 +27,7 @@
 Name[gl]=Escáner
 Name[he]=סורק
 Name[hi]=स्केनर
+Name[hr]=Skener
 Name[hu]=Lapolvasó
 Name[is]=Skanni
 Name[ja]=スキャナー
@@ -57,6 +59,7 @@
 Name[zu]=Umhloli
 Comment=Open images from your scanner into KView
 Comment[af]=Open beelde van jou skandeerder binnein K-bekyk
+Comment[ar]=يفتح الصور الموجودة على ماسحك الضوئي في برنامج KView
 Comment[bg]=Сканиране и отваряне на изображения от скенера
 Comment[bs]=Otvara slike sa skenera u KView
 Comment[ca]=Obre les imatges de l'escànner a KView
@@ -71,7 +74,7 @@
 Comment[fr]=Ouvre des images de votre scanner dans KView
 Comment[gl]=Obre as imaxes de  KView dende o teu escáner
 Comment[he]=פתיחת תמונות מהסורק שלך ב-KView
-Comment[hi]=आपके स्केनर से के-व्यू में चित्र खोलता है
+Comment[hi]=आपके स्केनर से के-व्यू में छवि खोलता है
 Comment[hu]=Kép beolvasása lapolvasóról a KView-ba
 Comment[is]=Senda myndir frá skannanum í KView
 Comment[it]=Apre le immagini provenienti dallo scanner in KView
diff -urN -x CVS kdegraphics.orig/kview/modules/template/kviewtemplate.desktop kdegraphics/kview/modules/template/kviewtemplate.desktop
--- kdegraphics.orig/kview/modules/template/kviewtemplate.desktop	Fri Apr  2 14:09:51 2004
+++ kdegraphics/kview/modules/template/kviewtemplate.desktop	Sat Apr 17 14:36:46 2004
@@ -77,6 +77,7 @@
 Comment[gl]=Unha máis longa descrición do que pode facer a extensión (plugin)
 Comment[he]=תיאור מפורט יותר לגבי מה שהתוסף עושה
 Comment[hi]=प्लगइन के बारे में लंबा वर्णन कि वह क्या करता है
+Comment[hr]=Duži opis umetka
 Comment[hu]=Ez a bővítőmodul részletesebb leírása
 Comment[is]=hér ætti að vera smá lýsing á hvað þetta gerir
 Comment[it]=Una descrizione più lunga di quello che fa il plugin
diff -urN -x CVS kdegraphics.orig/libkscan/massscandialog.cpp kdegraphics/libkscan/massscandialog.cpp
--- kdegraphics.orig/libkscan/massscandialog.cpp	Mon Feb 24 13:51:07 2003
+++ kdegraphics/libkscan/massscandialog.cpp	Wed Apr 14 19:47:55 2004
@@ -1,5 +1,5 @@
 /* This file is part of the KDE Project
-   Copyright (C) 2000 Klaas Freitag <freitag@suse.de>  
+   Copyright (C) 2000 Klaas Freitag <freitag@suse.de>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -16,7 +16,7 @@
    the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
    Boston, MA 02111-1307, USA.
 
-   $Id$
+   $Id$
 */
 
 #include <qlayout.h>
@@ -25,10 +25,10 @@
 #include <qgroupbox.h>
 #include <qframe.h>
 
-#include <qpushbutton.h>
-
 #include <klocale.h>
 #include <kdebug.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include "massscandialog.h"
 
@@ -41,11 +41,11 @@
    QVBoxLayout *bigdad = new QVBoxLayout( this, 5 );
    // QHBoxLayout *hl1= new QHBoxLayout( );      // Caption
    QHBoxLayout *l_but  = new QHBoxLayout( 10 );  // Buttons
- 	
+
  	/* Caption */
  	QLabel *l1 = new QLabel( i18n( "<B>Mass Scanning</B>" ), this);
    bigdad->addWidget( l1, 1);
- 	
+
  	/* Scan parameter information */
  	QGroupBox *f1 = new QGroupBox( i18n("Scan Parameter"), this );
  	f1->setFrameStyle( QFrame::Box | QFrame::Sunken );
@@ -53,7 +53,7 @@
  	f1->setLineWidth( 1 );
    QVBoxLayout *l_main = new QVBoxLayout( f1, f1->frameWidth()+3, 3 );
  	bigdad->addWidget( f1, 6 );
- 	
+
    scanopts = i18n("Scanning <B>%s</B> with <B>%d</B> dpi");
  	l_scanopts = new QLabel( scanopts, f1 );
  	l_main->addWidget( l_scanopts );
@@ -61,7 +61,7 @@
    tofolder = i18n("Storing new images in folder <B>%s</B>");
  	l_tofolder = new QLabel( tofolder, f1 );
  	l_main->addWidget( l_tofolder );
- 	
+
  	/* Scan Progress information */
  	QGroupBox *f2 = new QGroupBox( i18n("Scan Progress"), this );
  	f2->setFrameStyle( QFrame::Box | QFrame::Sunken );
@@ -91,7 +91,7 @@
    QPushButton *b_cancel = new QPushButton( i18n("Stop"), this, "ButtCancel" );
    connect( b_cancel, SIGNAL(clicked()), this, SLOT(slStopScan()) );
 
-   QPushButton *b_finish = new QPushButton( i18n("Close"), this, "ButtFinish" );
+   QPushButton *b_finish = new KPushButton( KStdGuiItem::close(), this, "ButtFinish" );
    connect( b_finish, SIGNAL(clicked()), this, SLOT(slFinished()) );
 
    l_but->addWidget( b_start );
