Index: kpdf/part.cpp
===================================================================
--- kpdf/part.cpp	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kpdf/part.cpp	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -697,6 +697,11 @@
     KURL saveURL = KFileDialog::getSaveURL( url().isLocalFile() ? url().url() : url().fileName(), QString::null, widget() );
     if ( saveURL.isValid() && !saveURL.isEmpty() )
     {
+        if (saveURL == url())
+        {
+            KMessageBox::information( widget(), i18n("You are trying to overwrite \"%1\" with itself. This is not allowed. Please save it in another location.").arg(saveURL.filename()) );
+            return;
+        }
         if ( KIO::NetAccess::exists( saveURL, false, widget() ) )
         {
             if (KMessageBox::warningContinueCancel( widget(), i18n("A file named \"%1\" already exists. Are you sure you want to overwrite it?").arg(saveURL.filename()), QString::null, i18n("Overwrite")) != KMessageBox::Continue)
Index: kpdf/configure.in.in
===================================================================
--- kpdf/configure.in.in	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kpdf/configure.in.in	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -66,7 +66,7 @@
 fi
 
 dnl #### Check for FSEEK variants
-AC_SYS_LARGEFILE
+KDE_CHECK_LARGEFILE
 AC_FUNC_FSEEKO
 AC_CHECK_FUNCS(fseek64, xpdf_cv_func_fseek64=yes, xpdf_cv_func_fseek64=no)
 AC_CHECK_FUNCS(ftell64, xpdf_cv_func_ftell64=yes, xpdf_cv_func_ftell64=no)
Index: kpdf/xpdf/xpdf/GfxFont.cc
===================================================================
--- kpdf/xpdf/xpdf/GfxFont.cc	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kpdf/xpdf/xpdf/GfxFont.cc	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -1493,6 +1493,7 @@
   unsigned int cidlen;
 
   *mapsizep = 0;
+  if (!ctu) return NULL;
 
   /* we use only unicode cmap */
   cmap = -1;
Index: kpdf/xpdf/splash/SplashXPathScanner.cc
===================================================================
--- kpdf/xpdf/splash/SplashXPathScanner.cc	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kpdf/xpdf/splash/SplashXPathScanner.cc	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -186,7 +186,7 @@
 }
 
 void SplashXPathScanner::computeIntersections(int y) {
-  SplashCoord ySegMin, ySegMax, xx0, xx1;
+  SplashCoord xSegMin, xSegMax, ySegMin, ySegMax, xx0, xx1;
   SplashXPathSeg *seg;
   int i, j;
 
@@ -236,20 +236,28 @@
     } else if (seg->flags & splashXPathVert) {
       xx0 = xx1 = seg->x0;
     } else {
-      if (ySegMin <= y) {
-	// intersection with top edge
-	xx0 = seg->x0 + ((SplashCoord)y - seg->y0) * seg->dxdy;
+      if (seg->x0 < seg->x1) {
+	xSegMin = seg->x0;
+	xSegMax = seg->x1;
       } else {
-	// x coord of segment endpoint with min y coord
-	xx0 = (seg->flags & splashXPathFlip) ? seg->x1 : seg->x0;
+	xSegMin = seg->x1;
+	xSegMax = seg->x0;
       }
-      if (ySegMax >= y + 1) {
-	// intersection with bottom edge
-	xx1 = seg->x0 + ((SplashCoord)y + 1 - seg->y0) * seg->dxdy;
-      } else {
-	// x coord of segment endpoint with max y coord
-	xx1 = (seg->flags & splashXPathFlip) ? seg->x0 : seg->x1;
+      // intersection with top edge
+      xx0 = seg->x0 + ((SplashCoord)y - seg->y0) * seg->dxdy;
+      // intersection with bottom edge
+      xx1 = seg->x0 + ((SplashCoord)y + 1 - seg->y0) * seg->dxdy;
+      // the segment may not actually extend to the top and/or bottom edges
+      if (xx0 < xSegMin) {
+	xx0 = xSegMin;
+      } else if (xx0 > xSegMax) {
+	xx0 = xSegMax;
       }
+      if (xx1 < xSegMin) {
+	xx1 = xSegMin;
+      } else if (xx1 > xSegMax) {
+	xx1 = xSegMax;
+      }
     }
     if (xx0 < xx1) {
       inter[interLen].x0 = splashFloor(xx0);
Index: kfaxview/kfaxmultipage_tiff.desktop
===================================================================
--- kfaxview/kfaxmultipage_tiff.desktop	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kfaxview/kfaxmultipage_tiff.desktop	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -18,7 +18,7 @@
 Comment[et]=KView faksifailide plugin
 Comment[eu]=Fax fitxategientzatko KViewShell-en plugina
 Comment[fi]=KViewShell sovelma faksitiedostoille
-Comment[fr]=Module KViewShell pour les fichiers fax
+Comment[fr]=Module KViewShell pour les fichiers de fax
 Comment[hu]=KViewShell-modul faxfájlokhoz
 Comment[is]=KViewShell íforrit fyrir faxskrár
 Comment[it]=Plugin KViewShell per file di fax
Index: kfaxview/kfaxmultipage.desktop
===================================================================
--- kfaxview/kfaxmultipage.desktop	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kfaxview/kfaxmultipage.desktop	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -20,7 +20,7 @@
 Comment[et]=KView faksifailide plugin
 Comment[eu]=Fax fitxategientzatko KViewShell-en plugina
 Comment[fi]=KViewShell sovelma faksitiedostoille
-Comment[fr]=Module KViewShell pour les fichiers fax
+Comment[fr]=Module KViewShell pour les fichiers de fax
 Comment[hu]=KViewShell-modul faxfájlokhoz
 Comment[is]=KViewShell íforrit fyrir faxskrár
 Comment[it]=Plugin KViewShell per file di fax
Index: kdvi/special.cpp
===================================================================
--- kdvi/special.cpp	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kdvi/special.cpp	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -697,6 +697,7 @@
       (strncasecmp(cp, "html:<A name=", 13) == 0) ||
       (strncasecmp(cp, "ps:", 3) == 0) ||
       (strncasecmp(cp, "papersize", 9) == 0) ||
+      (strncasecmp(cp, "header", 6) == 0) ||
       (strncasecmp(cp, "background", 10) == 0) )
     return;
   
Index: kdvi/dviRenderer_prescan.cpp
===================================================================
--- kdvi/dviRenderer_prescan.cpp	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kdvi/dviRenderer_prescan.cpp	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -284,8 +284,20 @@
   kdDebug(4300) << "PostScript-special, header " << cp.latin1() << endl;
 #endif
 
-  if (QFile::exists(cp)) 
-    PS_interface->PostScriptHeaderString->append( QString(" (%1) run\n").arg(cp) );
+  QString _file = cp;
+  
+  // If the file is not found in the current directory, use kpsewhich
+  // to find it.
+  if (!QFile::exists(_file)) {
+    // Otherwise, use kpsewhich to find the eps file.
+    KProcIO proc;
+    proc << "kpsewhich" << cp;
+    proc.start(KProcess::Block);
+    proc.readln(_file);
+  }
+  
+  if (QFile::exists(_file))
+    PS_interface->PostScriptHeaderString->append( QString(" (%1) run\n").arg(_file) );
 }
 
 
@@ -305,7 +317,6 @@
 {
 #ifdef DEBUG_PRESCAN
   kdError(4300) << "PostScript-special, literal PostScript " << cp.latin1() << endl;
-
 #endif
   
   double PS_H = (currinf.data.dvi_h*300.0)/(65536*1200)-300;
Index: kuickshow/src/kuickshow.cpp
===================================================================
--- kuickshow/src/kuickshow.cpp	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kuickshow/src/kuickshow.cpp	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -646,7 +646,7 @@
     }
 
     tryShowNextImage();
-    fileWidget->trash(list, false, false);
+    fileWidget->trash(list, parent, false, false);
 }
 
 void KuickShow::tryShowNextImage()
Index: kamera/kcontrol/kamera.desktop
===================================================================
--- kamera/kcontrol/kamera.desktop	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kamera/kcontrol/kamera.desktop	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -78,7 +78,7 @@
 Keywords[et]=gphoto,kaamera,digitaalkaamera,veebikaamera,kamera
 Keywords[eu]=gphoto,kamera,digicam,webcam,kamera
 Keywords[fi]=gphoto,kamera,digicam,webcam
-Keywords[fr]=gphoto,camera,digicam,webcam,kamera,caméscope,caméra
+Keywords[fr]=gphoto,camera,digicam,webcam,kamera,caméscope,caméra,appareil photo
 Keywords[he]=gphoto,kamera,מצלמה,מצלמת רשת,מצלמה דיגיטלית, amera,digicam,webcam
 Keywords[hi]=जीफोटो,केमरा,डिजिकेम,वेबकेम,केमरा
 Keywords[hu]=gphoto,fényképezőgép,digitális fényképezőgép,webkamera,videókamera
@@ -116,7 +116,7 @@
 Name[br]=Kamera niverel
 Name[bs]=Digitalna kamera
 Name[ca]=Càmera digital
-Name[cs]=Digitální kamera
+Name[cs]=Digitální fotoaparát
 Name[cy]=Camera Digidol
 Name[da]=Digitalt kamera
 Name[de]=Digitalkamera
Index: kiconedit/kiconedit.desktop
===================================================================
--- kiconedit/kiconedit.desktop	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kiconedit/kiconedit.desktop	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -19,7 +19,7 @@
 GenericName[et]=Ikoonide redaktor
 GenericName[eu]=Ikono editorea
 GenericName[fi]=Kuvakemuokkain
-GenericName[fr]=Éditeur d'icône
+GenericName[fr]=Éditeur d'icônes
 GenericName[ga]=Eagarthóir Deilbhíní
 GenericName[gl]=Editor de iconas
 GenericName[he]=עורך סמלים
Index: kfile-plugins/raw/x-image-raw.desktop
===================================================================
--- kfile-plugins/raw/x-image-raw.desktop	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kfile-plugins/raw/x-image-raw.desktop	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -15,7 +15,7 @@
 Comment[et]=Toores kaamerafail (RAW)
 Comment[eu]=RAW kamera-fitxategiak
 Comment[fi]=RAW-kuvatiedosto
-Comment[fr]=Fichiers brut d'appareil photo numériques
+Comment[fr]=Fichiers RAW d'appareil photo
 Comment[ga]=Comhad Ceamara RAW
 Comment[hu]=RAW-fájl
 Comment[is]=RAW myndavélaskrá
Index: kfile-plugins/raw/kfile_raw.desktop
===================================================================
--- kfile-plugins/raw/kfile_raw.desktop	(.../tags/KDE/3.5.1/kdegraphics)	(revision 505113)
+++ kfile-plugins/raw/kfile_raw.desktop	(.../branches/KDE/3.5/kdegraphics)	(revision 505113)
@@ -12,8 +12,9 @@
 Name[et]=Toored kaamerafailid
 Name[eu]=RAW kamera-fitxategiak
 Name[fi]=RAW-kuvatiedostot
-Name[fr]=Fichiers bruts d'appareil photo numériques
+Name[fr]=Fichiers RAW d'appareil photo numérique
 Name[ga]=Comhaid Cheamara RAW
+Name[he]=קבצי מצלמה דיגיטלית גולמיים
 Name[hu]=RAW-fájlok
 Name[is]=RAW myndavélaskrár
 Name[it]=File grezzi fotocamera digitale

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


