--- kdegraphics-3.1.2/kpovmodeler/configure.in.in.orig	2003-05-13 18:21:09.000000000 +0200
+++ kdegraphics-3.1.2/kpovmodeler/configure.in.in	2003-05-13 18:22:05.000000000 +0200
@@ -7,9 +7,6 @@
 if test "$ac_cv_header_GL_glu_h" = "no"; then
   KPOVMODELER_OPENGL=no;
 fi
-if test "$ac_cv_header_GL_glut_h" = "no"; then
-  KPOVMODELER_OPENGL=no;
-fi
 if test "$ac_cv_header_GL_glx_h" = "no"; then
   KPOVMODELER_OPENGL=no;
 fi
--- kdegraphics-3.1.2/kpovmodeler/configure.in.bot.orig	2003-05-13 18:20:36.000000000 +0200
+++ kdegraphics-3.1.2/kpovmodeler/configure.in.bot	2003-05-13 18:20:59.000000000 +0200
@@ -7,9 +7,6 @@
   if test "$ac_cv_header_GL_glu_h" = "no"; then
     echo "    - GLU"
   fi
-  if test "$ac_cv_header_GL_glut_h" = "no"; then
-    echo "    - GLUT"
-  fi
   if test "$ac_cv_header_GL_glx_h" = "no"; then
     echo "    - GLX"
   fi
--- kdegraphics-3.1.2/kpovmodeler/pmrendermanager.cpp.orig	2003-05-13 18:24:37.000000000 +0200
+++ kdegraphics-3.1.2/kpovmodeler/pmrendermanager.cpp	2003-05-13 18:24:49.000000000 +0200
@@ -43,7 +43,7 @@
 
 #include <GL/gl.h>
 #include <GL/glx.h>
-#include <GL/glut.h> // Only needed for gluPerspective
+#include <GL/glu.h> // Only needed for gluPerspective
 
 #include "pmglview.h"
 
--- kdegraphics-3.1.2/kpovmodeler/Makefile.am.orig	2003-05-19 16:38:56.000000000 +0200
+++ kdegraphics-3.1.2/kpovmodeler/Makefile.am	2003-05-19 16:39:15.000000000 +0200
@@ -160,7 +160,7 @@
 	version.h pmdocumentformat.h
 
 libkpovmodelerpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) $(KDE_PLUGIN) $(LIBFREETYPE_RPATH) 
-libkpovmodelerpart_la_LIBADD = $(LIBFREETYPE_LIBS) $(LIB_KPARTS) -lglut -lGLU -lGL -lXmu -lXi
+libkpovmodelerpart_la_LIBADD = $(LIBFREETYPE_LIBS) $(LIB_KPARTS) -lGLU -lGL -lXmu -lXi
 libkpovmodelerpart_la_METASOURCES = AUTO
 
 kpovmodeler_LDFLAGS = $(all_libraries) $(KDE_RPATH)
@@ -169,7 +169,7 @@
 
 #kpovmodeler_SOURCES = $(pmallsources) main.cpp
 #kpovmodeler_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-#kpovmodeler_LDADD = $(LIB_KFILE) $(LIB_KPARTS) -lglut -lGLU -lGL -lXmu -lXi
+#kpovmodeler_LDADD = $(LIB_KFILE) $(LIB_KPARTS) -lGLU -lGL -lXmu -lXi
 #kpovmodeler_METASOURCES = USE_AUTOMOC
 
 rcdir = $(kde_datadir)/kpovmodeler
