--- kdegraphics-3.0.3/kamera/kcontrol/kameradevice.cpp--	2002-09-12 14:33:03.000000000 +0200
+++ kdegraphics-3.0.3/kamera/kcontrol/kameradevice.cpp	2002-09-12 14:33:40.000000000 +0200
@@ -338,6 +338,8 @@ KameraDeviceSelectDialog::KameraDeviceSe
 
 	populateCameraListView();
 	load();
+	enableButtonOK( false );
+
 }
 
 bool KameraDeviceSelectDialog::populateCameraListView()
@@ -391,6 +393,8 @@ void KameraDeviceSelectDialog::load()
 
 void KameraDeviceSelectDialog::slot_setModel(QListViewItem *item)
 {
+  enableButtonOK( item );
+
 	QString model = item->text(0);
 	
 	CameraAbilities abilities;
