--- kdegraphics-030904/kamera/kcontrol/kamera.desktop~	2003-09-04 15:36:04.000000000 +0200
+++ kdegraphics-030904/kamera/kcontrol/kamera.desktop	2003-09-04 16:52:35.000000000 +0200
@@ -143,3 +143,4 @@
 Icon=camera
 Exec=kcmshell kamera
 DocPath=kamera/index.html
+Categories=Qt;KDE;X-KDE-settings-peripherals
--- kdegraphics-030329/kdvi/kdvi.desktop~	Wed Jan 22 00:43:40 2003
+++ kdegraphics-030329/kdvi/kdvi.desktop	Tue Apr  1 23:03:57 2003
@@ -71,4 +71,4 @@
 X-KDE-BrowserView-Args=dvi
 DocPath=kdvi/index.html
 X-KDE-StartupNotify=true
-Categories=Qt;KDE;Graphics
+Categories=Qt;KDE;Graphics;X-Viewer
--- kdegraphics-030329/kfax/kfax.desktop~	Fri Jan 10 00:47:21 2003
+++ kdegraphics-030329/kfax/kfax.desktop	Tue Apr  1 23:03:23 2003
@@ -62,4 +62,4 @@
 Terminal=0
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Graphics
+Categories=Qt;KDE;Graphics;X-Viewer
--- kdegraphics-030329/kghostview/kghostview.desktop~	Fri Jan 10 00:47:21 2003
+++ kdegraphics-030329/kghostview/kghostview.desktop	Tue Apr  1 23:04:52 2003
@@ -65,4 +65,4 @@
 X-KDE-Library=libkghostviewpart
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Graphics
+Categories=Qt;KDE;Graphics;X-Viewer
--- kdegraphics-030924/kpdf/kpdf/kpdf.desktop~	2003-09-23 21:48:34.000000000 +0200
+++ kdegraphics-030924/kpdf/kpdf/kpdf.desktop	2003-09-25 10:34:55.000000000 +0200
@@ -38,5 +38,5 @@
 Type=Application
 DocPath=kpdf/kpdf.html
 InitialPreference=4
-Categories=Qt;KDE;Graphics;PDFViewer
+Categories=Qt;KDE;Graphics;PDFViewer;X-Viewer
 
--- kdegraphics-030329/kuickshow/src/kuickshow.desktop~	Sat Mar  1 00:55:35 2003
+++ kdegraphics-030329/kuickshow/src/kuickshow.desktop	Tue Apr  1 23:06:36 2003
@@ -67,4 +67,4 @@
 GenericName[zh_TW]=影像檢視程式
 GenericName[zu]=Umbonisi Womfanekiso
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Graphics
+Categories=Qt;KDE;Graphics;X-Viewer
--- kdegraphics-030329/kview/kview.desktop~	Sat Mar  8 01:10:00 2003
+++ kdegraphics-030329/kview/kview.desktop	Tue Apr  1 23:07:49 2003
@@ -70,4 +70,4 @@
 InitialPreference=3
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Graphics
+Categories=Qt;KDE;Graphics;X-Viewer
