Index: kpdf/part.cpp
===================================================================
--- kpdf/part.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/part.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -186,9 +186,13 @@
 
 	int index;
 	// [left toolbox: Table of Contents] | []
-	m_tocFrame = new TOC( m_toolBox, m_document );
+	// dummy wrapper with layout to enable horizontal scroll bars (bug: 147233)
+	QWidget *tocWrapper = new QWidget(m_toolBox);
+	QVBoxLayout *tocWrapperLayout = new QVBoxLayout(tocWrapper);
+	m_tocFrame = new TOC( tocWrapper, m_document );
+	tocWrapperLayout->add(m_tocFrame);
 	connect(m_tocFrame, SIGNAL(hasTOC(bool)), this, SLOT(enableTOC(bool)));
-	index = m_toolBox->addItem( m_tocFrame, QIconSet(SmallIcon("text_left")), i18n("Contents") );
+	index = m_toolBox->addItem( tocWrapper, QIconSet(SmallIcon("text_left")), i18n("Contents") );
 	m_toolBox->setItemToolTip(index, i18n("Contents"));
 	enableTOC( false );
 
@@ -619,6 +623,7 @@
     }
     if (m_wasPresentationOpen) slotShowPresentation();
     emit enablePrintAction(true);
+    emit setWindowCaption(url().filename());
   }
   else
   {
Index: kpdf/core/document.cpp
===================================================================
--- kpdf/core/document.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/core/document.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1117,15 +1117,21 @@
             break;
         }
 
-    // [MEM] 1.2 append memory allocation descriptor to the FIFO
-    int memoryBytes = 4 * req->width * req->height;
-    AllocatedPixmap * memoryPage = new AllocatedPixmap( req->id, req->pageNumber, memoryBytes );
-    d->allocatedPixmapsFifo.append( memoryPage );
-    d->allocatedPixmapsTotalMemory += memoryBytes;
+    if ( d->observers.contains( req->id ) )
+    {
+        // [MEM] 1.2 append memory allocation descriptor to the FIFO
+        int memoryBytes = 4 * req->width * req->height;
+        AllocatedPixmap * memoryPage = new AllocatedPixmap( req->id, req->pageNumber, memoryBytes );
+        d->allocatedPixmapsFifo.append( memoryPage );
+        d->allocatedPixmapsTotalMemory += memoryBytes;
 
-    // 2. notify an observer that its pixmap changed
-    if ( d->observers.contains( req->id ) )
+        // 2. notify an observer that its pixmap changed
         d->observers[ req->id ]->notifyPageChanged( req->pageNumber, DocumentObserver::Pixmap );
+    }
+#ifndef NDEBUG
+    else
+        kdWarning() << "Receiving a done request for the defunct observer " << req->id << endl;
+#endif
 
     // 3. delete request
     delete req;
Index: kpdf/core/generator_pdf/generator_pdf.cpp
===================================================================
--- kpdf/core/generator_pdf/generator_pdf.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/core/generator_pdf/generator_pdf.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -316,7 +316,7 @@
     // 1. Set OutputDev parameters and Generate contents
     // note: thread safety is set on 'false' for the GUI (this) thread
     kpdfOutputDev->setParams( request->width, request->height, genObjectRects, genObjectRects, false );
-    pdfdoc->displayPage( kpdfOutputDev, page->number() + 1, fakeDpiX, fakeDpiY, 0, false, true, genObjectRects );
+    pdfdoc->displayPage( kpdfOutputDev, page->number() + 1, fakeDpiX, fakeDpiY, 0, false, true, false );
     if ( genObjectRects )
         pdfdoc->processLinks( kpdfOutputDev, page->number() + 1 );
 
@@ -498,7 +498,7 @@
         docLock.lock();
         for ( ; pIt != pEnd; ++pIt )
         {
-            pdfdoc->displayPage(psOut, *pIt, 72, 72, 0, false, globalParams->getPSCrop(), gFalse);
+            pdfdoc->displayPage(psOut, *pIt, 72, 72, 0, false, globalParams->getPSCrop(), gTrue);
         }
         docLock.unlock();
 
@@ -1197,7 +1197,7 @@
     d->generator->kpdfOutputDev->setParams( width, height, 
                                             genObjectRects, genObjectRects, TRUE /*thread safety*/ );
     d->generator->pdfdoc->displayPage( d->generator->kpdfOutputDev, page->number() + 1,
-                                       fakeDpiX, fakeDpiY, 0, false, true, genObjectRects );
+                                       fakeDpiX, fakeDpiY, 0, false, true, false );
     if ( genObjectRects )
         d->generator->pdfdoc->processLinks( d->generator->kpdfOutputDev, page->number() + 1 );
 
Index: kpdf/xpdf/xpdf/Stream.cc
===================================================================
--- kpdf/xpdf/xpdf/Stream.cc	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/xpdf/xpdf/Stream.cc	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -411,9 +411,9 @@
 
   nVals = width * nComps;
   if (width <= 0 || nComps <= 0 || nBits <= 0 ||
-      nComps >= INT_MAX / nBits ||
-      width >= INT_MAX / nComps / nBits ||
-      nVals * nBits + 7 < 0) {
+      nComps > gfxColorMaxComps || nBits > 16 ||
+      width >= INT_MAX / nComps ||
+      nVals >= (INT_MAX - 7) / nBits) {
     return;
   }
   pixBytes = (nComps * nBits + 7) >> 3;
Index: kpdf/shell/kpdf.desktop
===================================================================
--- kpdf/shell/kpdf.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/shell/kpdf.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-MimeType=application/pdf
+MimeType=application/pdf;
 Terminal=false
 Name=KPDF
 Name[ar]=برنامج KPDF
Index: kpdf/ui/pagepainter.cpp
===================================================================
--- kpdf/ui/pagepainter.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/ui/pagepainter.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -49,11 +49,24 @@
     // if have no pixmap, draw blank page with gray cross and exit
     if ( !pixmap )
     {
-        if ( KpdfSettings::changeColors() &&
-             KpdfSettings::renderMode() == KpdfSettings::EnumRenderMode::Paper )
-            destPainter->fillRect( limits, KpdfSettings::paperColor() );
-        else
-            destPainter->fillRect( limits, Qt::white );
+        QColor color = Qt::white;
+        if ( KpdfSettings::changeColors() )
+        {
+            switch ( KpdfSettings::renderMode() )
+            {
+                case KpdfSettings::EnumRenderMode::Inverted:
+                    color = Qt::black;
+                    break;
+                case KpdfSettings::EnumRenderMode::Paper:
+                    color = KpdfSettings::paperColor();
+                    break;
+                case KpdfSettings::EnumRenderMode::Recolor:
+                    color = KpdfSettings::recolorBackground();
+                    break;
+                default: ;
+            }
+        }
+        destPainter->fillRect( limits, color );
 
         // draw a cross (to  that the pixmap as not yet been loaded)
         // helps a lot on pages that take much to render
Index: kpdf/ui/pageview.cpp
===================================================================
--- kpdf/ui/pageview.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/ui/pageview.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -59,53 +59,6 @@
 // definition of searchID for this class
 #define PAGEVIEW_SEARCH_ID 2
 
-class PageViewTip : public QToolTip
-{
-    public:
-        PageViewTip( PageView * view )
-            : QToolTip( view->viewport() ), m_view( view )
-        {
-        }
-
-        ~PageViewTip()
-        {
-            remove( m_view->viewport() );
-        }
-
-
-    protected:
-        void maybeTip( const QPoint &p );
-
-    private:
-        PageView * m_view;
-};
-
-void PageViewTip::maybeTip( const QPoint &_p )
-{
-    QPoint p( _p.x() + m_view->contentsX(), _p.y() + m_view->contentsY() );
-    PageViewItem * pageItem = m_view->pickItemOnPoint( p.x(), p.y() );
-    if ( pageItem )
-    {
-        double nX = (double)(p.x() - pageItem->geometry().left()) / (double)pageItem->width(),
-               nY = (double)(p.y() - pageItem->geometry().top()) / (double)pageItem->height();
-
-        // if over a ObjectRect (of type Link) change cursor to hand
-        const ObjectRect * object = pageItem->page()->hasObject( ObjectRect::Link, nX, nY );
-        if ( object )
-        {
-            // set tooltip over link's rect
-            KPDFLink *link = (KPDFLink *)object->pointer();
-            QString strtip = link->linkTip();
-            if ( !strtip.isEmpty() )
-            {
-                QRect linkRect = object->geometry( pageItem->width(), pageItem->height() );
-                linkRect.moveBy( - m_view->contentsX() + pageItem->geometry().left(), - m_view->contentsY() + pageItem->geometry().top() );
-                tip( linkRect, strtip );
-            }
-        }
-    }
-}
-
 // structure used internally by PageView for data storage
 class PageViewPrivate
 {
@@ -165,6 +118,55 @@
 
 
 
+class PageViewTip : public QToolTip
+{
+    public:
+        PageViewTip( PageView * view )
+            : QToolTip( view->viewport() ), m_view( view )
+        {
+        }
+
+        ~PageViewTip()
+        {
+            remove( m_view->viewport() );
+        }
+
+
+    protected:
+        void maybeTip( const QPoint &p );
+
+    private:
+        PageView * m_view;
+};
+
+void PageViewTip::maybeTip( const QPoint &_p )
+{
+    QPoint p( _p.x() + m_view->contentsX(), _p.y() + m_view->contentsY() );
+    PageViewItem * pageItem = m_view->pickItemOnPoint( p.x(), p.y() );
+    if ( pageItem && m_view->d->mouseMode == PageView::MouseNormal )
+    {
+        double nX = (double)(p.x() - pageItem->geometry().left()) / (double)pageItem->width(),
+               nY = (double)(p.y() - pageItem->geometry().top()) / (double)pageItem->height();
+
+        // if over a ObjectRect (of type Link) change cursor to hand
+        const ObjectRect * object = pageItem->page()->hasObject( ObjectRect::Link, nX, nY );
+        if ( object )
+        {
+            // set tooltip over link's rect
+            KPDFLink *link = (KPDFLink *)object->pointer();
+            QString strtip = link->linkTip();
+            if ( !strtip.isEmpty() )
+            {
+                QRect linkRect = object->geometry( pageItem->width(), pageItem->height() );
+                linkRect.moveBy( - m_view->contentsX() + pageItem->geometry().left(), - m_view->contentsY() + pageItem->geometry().top() );
+                tip( linkRect, strtip );
+            }
+        }
+    }
+}
+
+
+
 /* PageView. What's in this file? -> quick overview.
  * Code weight (in rows) and meaning:
  *  160 - constructor and creating actions plus their connected slots (empty stuff)
@@ -1605,7 +1607,7 @@
 {
     // detect the underlaying page (if present)
     PageViewItem * pageItem = pickItemOnPoint( p.x(), p.y() );
-    if ( pageItem )
+    if ( pageItem && d->mouseMode == MouseNormal )
     {
         double nX = (double)(p.x() - pageItem->geometry().left()) / (double)pageItem->width(),
                nY = (double)(p.y() - pageItem->geometry().top()) / (double)pageItem->height();
Index: kpdf/ui/toc.cpp
===================================================================
--- kpdf/ui/toc.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpdf/ui/toc.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -64,7 +64,8 @@
 #endif
     setSorting(-1);
     setRootIsDecorated(true);
-    setResizeMode(AllColumns);
+    // the next line causes bug:147233
+//    setResizeMode(AllColumns);
     setAllColumnsShowFocus(true);
     connect(this, SIGNAL(clicked(QListViewItem *)), this, SLOT(slotExecuted(QListViewItem *)));
     connect(this, SIGNAL(returnPressed(QListViewItem *)), this, SLOT(slotExecuted(QListViewItem *)));
Index: kfaxview/kfaxmultipage_tiff.desktop
===================================================================
--- kfaxview/kfaxmultipage_tiff.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfaxview/kfaxmultipage_tiff.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -45,6 +45,7 @@
 Comment[sr]=KViewShell-ов прикључак за факс фајлове
 Comment[sr@Latn]=KViewShell-ov priključak za faks fajlove
 Comment[sv]=Kviewshell-insticksprogram för telefaxfiler
+Comment[th]=ปลั๊กอินสำหรับแสดงแฟ้มโทรสารของ KViewShell
 Comment[tr]=Faks dosyaları için KViewShell eklentisi
 Comment[uk]=Втулок перегляду файлів факсів для KViewShell
 Comment[zh_CN]=传真文件的 KViewShell 插件
Index: kfaxview/kfaxmultipage.desktop
===================================================================
--- kfaxview/kfaxmultipage.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfaxview/kfaxmultipage.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -48,6 +48,7 @@
 Comment[sr]=KViewShell-ов прикључак за факс фајлове
 Comment[sr@Latn]=KViewShell-ov priključak za faks fajlove
 Comment[sv]=Kviewshell-insticksprogram för telefaxfiler
+Comment[th]=ปลั๊กอินสำหรับแสดงแฟ้มโทรสารของ KViewShell
 Comment[tr]=Faks dosyaları için KViewShell eklentisi
 Comment[uk]=Втулок перегляду файлів факсів для KViewShell
 Comment[zh_CN]=传真文件的 KViewShell 插件
Index: kfaxview/kfaxview.desktop
===================================================================
--- kfaxview/kfaxview.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfaxview/kfaxview.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -34,7 +34,7 @@
 GenericName[ms]=Pemapar Faks
 GenericName[nb]=Faksfremviser
 GenericName[nds]=Faxkieker
-GenericName[ne]=फ्याक्स दर्शक
+GenericName[ne]=फ्याक्स द्रष्टा
 GenericName[nl]=Faxweergaveprogramma
 GenericName[nn]=Faksvisar
 GenericName[pa]=ਫੈਕਸ ਦਰਸ਼ਕ
@@ -67,7 +67,7 @@
 Name[ne]=केडीई फ्याक्स दृश्य
 Name[sv]=Kfaxview
 Name[zh_TW]=KFaxView 傳真檢視
-MimeType=image/fax-g3
+MimeType=image/fax-g3;
 InitialPreference=6
 Exec=kfaxview %f
 Icon=kfaxview
Index: kview/kview.desktop
===================================================================
--- kview/kview.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kview/kview.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -72,7 +72,7 @@
 GenericName[zh_HK]=圖像檢視器
 GenericName[zh_TW]=影像檢視程式
 GenericName[zu]=Umbonisi Womfanekiso
-MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/x-bmp;image/png;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap;image/tiff;image/jp2
+MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/x-bmp;image/png;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap;image/tiff;image/jp2;
 Name=KView
 Name[af]=K-bekyk
 Name[ar]=برنامج KView
Index: kview/config/kview.setdlg
===================================================================
--- kview/config/kview.setdlg	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kview/config/kview.setdlg	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -193,7 +193,7 @@
 Comment[ms]=Di sini anda boleh set opsyen yang khas kepada aplikasi KView apabila dilancar secara tunggal. Apa yang anda ubah di sini tidak akan memberikan kesan kepada aplikasi lain.
 Comment[nb]=Her finner du innstillinger som bare påvirker KView som frittstående program. Disse innstillingene har ingenting å si for andre programmer.
 Comment[nds]=Hier laat sik de Instellen fastleggen, de bloots för dat KView-Enkelprogramm gellt. Wat Du hier ännerst, bedröppt keen anner Programmen.
-Comment[ne]=यहाँ तपाईँले स्यान्ड अलोन चलिरहेको बेलामा केडीई दृश्य अनुप्रयोगका लागि विशेष विकल्पहरू सेट गर्न सक्नुहुन्छ । तपाईँको यहाँको परिवर्तनले अन्य अनुप्रयोगलाई प्रभाव पार्ने छैन ।
+Comment[ne]=यहाँ तपाईँले स्यान्ड अलोन चलिरहेको बेलामा केडीई दृश्य अनुप्रयोगका लागि विशेष विकल्प सेट गर्न सक्नुहुन्छ । तपाईँको यहाँको परिवर्तनले अन्य अनुप्रयोगलाई प्रभाव पार्ने छैन ।
 Comment[nl]=Deze instellingen gelden alleen voor de KView-toepassing op zichzelf. Wat u hier instelt heeft geen invloed op andere toepassingen.
 Comment[nn]=Her finn du innstillingar som berre påverkar KView som frittståande program. Desse innstillingane har ingenting å seia for andre program.
 Comment[pl]=Tutaj można zmienić ustawienia, które będą używane, gdy KView jest uruchamiane jako oddzielny program. Zmiany dokonane tutaj nie będą miały wpływu na inne programy.
Index: kview/kviewviewer/kviewviewer.desktop
===================================================================
--- kview/kviewviewer/kviewviewer.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kview/kviewviewer/kviewviewer.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -89,7 +89,7 @@
 Name[ms]=Pemapar Imej KView
 Name[nb]=Bildeviseren KView
 Name[nds]=Bildkieker KView
-Name[ne]=के दृश्य छवि दर्शक
+Name[ne]=केडीई दृश्य छवि दर्शक
 Name[nl]=KView afbeeldingenweergave
 Name[nn]=Biletvisaren KView
 Name[pa]=KView ਚਿੱਤਰ ਦਰਸ਼ਕ
Index: kview/modules/effects/kvieweffects.desktop
===================================================================
--- kview/modules/effects/kvieweffects.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kview/modules/effects/kvieweffects.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -93,7 +93,7 @@
 Comment[ms]=Menyediakan beberapa kesan imej
 Comment[nb]=Utfører noen effekter på bilder
 Comment[nds]=Stellt en poor Bildeffekten praat
-Comment[ne]=केहि छवि असर प्रदान गर्दछ
+Comment[ne]=केही छवि असर प्रदान गर्दछ
 Comment[nl]=Biedt enkele effecten om afbeeldingen te bewerken
 Comment[nn]=Utfører nokre effektar på bilete
 Comment[pl]=Dodaje kilka efektów do obrazków
Index: libkscan/kscandevice.cpp
===================================================================
--- libkscan/kscandevice.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ libkscan/kscandevice.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -60,12 +60,13 @@
 {
 public:
     KScanDevicePrivate()
-	: currScanResolution(0)
+	: currScanResolutionX(0),
+	  currScanResolutionY(0)
 	{
 	    
 	}
 
-    int currScanResolution;
+    int currScanResolutionX, currScanResolutionY;
     
 };
 
@@ -788,13 +789,16 @@
    }
 
    /* Set scan resolution for preview. */
-   if( optionExists( SANE_NAME_SCAN_Y_RESOLUTION ) )
+   if( !optionExists( SANE_NAME_SCAN_Y_RESOLUTION ) )
+      d->currScanResolutionY = 0;
+   else
    {
       KScanOption yres ( SANE_NAME_SCAN_Y_RESOLUTION );
       /* if active ? */
       storeOptions->backupOption( yres );
       yres.set( set_dpi );
       apply( &yres );
+      yres.get( &d->currScanResolutionY );
 
       /* Resolution bind switch ? */
       if( optionExists( SANE_NAME_RESOLUTION_BIND ) )
@@ -811,8 +815,10 @@
    apply( &res );
 
    /* Store the resulting preview for additional image information */
-   res.get( &d->currScanResolution );
+   res.get( &d->currScanResolutionX );
 
+   if ( d->currScanResolutionY == 0 )
+      d->currScanResolutionY = d->currScanResolutionX;
 
    /* Start scanning */
    KScanStat stat = acquire_data( true );
@@ -914,6 +920,18 @@
  	        kdDebug(29000) << "Option <" << so->getName() << "> is not active !" << endl;
  	    }
  	}
+
+	/** Scan Resolution should always exist. **/
+	KScanOption res( SANE_NAME_SCAN_RESOLUTION );
+	res.get( &d->currScanResolutionX );
+        if ( !optionExists( SANE_NAME_SCAN_Y_RESOLUTION ) )
+           d->currScanResolutionY = d->currScanResolutionX;
+        else
+        {
+           KScanOption yres( SANE_NAME_SCAN_Y_RESOLUTION );
+           yres.get( &d->currScanResolutionY );
+        }
+
 	return( acquire_data( false ));
     }
     else
@@ -1169,13 +1187,14 @@
     if( status == KSCAN_OK && img )
     {
 	ImgScanInfo info;
-	/* FIXME: Both resolutions are equal here but should be
-	 *        x and y different
-	 */
-	info.setXResolution(d->currScanResolution);
-	info.setYResolution(d->currScanResolution);
+	info.setXResolution(d->currScanResolutionX);
+	info.setYResolution(d->currScanResolutionY);
 	info.setScannerName(shortScannerName());
 
+	// put the resolution also into the image itself
+	img->setDotsPerMeterX(static_cast<int>(d->currScanResolutionX / 0.0254 + 0.5));
+	img->setDotsPerMeterY(static_cast<int>(d->currScanResolutionY / 0.0254 + 0.5));
+
 	if( scanningPreview )
 	{
 	    kdDebug(29000) << "Scanning a preview !" << endl;
Index: libkscan/kscanoption.cpp
===================================================================
--- libkscan/kscanoption.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ libkscan/kscanoption.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -156,7 +156,7 @@
 	}
     }
 
-   return( desc > 0 );
+   return desc;
 }
 
 KScanOption::KScanOption( const KScanOption &so ) :
Index: kpovmodeler/main.cpp
===================================================================
--- kpovmodeler/main.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kpovmodeler/main.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -64,7 +64,6 @@
       shell = new PMShell;
       shell->show( );
    }
-
+   args->clear();   
    return app.exec( );
-   delete shell;
 }
Index: kgamma/kcmkgamma/kgamma.desktop
===================================================================
--- kgamma/kcmkgamma/kgamma.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kgamma/kcmkgamma/kgamma.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -42,6 +42,7 @@
 Comment[sv]=Kalibreringsverktyg för skärmen
 Comment[ta]=திரை நிலைக்கருவி
 Comment[tg]=Утилита барои калибратсия кардани монитор
+Comment[th]=เครื่องมือปรับความเที่ยงตรงของจอภาพ
 Comment[tr]=Monitör kalibrasyon aracı
 Comment[uk]=Засіб для калібрування монітора
 Comment[zh_CN]=监视器校准工具
Index: kdvi/squeeze.c
===================================================================
--- kdvi/squeeze.c	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kdvi/squeeze.c	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -54,8 +54,7 @@
  *   we simply put it out, since any special character terminates the
  *   preceding token.
  */
-void specialout(c)
-char c ;
+void specialout(char c)
 {
    if (linepos + 1 > LINELENGTH) {
       (void)fputs("\\n\\\n", out);
Index: kcoloredit/kcoloredit.desktop
===================================================================
--- kcoloredit/kcoloredit.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kcoloredit/kcoloredit.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -61,7 +61,7 @@
 GenericName[ta]=வண்ணகளஞ்சியம்
 GenericName[tg]=Муҳаррири палитра
 GenericName[th]=เครื่องมือแก้ไขจานสีของ KDE
-GenericName[tr]=Renk Paledi Düzenleyici
+GenericName[tr]=Renk Paleti Düzenleyici
 GenericName[uk]=Редактор палітри кольорів
 GenericName[ven]=Musengulusi wa phalete ya muvhala
 GenericName[wa]=Aspougneu del palete di coleurs
Index: kooka/main.cpp
===================================================================
--- kooka/main.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kooka/main.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -113,7 +113,7 @@
    kooka->show();
    app.processEvents();
    kooka->startup();
-
+   args->clear();
    int ret = app.exec();
 
    return ret;
Index: kuickshow/src/kuickshow.desktop
===================================================================
--- kuickshow/src/kuickshow.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kuickshow/src/kuickshow.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -18,7 +18,7 @@
 Exec=kuickshow %i %m -caption "%c" %U
 Icon=kuickshow
 Type=Application
-MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/png;image/tiff;image/x-bmp;image/x-psd;image/x-eim;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap
+MimeType=image/gif;image/x-xpm;image/x-xbm;image/jpeg;image/png;image/tiff;image/x-bmp;image/x-psd;image/x-eim;image/x-portable-bitmap;image/x-portable-pixmap;image/x-portable-greymap;
 DocPath=kuickshow/index.html
 Terminal=false
 InitialPreference=6
Index: kolourpaint/VERSION
===================================================================
--- kolourpaint/VERSION	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/VERSION	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1 +1 @@
-1.4.7_relight
+1.4.8_relight-pre
Index: kolourpaint/kpdocument.cpp
===================================================================
--- kolourpaint/kpdocument.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/kpdocument.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1141,7 +1141,12 @@
         if (m_selection->pixmap ())
             slotContentsChanged (m_selection->boundingRect ());
         else
+            // TODO: Should emit contentsChanged() instead?
+            //       I don't think it matters since contentsChanged() is
+            //       connected to updateViews() anyway (see
+            //       kpMainWindow::setDocument ()).
             vm->updateViews (m_selection->boundingRect ());
+
         delete m_selection;
     }
 
@@ -1187,8 +1192,18 @@
     if (m_selection->pixmap ())
         slotContentsChanged (m_selection->boundingRect ());
     else
+        // TODO: Should emit contentsChanged() instead?
+        //       I don't think it matters since contentsChanged() is
+        //       connected to updateViews() anyway (see
+        //       kpMainWindow::setDocument ()).
         vm->updateViews (m_selection->boundingRect ());
 
+    // There's no need to disconnect() the old selection since we:
+    //
+    // 1. Connect our _copy_ of the given selection.
+    // 2. We delete our copy when setSelection() is called again.
+    //
+    // See code above for both.
     connect (m_selection, SIGNAL (changed (const QRect &)),
              this, SLOT (slotContentsChanged (const QRect &)));
 
@@ -1395,6 +1410,8 @@
 
     if (!sel->isText ())
     {
+        // We can't use kpSelection::paint() since that always uses the
+        // transparent pixmap.
         paintPixmapAt (useTransparentPixmap ? sel->transparentPixmap () : sel->opaquePixmap (),
                        boundingRect.topLeft ());
     }
Index: kolourpaint/BUGS
===================================================================
--- kolourpaint/BUGS	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/BUGS	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1,9 +1,8 @@
 
-Please send bug reports and feature requests to http://bugs.kde.org/ or
-http://www.kolourpaint.org/support.  Don't hesitate to report bugs
-nor hesitate to send us your wishes - it provides valuable feedback that
-will help to improve future versions of KolourPaint and you will not
-receive flames for reporting duplicates.
+Please send bug reports and feature requests to http://bugs.kde.org/.
+Don't hesitate to report bugs nor hesitate to send us your wishes - it
+provides valuable feedback that will help to improve future versions of
+KolourPaint and you will not receive flames for reporting duplicates.
 
 
 This file lists known bugs in this version that are not considered
Index: kolourpaint/NEWS
===================================================================
--- kolourpaint/NEWS	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/NEWS	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -2,6 +2,12 @@
 KolourPaint 1.4_relight Series (branches/KDE/3.5/)
 ===============================
 
+KolourPaint 1.4.8_relight (Frozen ???)
+
+   * Add error dialogs for:
+    - if scanning support is unavailable
+    - running out of graphics memory during a scan
+
 KolourPaint 1.4.7_relight (Frozen 2007-05-14)
 
    * Save local files atomically - KolourPaint will no longer truncate
Index: kolourpaint/kpmainwindow_file.cpp
===================================================================
--- kolourpaint/kpmainwindow_file.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/kpmainwindow_file.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -432,35 +432,19 @@
         #if DEBUG_KP_MAIN_WINDOW
             kdDebug () << "\tcould not create scan dialog" << endl;
         #endif
-            // STRING: We really need a dialog here.  We can't due to string freeze.
-            //         In order to signal to the user -- without a dialog -- that
-            //         scanning support is not installed, we disable the action
-            //         in every window.
+            // Instead, we could try to create the scan dialog in the ctor
+            // and just disable the action in the first place, removing
+            // the need for this dialog.
             //
-            //         Or we could try to create the scan dialog in the ctor
-            //         and just disable the action in the first place.  But
-            //         this increases startup time and is also too risky on
-            //         the stable branch (e.g. if the scan support hangs,
-            //         KolourPaint would not be able to be started at all).
-
-            // TODO: PROPAGATE: interprocess
-            if (KMainWindow::memberList)
-            {
-                for (QPtrList <KMainWindow>::const_iterator it = KMainWindow::memberList->begin ();
-                     it != KMainWindow::memberList->end ();
-                     it++)
-                {
-                    kpMainWindow *mw = dynamic_cast <kpMainWindow *> (*it);
-                    Q_ASSERT (mw);
-
-                #if DEBUG_KP_MAIN_WINDOW
-                    kdDebug () << "\t\tmw=" << mw << endl;
-                #endif
-
-                    mw->m_actionScan->setEnabled (false);
-                }
-            }
-
+            // But this increases startup time and is a bit risky e.g. if
+            // the scan support hangs, KolourPaint would not be able to be
+            // started at all.
+            //
+            // Also, disabling the action is bad because the scan support
+            // can be installed while KolourPaint is still running.
+            KMessageBox::sorry (this,
+                         i18n ("Scanning support is not installed."),
+                         i18n ("No Scanning Support"));
             return;
         }
 
@@ -531,8 +515,6 @@
     kpDocumentMetaInfo metaInfo;
     const QPixmap pixmap = kpDocument::convertToPixmapAsLosslessAsPossible (
         image,
-        // STRING: After string freeze, have a custom message about the
-        //         "scanned image".
         kpMainWindow::pasteWarnAboutLossInfo (),
         &saveOptions,
         &metaInfo);
@@ -542,9 +524,9 @@
     #if DEBUG_KP_MAIN_WINDOW
         kdDebug () << "\tcould not convert to pixmap" << endl;
     #endif
-        // STRING: After string freeze, we need a message like
-        //         "out of graphics memory" (see
-        //         kpDocument::getPixmapFromFile()).
+        KMessageBox::sorry (this,
+                            i18n ("Cannot scan - out of graphics memory."),
+                            i18n ("Cannot Scan"));
         return;
     }
 
Index: kolourpaint/kolourpaint.cpp
===================================================================
--- kolourpaint/kolourpaint.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/kolourpaint.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -65,14 +65,7 @@
         I18N_NOOP ("Paint Program for KDE"),
         KAboutData::License_Custom,
         0/*copyright statement - see licence instead*/,
-#if 1
-        0/*free text*/,
-// STRING: After string freeze, make below the "free text" but point to
-//         http://www.kolourpaint.org/support.
-#else
-        I18N_NOOP ("Support / Feedback:\n"
-                   "kolourpaint-support@lists.sourceforge.net\n"),
-#endif
+        0/*no free text*/,
         "http://www.kolourpaint.org/"
     );
 
@@ -81,32 +74,6 @@
     aboutData.setLicenseText (kpLicenseText);
 
 
-// STRING: After string freeze, make below the custom author text but point to
-//         http://www.kolourpaint.org/support.
-#if 0
-    aboutData.setCustomAuthorText (
-        I18N_NOOP
-        (
-            "\n"
-            "For support, or to report bugs and feature requests, please email\n"
-            "<kolourpaint-support@lists.sourceforge.net>"
-            " - the free and friendly\n"
-            "KolourPaint support service.\n"
-            "\n"
-        ),
-        I18N_NOOP
-        (
-            "<qt>"
-            "For support, or to report bugs and feature requests, please email<br>"
-            "<a href=\"mailto:kolourpaint-support@lists.sourceforge.net\">kolourpaint-support@lists.sourceforge.net</a>"
-            " - the free and friendly<br>"
-            "KolourPaint support service.<br>"
-            "<br>"
-            "</qt>"
-        ));
-#endif
-
-
     // SYNC: with AUTHORS
 
     aboutData.addAuthor ("Clarence Dang", I18N_NOOP ("Maintainer"), "dang@kde.org");
Index: kolourpaint/pixmapfx/kpeffectbalance.cpp
===================================================================
--- kolourpaint/pixmapfx/kpeffectbalance.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/pixmapfx/kpeffectbalance.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -274,6 +274,8 @@
     m_gammaInput->setRange (-50, 50, 1/*step*/, true/*slider*/);
     // TODO: This is what should be shown in the m_gammaInput spinbox
     m_gammaLabel = new QLabel (this);
+    // TODO: This doesn't seem to be wide enough with some fonts so the
+    //       whole layout moves when we drag the gamma slider.
     m_gammaLabel->setMinimumWidth (m_gammaLabel->fontMetrics ().width (" 10.00 "));
     m_gammaLabel->setAlignment (m_gammaLabel->alignment () | Qt::AlignRight);
     QPushButton *gammaResetPushButton = new QPushButton (i18n ("Rese&t"), this);
Index: kolourpaint/kolourpaint.desktop
===================================================================
--- kolourpaint/kolourpaint.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/kolourpaint.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -83,7 +83,7 @@
 # SYNC: Run branches/kolourpaint/control/scripts/gen_mimetype_line.sh in
 #       the version of kdelibs/kimgio/ (e.g. KDE 3.5) KolourPaint is
 #       shipped with.
-MimeType=image/fax-g3;image/gif;image/jp2;image/jpeg;image/png;image/tiff;image/x-bmp;image/x-dds;image/x-eps;image/x-exr;image/x-hdr;image/x-ico;image/x-pcx;image/x-portable-bitmap;image/x-portable-greymap;image/x-portable-pixmap;image/x-rgb;image/x-targa;image/x-vnd.adobe.photoshop;image/x-xbm;image/x-xcf-gimp;image/x-xpm;video/x-mng
+MimeType=image/fax-g3;image/gif;image/jp2;image/jpeg;image/png;image/tiff;image/x-bmp;image/x-dds;image/x-eps;image/x-exr;image/x-hdr;image/x-ico;image/x-pcx;image/x-portable-bitmap;image/x-portable-greymap;image/x-portable-pixmap;image/x-rgb;image/x-targa;image/x-vnd.adobe.photoshop;image/x-xbm;image/x-xcf-gimp;image/x-xpm;video/x-mng;
 
 Categories=Qt;KDE;Graphics;
 Terminal=false
Index: kolourpaint/README
===================================================================
--- kolourpaint/README	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kolourpaint/README	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1,5 +1,5 @@
 
-KolourPaint Version 1.4.7_relight (KDE 3.5.7 Release Frozen 2007-05-14)
+KolourPaint Version 1.4.8_relight (KDE 3.5.8 Release Frozen ???)
 http://www.kolourpaint.org/
 
 Copyright (c) 2003,2004,2005,2006 Clarence Dang <dang@kde.org>
@@ -85,7 +85,7 @@
 4. Support
 ==========
 
-Visit: http://www.kolourpaint.org/support
+Visit: http://www.kolourpaint.org/
 
 If you have any questions about compiling, installing or using KolourPaint,
 don't be afraid to contact us.  We try to support all versions of
@@ -95,9 +95,8 @@
 5. Feedback
 ===========
 
-Please send bug reports and feature requests to http://bugs.kde.org/ or
-http://www.kolourpaint.org/support.  Don't hesitate to report bugs nor
-hesitate to send us your wishes - it provides valuable feedback that will
-help to improve future versions of KolourPaint and you will not receive
-flames for reporting duplicates.
+Please send bug reports and feature requests to http://bugs.kde.org/.
+Don't hesitate to report bugs nor hesitate to send us your wishes - it
+provides valuable feedback that will help to improve future versions of
+KolourPaint and you will not receive flames for reporting duplicates.
 
Index: ksnapshot/ksnapshot.desktop
===================================================================
--- ksnapshot/ksnapshot.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ ksnapshot/ksnapshot.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -71,7 +71,7 @@
 Name[fr]=KSnapShot
 Name[hi]=के-स्नेपशॉट
 Name[lv]=KSnapšots
-Name[ne]=केडीई स्न्यापशट
+Name[ne]=केडीई स्न्यापसट
 Name[pl]=Zrzuty ekranu
 Name[sv]=Ksnapshot
 Name[ta]=கேதிரையை நகலெடுத்தல்
Index: ksnapshot/main.cpp
===================================================================
--- ksnapshot/main.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ ksnapshot/main.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -67,6 +67,7 @@
   else
      toplevel = new KSnapshot();
 
+  args->clear();
   app.dcopClient()->setDefaultObject( toplevel->objId() );
   toplevel->setCaption( app.makeStdCaption("") );
   app.setMainWidget(toplevel);
Index: kviewshell/plugins/djvu/libdjvu/DataPool.cpp
===================================================================
--- kviewshell/plugins/djvu/libdjvu/DataPool.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kviewshell/plugins/djvu/libdjvu/DataPool.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -1438,7 +1438,7 @@
         OpenFiles::get()->stream_released(f->stream, this);
       }
       fstream=0;
-   } else DEBUG_MSG("Not connected\n");
+   } else { DEBUG_MSG("Not connected\n"); }
 }
 
 void
Index: kmrml/kmrml/mrml.desktop
===================================================================
--- kmrml/kmrml/mrml.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kmrml/kmrml/mrml.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -44,6 +44,7 @@
 Comment[sv]=Multimedia Retrieval Markup Language-dokument
 Comment[ta]=பல் ஊடக திரும்பப்பெறு அடையாள மொழி ஆவணம்
 Comment[tg]=Санади MRML (Multimedia Retrieval Markup Language)
+Comment[th]=เอกสาร Multimedia Retrieval Markup Language
 Comment[tr]=Multimedia Retrieval Markup Language Belgesi
 Comment[uk]=Документ формату зберігання мультимедіа
 Comment[ven]=Manwalwa a luambo lwau humbula zwa khasho nnzhi
Index: kmrml/kmrml/kcontrol/kcmkmrml.desktop
===================================================================
--- kmrml/kmrml/kcontrol/kcmkmrml.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kmrml/kmrml/kcontrol/kcmkmrml.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -68,7 +68,7 @@
 
 Comment=Configuration for using the GNU Image Finding Tool
 Comment[ar]=اعدادات لاستخدام أداة GNU للبحث عن الصور
-Comment[bg]=Настройки на програмата за индексиране и търсене на изображения
+Comment[bg]=Настройване на програмата за индексиране и търсене на изображения
 Comment[bs]=Podešavanje za upotrebu GNU Alata za pronalaženje slika
 Comment[ca]=Configuració per a l'ús de l'eina de cerca d'imatges GNU
 Comment[cs]=Konfigurace používání nástroje GNU Image Finding Tool
Index: kmrml/kmrml/mrml-servicemenu.desktop
===================================================================
--- kmrml/kmrml/mrml-servicemenu.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kmrml/kmrml/mrml-servicemenu.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -50,6 +50,7 @@
 Name[sv]=Sök efter liknande bilder...
 Name[ta]=இதே போன்ற பிம்பங்களை தேடுக...
 Name[tg]=Ҷустуҷӯи тасвироти якхела...
+Name[th]=ค้นหาภาพที่เหมือนกัน...
 Name[tr]=Benzer Resimleri Ara...
 Name[uk]=Пошук схожих зображень...
 Name[uz]=Ўхшаш расмларни қидириш
Index: kmrml/kmrml/mrml_part.desktop
===================================================================
--- kmrml/kmrml/mrml_part.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kmrml/kmrml/mrml_part.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -45,6 +45,7 @@
 Name[sr@Latn]=MRML prikazivač
 Name[ta]=MRML காட்சி
 Name[tg]=Намоиши MRML
+Name[th]=ดู MRML
 Name[tr]=MRML Görünümü
 Name[uk]=Перегляд MRML
 Name[ven]=Mbonalelo ya MRML
Index: kmrml/kmrml/lib/kmrml_config.cpp
===================================================================
--- kmrml/kmrml/lib/kmrml_config.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kmrml/kmrml/lib/kmrml_config.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -291,7 +291,7 @@
 ServerSettings ServerSettings::defaults()
 {
     return ServerSettings( DEFAULT_HOST, DEFAULT_PORT,
-                           (QString(DEFAULT_HOST) == "localhost" && DEFAULT_PORT),
+                           (!strcmp(DEFAULT_HOST, "localhost") && DEFAULT_PORT),
                            DEFAULT_AUTH, DEFAULT_USER, DEFAULT_PASS );
 }
 
Index: kmrml/kmrml/server/daemonwatcher.desktop
===================================================================
--- kmrml/kmrml/server/daemonwatcher.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kmrml/kmrml/server/daemonwatcher.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -29,7 +29,7 @@
 Name[ms]=Pemerhati Daemon KDED KMRML
 Name[nb]=KDED KMRML nisseovervåker
 Name[nds]=KMRML-Luerdämoon för KDED
-Name[ne]=KDED KMRML डेइमोन दर्शक
+Name[ne]=KDED KMRML डेइमन दर्शक
 Name[nl]=KDED KMRML-daemonbeheer
 Name[nn]=KDED KMRML-nisseovervaking
 Name[pl]=Monitor usług KMRML
@@ -45,6 +45,7 @@
 Name[sv]=KDED KMRML-demonbevakare
 Name[ta]=KDED டிமென் வாட்சர்
 Name[tg]=Мудири демони KDED KMRML
+Name[th]=ตัวเฝ้าดูแดมอน KDED KMRML
 Name[tr]=KDED KMRML Aracı İzleyici
 Name[uk]=Спостерігач демону KDED KMRML
 Name[zh_CN]=KDED KMRML 守护程序监视器
@@ -76,7 +77,7 @@
 Comment[ms]=Mulakan daemons atas permintaan dan mula semula atas kegagalan
 Comment[nb]=Starter nisser på forespørsler og starter dem igjen ved feil.
 Comment[nds]=Start Achtergrundperzessen op Nafraag un bi Fehlers nieg
-Comment[ne]=माग गरेको बेलामा डेइमोन सुरु गर्दछ र अफफल भएमा फेरि सुरु गर्दछ
+Comment[ne]=माग गरेको बेलामा डेइमन सुरु गर्दछ र अफफल भएमा फेरि सुरु गर्दछ
 Comment[nl]=Start achtergrondprogramma's op en herstart deze indien nodig
 Comment[nn]=Startar nissar når dei trengst og startar dei om att ved feil
 Comment[pl]=Uruchamia usługi na żądanie i wznawia je po awarii
Index: kfax/kfax.desktop
===================================================================
--- kfax/kfax.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfax/kfax.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -35,7 +35,7 @@
 GenericName[ms]=Pemapar Faks
 GenericName[nb]=Faksfremviser
 GenericName[nds]=Faxkieker
-GenericName[ne]=फ्याक्स दर्शक
+GenericName[ne]=फ्याक्स द्रष्टा
 GenericName[nl]=Faxweergaveprogramma
 GenericName[nn]=Faksvisar
 GenericName[pa]=ਫੈਕਸ ਦਰਸ਼ਕ
@@ -73,7 +73,6 @@
 Name[ne]=केडीई फ्याक्स
 Name[sv]=Kfax
 Name[ta]=கேஃபாக்ஸ்
-Name[th]=โทรสาร - K
 Name[ven]=Fekisi ya K
 Name[wa]=KFacs
 Name[zh_TW]=KFax 傳真檢視器
Index: kghostview/kgv_miniwidget.cpp
===================================================================
--- kghostview/kgv_miniwidget.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kghostview/kgv_miniwidget.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -38,6 +38,7 @@
 #endif
 
 #include <config.h>
+#include <stdlib.h>
 
 #include "infodialog.h"
 #include "marklist.h"
Index: kghostview/dscparse.cpp
===================================================================
--- kghostview/dscparse.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kghostview/dscparse.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -83,7 +83,7 @@
 #define CDSC_NEEDMORE 11
 
 template<class T>
-inline const T min (const T& a, const T& b) { return a < b ? a : b; }
+inline T min (T a, T b) { return a < b ? a : b; }
 
 /* local prototypes */
 dsc_private void * dsc_memalloc(P2(CDSC *dsc, size_t size));
Index: kghostview/main.cpp
===================================================================
--- kghostview/main.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kghostview/main.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -64,6 +64,7 @@
 	}
 	shell->show();
     }
+    args->clear();
     return app.exec();
 }
 
Index: kghostview/kghostview_part.desktop
===================================================================
--- kghostview/kghostview_part.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kghostview/kghostview_part.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -9,7 +9,6 @@
 Name[ne]=केडीई घोस्ट दृश्य
 Name[sv]=Kghostview
 Name[ta]=கேமற்றொரு காட்சி
-Name[th]=อ่านแฟ้มโพสต์สคริปต์ - K
 Name[tr]=GhostView
 Name[ven]=Mbonalelo ya tshipuku tsha K
 Name[xh]=Imboniselo yeKGhost
Index: kghostview/kghostview.desktop
===================================================================
--- kghostview/kghostview.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kghostview/kghostview.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -9,11 +9,10 @@
 Name[ne]=केडीई घोस्ट दृश्य
 Name[sv]=Kghostview
 Name[ta]=கேமற்றொரு காட்சி
-Name[th]=อ่านแฟ้มโพสต์สคริปต์ - K
 Name[tr]=GhostView
 Name[ven]=Mbonalelo ya tshipuku tsha K
 Name[xh]=Imboniselo yeKGhost
-MimeType=application/pdf;application/postscript;image/x-eps;application/x-gzpostscript;application/illustrator
+MimeType=application/pdf;application/postscript;image/x-eps;application/x-gzpostscript;application/illustrator;
 InitialPreference=6
 Exec=kghostview %u -caption "%c" %i %m
 Icon=kghostview
Index: kamera/kcontrol/kamera.cpp
===================================================================
--- kamera/kcontrol/kamera.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kamera/kcontrol/kamera.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -41,12 +41,6 @@
 #include "kamera.h"
 #include "kamera.moc"
 
-// XXX HACK HACK HACK
-// XXX All tocstr(string) references can be safely replaced with
-// XXX string.latin1() as soon as the gphoto2 API uses 'const char *'
-// XXX instead of 'char *' in calls that don't modify the string
-#define tocstr(x) ((char *)((x).latin1()))
-
 typedef KGenericFactory<KKameraConfig, QWidget> KKameraConfigFactory;
 K_EXPORT_COMPONENT_FACTORY( kcm_kamera, KKameraConfigFactory( "kcmkamera" ) )
 
Index: kamera/kcontrol/kamera.desktop
===================================================================
--- kamera/kcontrol/kamera.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kamera/kcontrol/kamera.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -4,7 +4,7 @@
 Comment[af]=Konfigureer Kamera
 Comment[ar]=إعداد Kamera
 Comment[az]=Kameranı Quraşdır
-Comment[bg]=Настройки на цифров фотоапарат
+Comment[bg]=Настройване на цифров фотоапарат
 Comment[br]=Kefluniañ Kamera
 Comment[bs]=Podesi kameru
 Comment[ca]=Configura Kamera
@@ -57,7 +57,7 @@
 Comment[sv]=Anpassa kamera
 Comment[ta]=காமிராவை அமை
 Comment[tg]=Танзимоти камера
-Comment[th]=ปรับแต่งกล้อง
+Comment[th]=ปรับแต่ง Kamera
 Comment[tr]=Kamera'yı Yapılandır
 Comment[uk]=Налаштувати Kamera
 Comment[uz]=Фотоапаратни мослаш
Index: kamera/kioslave/kamera.cpp
===================================================================
--- kamera/kioslave/kamera.cpp	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kamera/kioslave/kamera.cpp	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -2,7 +2,7 @@
 
     Copyright (C) 2001 The Kompany
 		  2001-2003	Ilya Konstantinov <kde-devel@future.shiny.co.il>
-		  2001-2003	Marcus Meissner <marcus@jet.franken.de>
+		  2001-2007	Marcus Meissner <marcus@jet.franken.de>
 
     This program is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <fcntl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <signal.h>
@@ -33,6 +34,7 @@
 
 #include <kdebug.h>
 #include <kinstance.h>
+#include <kstandarddirs.h>
 #include <kconfig.h>
 #include <ksimpleconfig.h>
 #include <klocale.h>
@@ -43,6 +45,8 @@
 
 #include "kamera.h"
 
+#define MAXIDLETIME	30	/* seconds */
+
 #define tocstr(x) ((x).local8Bit())
 
 using namespace KIO;
@@ -63,18 +67,15 @@
 
 int kdemain(int argc, char **argv)
 {
-	KInstance instance("kio_kamera");
+	KInstance	instance("kio_kamera");
 
 	if(argc != 4) {
 		kdDebug(7123) << "Usage: kio_kamera protocol "
 			     "domain-socket1 domain-socket2" << endl;
 		exit(-1);
 	}
-
 	KameraProtocol slave(argv[2], argv[3]);
-
 	slave.dispatchLoop();
-
 	return 0;
 }
 
@@ -88,10 +89,44 @@
 	m_file = NULL;
 	m_config = new KSimpleConfig(KProtocolInfo::config("camera"));
 	m_context = gp_context_new();
+	actiondone = true;
+	cameraopen = false;
+	m_lockfile = locateLocal("tmp", "kamera");
+	idletime = 0;
 }
 
+// This handler is getting called every second. We use it to do the
+// delayed close of the camera.
+// Logic is:
+// 	- No more requests in the queue (signaled by actiondone) AND
+//		- We are MAXIDLETIME seconds idle OR
+//		- Another slave wants to have access to the camera.
+//
+// The existance of a lockfile is used to signify "please give up camera".
+//
+void KameraProtocol::special(const QByteArray&) {
+	kdDebug(7123) << "KameraProtocol::special() at " << getpid() << endl;
+
+	if (!actiondone && cameraopen) {
+		struct stat	stbuf;
+		if ((-1!=::stat(m_lockfile.utf8(),&stbuf)) || (idletime++ >= MAXIDLETIME)) {
+			kdDebug(7123) << "KameraProtocol::special() closing camera." << endl;
+			closeCamera();
+			setTimeoutSpecialCommand(-1);
+		} else {
+			// continue to wait
+			setTimeoutSpecialCommand(1);
+		}
+	} else {
+		// We let it run until the slave gets no actions anymore.
+		setTimeoutSpecialCommand(1);
+	}
+	actiondone = false;
+}
+
 KameraProtocol::~KameraProtocol()
 {
+	kdDebug(7123) << "KameraProtocol::~KameraProtocol()" << endl;
 	delete m_config;
 	if(m_camera) {
 		closeCamera();
@@ -101,21 +136,34 @@
 }
 
 // initializes the camera for usage - should be done before operations over the wire
-bool KameraProtocol::openCamera(void) {
+bool KameraProtocol::openCamera(QString &str) {
+	idletime = 0;
+	actiondone = true;
 	if (!m_camera) {
 		reparseConfiguration();
 	} else {
-		int ret, tries = 15;
-		
-		while (tries--) {
-			ret = gp_camera_init(m_camera, m_context);
-			if (	(ret == GP_ERROR_IO_USB_CLAIM) || 
-				(ret == GP_ERROR_IO_LOCK)) {
-				sleep(1);
-				continue;
+		if (!cameraopen) {
+			int ret, tries = 15;
+			kdDebug(7123) << "KameraProtocol::openCamera at " << getpid() << endl;
+			while (tries--) {
+				ret = gp_camera_init(m_camera, m_context);
+				if (	(ret == GP_ERROR_IO_USB_CLAIM) || 
+					(ret == GP_ERROR_IO_LOCK)) {
+					// just create / touch if not there
+					int fd = ::open(m_lockfile.utf8(),O_CREAT|O_WRONLY,0600);
+					if (fd != -1) ::close(fd);
+					::sleep(1);
+					kdDebug(7123) << "openCamera at " << getpid() << "- busy, ret " << ret << ", trying again." << endl;
+					continue;
+				}
+				if (ret == GP_OK) break;
+				str = gp_result_as_string(ret);
+				return false;
 			}
-			if (ret == GP_OK) break;
-			return false;
+			::unlink(m_lockfile.utf8());
+			setTimeoutSpecialCommand(1);
+			kdDebug(7123) << "openCamera succeeded at " << getpid() << endl;
+			cameraopen = true;
 		}
 	}
 	return true;
@@ -129,12 +177,14 @@
 	if (!m_camera)
 		return;
 
+	kdDebug(7123) << "KameraProtocol::closeCamera at " << getpid() << endl;
 	if ((gpr=gp_camera_exit(m_camera,m_context))!=GP_OK) {
 		kdDebug(7123) << "closeCamera failed with " << gp_result_as_string(gpr) << endl;
 	}
 	// HACK: gp_camera_exit() in gp 2.0 does not close the port if there
 	//       is no camera_exit function.
 	gp_port_close(m_camera->port);
+	cameraopen = false;
 	return;
 }
 
@@ -183,7 +233,6 @@
 		processedSize(strlen(xx.text));				\
 		chunkDataBuffer.resetRawData(xx.text, strlen(xx.text));	\
 		finished();						\
-		closeCamera();						\
 		return;							\
 	}
 
@@ -204,12 +253,11 @@
 	gpr = gp_camera_file_get_info(m_camera, tocstr(fix_foldername(url.directory(false))), tocstr(url.fileName()), &info, m_context);
 	if (gpr != GP_OK) {
 		// fprintf(stderr,"Folder %s / File %s not found, gpr is %d\n",folder.latin1(), url.fileName().latin1(), gpr);
-		gp_file_free(m_file);
+		gp_file_unref(m_file);
 		if ((gpr == GP_ERROR_FILE_NOT_FOUND) || (gpr == GP_ERROR_DIRECTORY_NOT_FOUND))
 			error(KIO::ERR_DOES_NOT_EXIST, url.path());
 		else
 			error(KIO::ERR_UNKNOWN, gp_result_as_string(gpr));
-		closeCamera();
 		return;
 	}
 
@@ -246,16 +294,14 @@
 			break;
 		case GP_ERROR_FILE_NOT_FOUND:
 		case GP_ERROR_DIRECTORY_NOT_FOUND:
-			gp_file_free(m_file);
+			gp_file_unref(m_file);
 			m_file = NULL;
 			error(KIO::ERR_DOES_NOT_EXIST, url.fileName());
-			closeCamera();
 			return ;
 		default:
-			gp_file_free(m_file);
+			gp_file_unref(m_file);
 			m_file = NULL;
 			error(KIO::ERR_UNKNOWN, gp_result_as_string(gpr));
-			closeCamera();
 			return;
 	}
 	// emit the mimetype
@@ -276,7 +322,6 @@
 		gp_file_free(m_file);
 		m_file = NULL;
 		error(KIO::ERR_UNKNOWN, gp_result_as_string(gpr));
-		closeCamera();
 		return;
 	}
 	// make sure we're not sending zero-sized chunks (=EOF)
@@ -306,9 +351,8 @@
 	}
 
 	finished();
-	gp_file_free(m_file);
+	gp_file_unref(m_file); /* just unref, might be stored in fs */
 	m_file = NULL;
-	closeCamera();
 }
 
 // The KIO slave "stat" function.
@@ -356,6 +400,10 @@
 	statEntry(entry);
 
 	finished();
+
+	// This call happens on autodetect by kdemm. So close the camera, but
+	// only if no more requests are pending. 
+	idletime = MAXIDLETIME;
 }
 
 // Implements a regular stat() of a file / directory, returning all we know about it
@@ -384,7 +432,6 @@
 		else
 			error(KIO::ERR_UNKNOWN, gp_result_as_string(gpr));
 		gp_list_free(dirList);
-		closeCamera();
 		return;
 	}
 
@@ -399,7 +446,6 @@
 		translateTextToUDS(entry,#xx".txt",xx.text);		\
 		statEntry(entry);					\
 		finished();						\
-		closeCamera();						\
 		return;							\
 	}
 	GPHOTO_TEXT_FILE(about);
@@ -416,7 +462,6 @@
 			translateDirectoryToUDS(entry, url.fileName());
 			statEntry(entry);
 			finished();
-			closeCamera();
 			return;
 		}
 	}
@@ -430,13 +475,11 @@
 			error(KIO::ERR_DOES_NOT_EXIST, url.path());
 		else
 			error(KIO::ERR_UNKNOWN, gp_result_as_string(gpr));
-		closeCamera();
 		return;
 	}
 	translateFileToUDS(entry, info, url.fileName());
 	statEntry(entry);
 	finished();
-	closeCamera();
 }
 
 // The KIO slave "del" function.
@@ -465,7 +508,6 @@
 			finished();
 		}
 	}
-	closeCamera();
 }
 
 // The KIO slave "listDir" function.
@@ -643,7 +685,6 @@
 	gpr = readCameraFolder(url.path(), dirList, fileList);
 	if(gpr != GP_OK) {
 		kdDebug(7123) << "read Camera Folder failed:" << gp_result_as_string(gpr) <<endl;
-		closeCamera();
 		gp_list_free(dirList);
 		gp_list_free(fileList);
 		gp_list_free(specialList);
@@ -687,7 +728,6 @@
 		}
 	}
 
-	closeCamera();
 
 	gp_list_free(fileList);
 	gp_list_free(dirList);
@@ -764,18 +804,14 @@
 		gp_camera_set_port_info(m_camera, port_info);
 		gp_camera_set_port_speed(m_camera, 0); // TODO: the value needs to be configurable
 		kdDebug(7123) << "Opening camera model " << user << " at " << host << endl;
-#if 0
-		// initialize the camera (might take time on a non-existant or disconnected camera)
-		gpr = gp_camera_init(m_camera, m_context);
-		if(gpr != GP_OK) {
-			gp_camera_unref(m_camera);
-			m_camera = NULL;
+
+		QString errstr;
+		if (!openCamera(errstr)) {
 			kdDebug(7123) << "Unable to init camera: " << gp_result_as_string(gpr) << endl;
-			error(KIO::ERR_UNKNOWN, gp_result_as_string(gpr));
+			error(KIO::ERR_SERVICE_NOT_AVAILABLE, errstr);
+			gp_camera_exit(m_camera, m_context);
 			return;
 		}
-		gp_camera_exit(m_camera, m_context);
-#endif
 	}
 }
 
Index: kamera/kioslave/kamera.h
===================================================================
--- kamera/kioslave/kamera.h	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kamera/kioslave/kamera.h	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -2,7 +2,7 @@
 
     Copyright (C) 2001 The Kompany
 		  2001-2003	Ilya Konstantinov <kde-devel@future.shiny.co.il>
-		  2001-2003	Marcus Meissner <marcus@jet.franken.de>
+		  2001-2007	Marcus Meissner <marcus@jet.franken.de>
 
     This program is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -38,12 +38,13 @@
 	virtual void get(const KURL &url);
 	virtual void stat(const KURL &url);
 	virtual void del(const KURL &url, bool isFile);
-        virtual void setHost(const QString& host, int port, const QString& user, const QString& pass );
+	virtual void setHost(const QString& host, int port, const QString& user, const QString& pass );
 	virtual void listDir(const KURL &url);
+	virtual void special(const QByteArray &data);
 
-        CameraFile *getFile() { return m_file; }
-        int getFileSize() { return m_fileSize; }
-        void setFileSize(int newfs) { m_fileSize = newfs; }
+	CameraFile *getFile() { return m_file; }
+	int getFileSize() { return m_fileSize; }
+	void setFileSize(int newfs) { m_fileSize = newfs; }
 
 private:
 	Camera *m_camera;
@@ -53,26 +54,28 @@
 	GPContext	*m_context;
 
 	void reparseConfiguration(void);
-	bool openCamera(void);
+	bool openCamera(QString& str);
+	bool openCamera(void ) {
+		QString errstr;
+		return openCamera(errstr);
+	}
 	void closeCamera(void);
 
 	void statRoot(void);
 	void statRegular(const KURL &url);
-        void translateTextToUDS(KIO::UDSEntry &udsEntry, const QString &info, const char *txt);
-        void translateFileToUDS(KIO::UDSEntry &udsEntry, const CameraFileInfo &info, QString name);
+	void translateTextToUDS(KIO::UDSEntry &udsEntry, const QString &info, const char *txt);
+	void translateFileToUDS(KIO::UDSEntry &udsEntry, const CameraFileInfo &info, QString name);
 	void translateDirectoryToUDS(KIO::UDSEntry &udsEntry, const QString &dirname);
 	bool cameraSupportsPreview(void);
 	bool cameraSupportsDel(void);
 	bool cameraSupportsPut(void);
 	int readCameraFolder(const QString &folder, CameraList *dirList, CameraList *fileList);
-	QString lockFileName();
-	void lock();
-	void unlock();
 
-	QString m_cfgModel;
-	QString m_cfgPath;
+	QString m_lockfile;
+	int	idletime;
 
 	int m_fileSize;
 	CameraFile *m_file;
+	bool actiondone, cameraopen;
 };
 #endif
Index: kruler/eventsrc
===================================================================
--- kruler/eventsrc	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kruler/eventsrc	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -163,7 +163,7 @@
 Comment[mt]=Ir-riga mxiet b'tikka waħda permezz tal-buttuni vleġeġ
 Comment[nb]=Linjalen ble flyttet punktvis ved bruk av piltastene
 Comment[nds]=Dat Lineaal warrt mit de Pieltasten pixelwies verschaven
-Comment[ne]=कर्सर कुञ्जी प्रयोग गरेर रूलर पिक्सवाइज सारियो
+Comment[ne]=कर्सर कुञ्जी प्रयोग गरेर रूलर पिक्सेलअनुसार सारियो
 Comment[nl]=De liniaal is enkele pixels verplaatst via de cursortoetsen
 Comment[nn]=Linjalen vert flytta piksel for piksel med piltastane
 Comment[nso]=Molaodi o tsamaile ka mokgwa wa pixelwise a somisa ditobetswa tsa cursor
Index: kruler/kruler.desktop
===================================================================
--- kruler/kruler.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kruler/kruler.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -68,7 +68,7 @@
 Name[is]=KReglustika
 Name[lv]=KRulers
 Name[ms]=KPembaris
-Name[ne]=के डी ई रूलर
+Name[ne]=केडीई रूलर
 Name[pl]=Linijka
 Name[pt_BR]=KRégua
 Name[ro]=Riglă
Index: kiconedit/kiconedit.desktop
===================================================================
--- kiconedit/kiconedit.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kiconedit/kiconedit.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -82,7 +82,6 @@
 Name[ro]=Editor iconiţe
 Name[sv]=Kiconedit
 Name[ta]=கேசின்னம் திருத்து
-Name[th]=แก้ไขไอคอน - K
 Name[tr]=K Icon Düzenleyici
 Name[ven]=U sengulusa ha aikhono ya K
 Name[zh_TW]=KIconEdit 圖示編輯器
Index: kfile-plugins/dvi/kfile_dvi.desktop
===================================================================
--- kfile-plugins/dvi/kfile_dvi.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/dvi/kfile_dvi.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -45,6 +45,7 @@
 Name[sv]=DVI-information
 Name[ta]=DVI தகவல்
 Name[tg]=Иттилоот оиди DVI
+Name[th]=ข้อมูลแฟ้ม DVI
 Name[tr]=DVI Bilgisi
 Name[uk]=Інформація по DVI
 Name[uz]=DVI ҳақида маълумот
Index: kfile-plugins/png/kfile_png.desktop
===================================================================
--- kfile-plugins/png/kfile_png.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/png/kfile_png.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -47,7 +47,7 @@
 Name[sv]=PNG-information
 Name[ta]=PNG தகவல்
 Name[tg]=Иттилоот оиди PNG
-Name[th]=ข้อมูล PNG
+Name[th]=ข้อมูลแฟ้ม PNG
 Name[tr]=PNG Bilgisi
 Name[uk]=Інформація про PNG
 Name[uz]=PNG ҳақида маълумот
Index: kfile-plugins/jpeg/kfile_jpeg.desktop
===================================================================
--- kfile-plugins/jpeg/kfile_jpeg.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/jpeg/kfile_jpeg.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -46,6 +46,7 @@
 Name[sv]=JPEG EXIF-information
 Name[ta]=JPEG EXIF தகவல்
 Name[tg]=Иттилоот оиди JPEG EXIF
+Name[th]=ข้อมูลแฟ้ม JPEG EXIF
 Name[tr]=JPEG EXIF Bilgisi
 Name[uk]=Інформація про JPEG EXIF
 Name[uz]=JPEG EXIF ҳақида маълумот
Index: kfile-plugins/xbm/kfile_xbm.desktop
===================================================================
--- kfile-plugins/xbm/kfile_xbm.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/xbm/kfile_xbm.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -47,6 +47,7 @@
 Name[sv]=XBM-information
 Name[ta]=XBM தகவல்
 Name[tg]=Иттилоот оиди XBM
+Name[th]=ข้อมูลแฟ้ม XBM
 Name[tr]=XBM Bilgisi
 Name[uk]=Інформація по XBM
 Name[uz]=XBM ҳақида маълумот
Index: kfile-plugins/tiff/kfile_tiff.desktop
===================================================================
--- kfile-plugins/tiff/kfile_tiff.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/tiff/kfile_tiff.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -49,6 +49,7 @@
 Name[sv]=Metainformation om TIFF-fil
 Name[ta]=TIFF மீக்கோப்பு தகவல்
 Name[tg]=Иттилоот оиди метафайли TIFF
+Name[th]=ข้อมูลเมตาแฟ้ม TIFF
 Name[tr]=TIFF Dosya Bilgisi
 Name[uk]=Метаінформація про файл TIFF
 Name[uz]=TIFF-файлининг мета-маълумоти
Index: kfile-plugins/raw/kfile_raw.desktop
===================================================================
--- kfile-plugins/raw/kfile_raw.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/raw/kfile_raw.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -41,6 +41,7 @@
 Name[sr]=RAW фајлови слика
 Name[sr@Latn]=RAW fajlovi slika
 Name[sv]=Obehandlade kamerafiler
+Name[th]=แฟ้มภาพ RAW จากกล้อง
 Name[tr]=RAW Kamera Dosyaları
 Name[uk]=Файли цифрової камери RAW
 Name[zh_CN]=RAW 相机文件
Index: kfile-plugins/pnm/kfile_pnm.desktop
===================================================================
--- kfile-plugins/pnm/kfile_pnm.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/pnm/kfile_pnm.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -44,7 +44,7 @@
 Name[sv]=PNM-information
 Name[ta]=PNM தகவல்
 Name[tg]=Иттилоот оиди PNM
-Name[th]=ข้อมูล PNM
+Name[th]=ข้อมูลแฟ้ม PNM
 Name[tr]=PNM Bilgisi
 Name[uk]=Інформація про PNM
 Name[uz]=PNM ҳақида маълумот
Index: kfile-plugins/pcx/kfile_pcx.desktop
===================================================================
--- kfile-plugins/pcx/kfile_pcx.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/pcx/kfile_pcx.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -46,6 +46,7 @@
 Name[sv]=Metainformation om PCX-fil
 Name[ta]=PCX File Meta தகவல்
 Name[tg]=Иттилоот оиди метафайли PCX
+Name[th]=ข้อมูลเมตาแฟ้ม PCX
 Name[tr]=PCX Dosya Bilgisi
 Name[uk]=Метаінформація про файл PCX
 Name[uz]=PCX-файлининг мета-маълумоти
Index: kfile-plugins/exr/kfile_exr.desktop
===================================================================
--- kfile-plugins/exr/kfile_exr.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/exr/kfile_exr.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -42,6 +42,7 @@
 Name[sv]=EXR-information
 Name[ta]=EXR தகவல்
 Name[tg]=Иттилоот оиди EXR
+Name[th]=ข้อมูลแฟ้ม EXR
 Name[tr]=EXR Bilgisi
 Name[uk]=Інформація по EXR
 Name[uz]=EXR ҳақида маълумот
Index: kfile-plugins/xpm/kfile_xpm.desktop
===================================================================
--- kfile-plugins/xpm/kfile_xpm.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/xpm/kfile_xpm.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -38,6 +38,7 @@
 Name[sr@Latn]=XPM informacije
 Name[sv]=XPM-information
 Name[ta]=XPM தகவல்
+Name[th]=ข้อมูลแฟ้ม XPM
 Name[tr]=XPM Bilgisi
 Name[uk]=Інформація про XPM
 Name[uz]=XPM ҳақида маълумот
Index: kfile-plugins/gif/kfile_gif.desktop
===================================================================
--- kfile-plugins/gif/kfile_gif.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/gif/kfile_gif.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -48,6 +48,7 @@
 Name[sv]=Gif-information
 Name[ta]=GIF தகவல்
 Name[tg]=Иттилоот оиди GIF
+Name[th]=ข้อมูลแฟ้ม GIF
 Name[tr]=GIF Bilgisi
 Name[uk]=Інформація по GIF
 Name[uz]=GIF ҳақида маълумот
Index: kfile-plugins/pdf/kfile_pdf.desktop
===================================================================
--- kfile-plugins/pdf/kfile_pdf.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/pdf/kfile_pdf.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -47,7 +47,7 @@
 Name[sv]=PDF-information
 Name[ta]=PDF தகவல்
 Name[tg]=Иттилоот оиди PDF
-Name[th]=ข้อมูล PDF
+Name[th]=ข้อมูลแฟ้ม PDF
 Name[tr]=PDF Bilgisi
 Name[uk]=Інформація про PDF
 Name[uz]=PDF ҳақида маълумот
Index: kfile-plugins/ico/kfile_ico.desktop
===================================================================
--- kfile-plugins/ico/kfile_ico.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/ico/kfile_ico.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -47,6 +47,7 @@
 Name[sv]=Ico-information
 Name[ta]=ICO தகவல்
 Name[tg]=Иттилоот оиди ICO
+Name[th]=ข้อมูลแฟ้ม ICO
 Name[tr]=ICO Bilgisi
 Name[uk]=Інформація по ICO
 Name[uz]=ICO ҳақида маълумот
Index: kfile-plugins/dds/kfile_dds.desktop
===================================================================
--- kfile-plugins/dds/kfile_dds.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/dds/kfile_dds.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -40,6 +40,7 @@
 Name[sr@Latn]=Informacije o DirectDraw površini
 Name[sv]=Directdraw ytinformation
 Name[ta]=நேரடி மேற்பரப்பு தகவல்
+Name[th]=ข้อมูลแฟ้มพื้นผิว DirectDraw
 Name[tr]=DirectDraw Yüzey Bilgisi
 Name[uk]=Інформація про поверхню DirectDraw
 Name[zh_CN]=DirectDraw 表面信息
Index: kfile-plugins/tga/kfile_tga.desktop
===================================================================
--- kfile-plugins/tga/kfile_tga.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/tga/kfile_tga.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -46,6 +46,7 @@
 Name[sv]=Information om Truevision Targa
 Name[ta]=சரியான பார்வை தார்கா தகவல்
 Name[tg]=Иттилоот оиди Truevision Targa
+Name[th]=ข้อมูลแฟ้ม Truevision Targa
 Name[tr]=Truevision Targa Bilgisi
 Name[uk]=Інформація по Truevision Targa
 Name[uz]=TGA ҳақида маълумот
Index: kfile-plugins/bmp/kfile_bmp.desktop
===================================================================
--- kfile-plugins/bmp/kfile_bmp.desktop	(.../tags/KDE/3.5.7/kdegraphics)	(wersja 699343)
+++ kfile-plugins/bmp/kfile_bmp.desktop	(.../branches/KDE/3.5/kdegraphics)	(wersja 699343)
@@ -48,6 +48,7 @@
 Name[sv]=BMP-information
 Name[ta]=BMP தகவல்
 Name[tg]=Иттилоот оиди BMP
+Name[th]=ข้อมูลแฟ้ม BMP
 Name[tr]=BMP Bilgisi
 Name[uk]=Інформація по BMP
 Name[uz]=BMP ҳақида маълумот

Zmiany atrybutów dla: .
___________________________________________________________________
Nazwa: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


