diff -Nurw kdegraphics-3.5.5/kpdf/core/bookmark.cpp kdegraphics-3.5.5-bw/kpdf/core/bookmark.cpp
--- kdegraphics-3.5.5/kpdf/core/bookmark.cpp	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/kpdf/core/bookmark.cpp	2007-01-15 16:53:07.000000000 -0800
@@ -0,0 +1,106 @@
+/***************************************************************************
+ *   Copyright (C) 2007 by Bryan Wilkerson <bryanwilkerson@yahoo.com>      *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ ***************************************************************************/
+
+
+#include <qobject.h>
+#include <qstring.h>
+#include <qdom.h>
+#include <kdebug.h>
+
+#include "bookmark.h"
+
+#define PAGE_ATTR        "page"
+#define RECT_LEFT_ATTR   "rect_left"
+#define RECT_TOP_ATTR    "rect_top"
+#define RECT_RIGHT_ATTR  "rect_right"
+#define RECT_BOTTOM_ATTR "rect_bottom"
+#define NAME_ATTR        "name"
+#define IS_PAGE_ATTR     "is_page_mark"
+
+
+
+
+KPDFBookmark::KPDFBookmark(int page, QRect rect, const char *name, const char *comment)
+{
+    m_page = page;
+    m_rect = rect;
+    m_name = name;
+    m_comment = comment;
+	m_isPageMark = false;
+
+}
+
+KPDFBookmark::KPDFBookmark(int page, QPoint point, const char *name, const char *comment)
+{
+	m_page = page;
+	m_rect.setX(point.x());
+	m_rect.setY(point.y());
+	m_rect.setHeight(40);
+	m_rect.setWidth(40);
+	m_isPageMark = true;
+	m_name = name;
+	m_comment = comment;
+
+}
+
+KPDFBookmark::KPDFBookmark(QDomElement &e)
+{
+    bool ok;
+
+#if 0
+    kdDebug() << "Parsing Element: " << e.tagName() << endl;
+    kdDebug() << "   Attributes: " << endl;
+
+    QDomNamedNodeMap attribs = e.attributes();
+    for( unsigned int i = 0; i < attribs.count(); i++ )
+    {
+        QDomNode node = attribs.item(i);
+        if( node.isAttr() )
+            kdDebug() << node.nodeName() << " = " << node.nodeValue() << endl;
+    }
+#endif 
+
+    if ( e.hasAttribute(PAGE_ATTR) )
+        m_page = e.attribute(PAGE_ATTR).toInt(&ok);
+    if ( e.hasAttribute(RECT_LEFT_ATTR) )
+        m_rect.setLeft(e.attribute(RECT_LEFT_ATTR).toInt(&ok));
+    if ( e.hasAttribute(RECT_TOP_ATTR) )
+        m_rect.setTop(e.attribute(RECT_TOP_ATTR).toInt(&ok));
+    if ( e.hasAttribute( RECT_RIGHT_ATTR ) )
+        m_rect.setRight(e.attribute(RECT_RIGHT_ATTR).toInt(&ok));
+    if ( e.hasAttribute( RECT_BOTTOM_ATTR ) )
+        m_rect.setBottom(e.attribute(RECT_BOTTOM_ATTR).toInt(&ok));
+    if ( e.hasAttribute( NAME_ATTR ) )
+        m_name = e.attribute(NAME_ATTR);
+	if ( e.hasAttribute( IS_PAGE_ATTR ) )
+		m_isPageMark = (bool)e.attribute(IS_PAGE_ATTR).toInt(&ok);
+	m_comment = e.text();
+
+}
+
+KPDFBookmark::~KPDFBookmark()
+{
+
+
+}
+
+QDomElement KPDFBookmark::toXML(QDomDocument &doc)
+{
+    QDomElement e = doc.createElement( "bookmark" );
+    e.setAttribute(PAGE_ATTR        ,  m_page);
+    e.setAttribute(RECT_LEFT_ATTR   ,  m_rect.left());
+    e.setAttribute(RECT_TOP_ATTR    ,  m_rect.top());
+    e.setAttribute(RECT_RIGHT_ATTR  ,  m_rect.right());
+    e.setAttribute(RECT_BOTTOM_ATTR ,  m_rect.bottom());
+    e.setAttribute(NAME_ATTR        ,  m_name);
+	e.setAttribute(IS_PAGE_ATTR     ,  m_isPageMark);
+    e.appendChild( doc.createTextNode(m_comment) );
+
+    return e;
+}
diff -Nurw kdegraphics-3.5.5/kpdf/core/bookmark.h kdegraphics-3.5.5-bw/kpdf/core/bookmark.h
--- kdegraphics-3.5.5/kpdf/core/bookmark.h	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/kpdf/core/bookmark.h	2007-01-15 15:45:49.000000000 -0800
@@ -0,0 +1,49 @@
+/***************************************************************************
+ *   Copyright (C) 2007 by Bryan Wilkerson <bryanwilkerson@yahoo.com       *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ ***************************************************************************/
+
+#ifndef _KPDF_BOOKMARK_H_
+#define _KPDF_BOOKMARK_H_
+
+#include <qobject.h>
+#include <qstring.h>
+#include <qdom.h>
+
+
+class KPDFBookmark : public QObject
+{
+	Q_OBJECT
+	public:
+		KPDFBookmark(int page, QRect rect, const char *name, const char *comment);
+		KPDFBookmark(QDomElement &e);
+		// this one creates a page level bookmark
+		KPDFBookmark(int page, QPoint pagepoint, const char *name, const char *comment);
+		~KPDFBookmark();
+		
+		int Page() const { return m_page; };
+		const QRect &Rect() const { return m_rect; };
+		const QString &Name()  const { return m_name; };
+		void Name(QString name) { m_name = name; };
+		const QString &Comment()  const { return m_comment; };
+		void Comment(QString comment) {m_comment = comment;};
+		QDomElement toXML(QDomDocument &doc);
+		const bool isPageMark() { return m_isPageMark; }
+
+
+	private:
+		int m_page;
+		bool m_isPageMark;
+		QRect m_rect;
+		QString m_name;
+		QString m_comment;
+	
+
+};
+
+
+#endif // _KPDF_BOOKMARK_H_
diff -Nurw kdegraphics-3.5.5/kpdf/core/document.cpp kdegraphics-3.5.5-bw/kpdf/core/document.cpp
--- kdegraphics-3.5.5/kpdf/core/document.cpp	2006-10-01 10:25:14.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/core/document.cpp	2007-01-15 15:39:50.000000000 -0800
@@ -80,7 +80,9 @@
     int page;
     int memory;
     // public constructor: initialize data
-    AllocatedPixmap( int i, int p, int m ) : id( i ), page( p ), memory( m ) {};
+	AllocatedPixmap( int i, int p, int m ) : id( i ), page( p ), memory( m )
+	{}
+	;
 };
 
 struct RunningSearch
@@ -427,7 +429,8 @@
 
 void KPDFDocument::putFontInfo(KListView *list)
 {
-    if (generator) generator->putFontInfo(list);
+	if (generator)
+		generator->putFontInfo(list);
 }
 
 void KPDFDocument::requestPixmaps( const QValueList< PixmapRequest * > & requests )
@@ -792,8 +795,7 @@
     }
     // 3. PREVMATCH //TODO
     else if ( type == PrevMatch )
-    {
-    }
+	{}
     // 4. GOOGLE* - process all document marking pages
     else if ( type == GoogleAll || type == GoogleAny )
     {
@@ -921,16 +923,44 @@
 }
 
 
-void KPDFDocument::toggleBookmark( int n )
+KPDFBookmark *KPDFDocument::getBookmark( int n, const QPoint &point )
+{
+	KPDFPage * page = ( n < (int)pages_vector.count() ) ? pages_vector[ n ] : 0;
+	if ( page )
+		return page->getBookmark(point);
+	return NULL;	
+}
+
+void KPDFDocument::addBookmark( int n, KPDFBookmark *bm )
+{
+	KPDFPage * page = ( n < (int)pages_vector.count() ) ? pages_vector[ n ] : 0;
+	if ( page )
+	{
+		page->setBookmark(bm);
+		foreachObserver( notifyPageChanged( n, DocumentObserver::Bookmark ) );
+	}
+}
+void KPDFDocument::removeBookmark( int n, const QPoint &point )
+{
+	KPDFPage * page = ( n < (int)pages_vector.count() ) ? pages_vector[ n ] : 0;
+	if ( page )
+	{
+		page->removeBookmark(point);
+		foreachObserver( notifyPageChanged( n, DocumentObserver::Bookmark ) );
+	}
+}
+void KPDFDocument::removeBookmarks( int n )
 {
     KPDFPage * page = ( n < (int)pages_vector.count() ) ? pages_vector[ n ] : 0;
     if ( page )
     {
-        page->setBookmark( !page->hasBookmark() );
+		page->removeAllBookmarks();
         foreachObserver( notifyPageChanged( n, DocumentObserver::Bookmark ) );
     }
 }
 
+
+
 void KPDFDocument::processLink( const KPDFLink * link )
 {
     if ( !link )
@@ -938,7 +968,8 @@
 
     switch( link->linkType() )
     {
-        case KPDFLink::Goto: {
+		case KPDFLink::Goto:
+			{
             const KPDFLinkGoto * go = static_cast< const KPDFLinkGoto * >( link );
             d->nextDocumentViewport = go->destViewport();
 
@@ -958,14 +989,17 @@
             }
             else
             {
-                if (d->nextDocumentViewport.pageNumber == -1) return;
+					if (d->nextDocumentViewport.pageNumber == -1)
+						return;
                 setViewport( d->nextDocumentViewport, -1, true );
                 d->nextDocumentViewport = DocumentViewport();
             }
 
-            } break;
+			}
+			break;
 
-        case KPDFLink::Execute: {
+		case KPDFLink::Execute:
+			{
             const KPDFLinkExecute * exe  = static_cast< const KPDFLinkExecute * >( link );
             QString fileName = exe->fileName();
             if ( fileName.endsWith( ".pdf" ) || fileName.endsWith( ".PDF" ) )
@@ -1012,9 +1046,11 @@
             }
             else
                 KMessageBox::information( 0, i18n( "No application found for opening file of mimetype %1." ).arg( mime->name() ) );
-            } break;
+			}
+			break;
 
-        case KPDFLink::Action: {
+		case KPDFLink::Action:
+			{
             const KPDFLinkAction * action = static_cast< const KPDFLinkAction * >( link );
             switch( action->actionType() )
             {
@@ -1057,9 +1093,11 @@
                     emit close();
                     break;
             }
-            } break;
+			}
+			break;
 
-        case KPDFLink::Browse: {
+		case KPDFLink::Browse:
+			{
             const KPDFLinkBrowse * browse = static_cast< const KPDFLinkBrowse * >( link );
             // if the url is a mailto one, invoke mailer
             if ( browse->url().startsWith( "mailto:", false ) )
@@ -1078,7 +1116,8 @@
                 // Albert: this is not a leak!
                 new KRun(url);
             }
-            } break;
+			}
+			break;
 
         case KPDFLink::Movie:
             //const KPDFLinkMovie * browse = static_cast< const KPDFLinkMovie * >( link );
@@ -1202,7 +1241,8 @@
                 pages_vector[ p->page ]->deletePixmap( p->id );
                 // delete allocation descriptor
                 delete p;
-            } else
+			}
+			else
                 ++pIt;
         }
         //p--rintf("freeMemory A:[%d -%d = %d] \n", d->allocatedPixmapsFifo.count() + pagesFreed, pagesFreed, d->allocatedPixmapsFifo.count() );
@@ -1270,7 +1310,7 @@
 // note: load data and stores it internally (document or pages). observers
 // are still uninitialized at this point so don't access them
 {
-    //kdDebug() << "Using '" << d->xmlFileName << "' as document info file." << endl;
+	kdDebug() << "Using '" << d->xmlFileName << "' as document info file." << endl;
     QFile infoFile( d->xmlFileName );
     if ( !infoFile.exists() || !infoFile.open( IO_ReadOnly ) )
         return;
@@ -1305,11 +1345,20 @@
             while ( n.isElement() )
             {
                 e = n.toElement();
+				// still supports old style bookmarks
                 if (e.tagName() == "page")
                 {
                     pageNumber = e.text().toInt(&ok);
                     if ( ok && pageNumber >= 0 && pageNumber < (int)pages_vector.count() )
-                        pages_vector[ pageNumber ]->setBookmark( true );
+						pages_vector[ pageNumber ]->setBookmark( new KPDFBookmark(pageNumber, QPoint(20, 20), NULL, NULL) );
+				}
+				else if (e.tagName() == "bookmark")
+				{
+					// This gets deleted by the page as it destroys itself
+					KPDFBookmark *bm = new KPDFBookmark(e);
+					pageNumber = bm->Page();
+					if ( pageNumber >= 0 && pageNumber < (int)pages_vector.count() )
+						pages_vector[ pageNumber ]->setBookmark( bm );
                 }
                 n = n.nextSibling();
             }
@@ -1397,15 +1446,7 @@
         root.appendChild( bookmarkList );
 
         for ( uint i = 0; i < pages_vector.count() ; i++ )
-        {
-            if ( pages_vector[i]->hasBookmark() )
-            {
-                QDomElement page = doc.createElement( "page" );
-                page.appendChild( doc.createTextNode( QString::number(i) ) );
-
-                bookmarkList.appendChild( page );
-            }
-        }
+			pages_vector[i]->addBookmarksToXML(doc, bookmarkList);
 
         // Add general info to DOM
         QDomElement generalInfo = doc.createElement( "generalInfo" );
@@ -1510,8 +1551,10 @@
             rePos.enabled = true;
             rePos.normalizedX = token.section( ':', 1, 1 ).toDouble();
             rePos.normalizedY = token.section( ':', 2, 2 ).toDouble();
-            if (token.section( ':', 3, 3 ).toInt() == 1) rePos.pos = Center;
-            else rePos.pos = TopLeft;
+			if (token.section( ':', 3, 3 ).toInt() == 1)
+				rePos.pos = Center;
+			else
+				rePos.pos = TopLeft;
         }
         else if ( token.startsWith( "AF1" ) )
         {
@@ -1569,7 +1612,8 @@
     appendChild( docElement );
 }
 
-void DocumentInfo::set( const QString &key, const QString &value,
+void DocumentInfo::set
+	( const QString &key, const QString &value,
                         const QString &title )
 {
     QDomElement docElement = documentElement();
@@ -1589,7 +1633,8 @@
         docElement.appendChild( element );
 }
 
-QString DocumentInfo::get( const QString &key ) const
+QString DocumentInfo::get
+	( const QString &key ) const
 {
     QDomElement docElement = documentElement();
     QDomElement element;
@@ -1612,3 +1657,4 @@
 }
 
 #include "document.moc"
+
diff -Nurw kdegraphics-3.5.5/kpdf/core/document.h kdegraphics-3.5.5-bw/kpdf/core/document.h
--- kdegraphics-3.5.5/kpdf/core/document.h	2006-10-01 10:25:14.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/core/document.h	2007-01-15 10:21:44.000000000 -0800
@@ -20,6 +20,7 @@
 
 class KPDFPage;
 class KPDFLink;
+class KPDFBookmark;
 class DocumentObserver;
 class DocumentViewport;
 class DocumentInfo;
@@ -98,7 +99,10 @@
         bool continueSearch( int searchID );
         void resetSearch( int searchID );
         bool continueLastSearch();
-        void toggleBookmark( int page );
+	void addBookmark( int page, KPDFBookmark * bm );
+	void removeBookmark( int n, const QPoint &point );
+	void removeBookmarks( int page );
+	KPDFBookmark *getBookmark( int page, const QPoint &point );
         void processLink( const KPDFLink * link );
         bool print( KPrinter &printer );
 
@@ -153,19 +157,23 @@
         enum Position { Center = 1, TopLeft = 2};
 
         // if reCenter.enabled, this contains the viewport center
-        struct {
+	struct
+	{
             bool enabled;
             double normalizedX;
             double normalizedY;
             Position pos;
-        } rePos;
+	}
+	rePos;
 
         // if autoFit.enabled, page must be autofitted in the viewport
-        struct {
+	struct
+	{
             bool enabled;
             bool width;
             bool height;
-        } autoFit;
+	}
+	autoFit;
 
         /** class methods **/
         // allowed constructors, don't use others
@@ -190,14 +198,16 @@
          * Sets a value for a special key. The title should be an i18n'ed
          * string, since it's used in the document information dialog.
          */
-        void set( const QString &key, const QString &value,
+	void set
+		( const QString &key, const QString &value,
                   const QString &title = QString() );
 
         /**
          * Returns the value for a given key or an empty string when the
          * key doesn't exist.
          */
-        QString get( const QString &key ) const;
+	QString get
+		( const QString &key ) const;
 };
 
 /**
diff -Nurw kdegraphics-3.5.5/kpdf/core/page.cpp kdegraphics-3.5.5-bw/kpdf/core/page.cpp
--- kdegraphics-3.5.5/kpdf/core/page.cpp	2005-09-10 01:18:42.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/core/page.cpp	2007-01-15 10:22:53.000000000 -0800
@@ -25,7 +25,7 @@
 
 KPDFPage::KPDFPage( uint page, float w, float h, int r )
     : m_number( page ), m_rotation( r ), m_width( w ), m_height( h ),
-    m_bookmarked( false ), m_text( 0 ), m_transition( 0 )
+    m_text( 0 ), m_transition( 0 )
 {
     // if landscape swap width <-> height (rotate 90deg CCW)
     if ( r == 90 || r == 270 )
@@ -44,6 +44,7 @@
 {
     deletePixmapsAndRects();
     deleteHighlights();
+    removeAllBookmarks();
     delete m_text;
     delete m_transition;
 }
@@ -64,9 +65,26 @@
     return m_text != 0;
 }
 
-bool KPDFPage::hasBookmark() const
+bool KPDFPage::hasBookmarks() const
 {
-    return m_bookmarked;
+    return !m_bookmarks.isEmpty();
+}
+
+bool KPDFPage::hasBookmark(const QPoint &point) const 
+{
+    QValueList< KPDFBookmark * >::const_iterator it = m_bookmarks.begin(), end = m_bookmarks.end();
+    while ( it != end )
+    {
+        KPDFBookmark* bm = *it;
+        QRect rect = bm->Rect();
+        int x = point.x(), y = point.y();
+        if( x > rect.left() && x < rect.right() && y > rect.top() && y < rect.bottom() )
+        {
+            return true;
+        }
+        ++it;
+    }
+    return false;
 }
 
 bool KPDFPage::hasObjectRect( double x, double y ) const
@@ -191,9 +209,10 @@
     m_text = tp;
 }
 
-void KPDFPage::setBookmark( bool state )
+void KPDFPage::setBookmark( KPDFBookmark *bm )
 {
-    m_bookmarked = state;
+    // Allow overlapping rects
+    m_bookmarks.append( bm );
 }
 
 void KPDFPage::setObjectRects( const QValueList< ObjectRect * > rects )
@@ -267,6 +286,62 @@
     }
 }
 
+void KPDFPage::addBookmarksToXML(QDomDocument &doc, QDomElement &domElement)
+{
+    QValueList< KPDFBookmark * >::iterator it = m_bookmarks.begin(), end = m_bookmarks.end();
+    while ( it != end )
+    {
+        KPDFBookmark* bm = *it;
+        domElement.appendChild(bm->toXML(doc));
+        ++it;
+    }
+}
+
+KPDFBookmark *KPDFPage::getBookmark(const QPoint &point)
+{
+	QValueList< KPDFBookmark * >::iterator it = m_bookmarks.begin(), end = m_bookmarks.end();
+	while ( it != end )
+	{
+		KPDFBookmark* bm = *it;
+		QRect rect = bm->Rect();
+		int x = point.x(), y = point.y();
+		if( x > rect.left() && x < rect.right() && y > rect.top() && y < rect.bottom() )
+		{
+			return bm;
+		}
+		++it;
+	}
+	return NULL;
+}
+
+void KPDFPage::removeBookmark(const QPoint &point)
+{
+    QValueList< KPDFBookmark * >::iterator it = m_bookmarks.begin(), end = m_bookmarks.end();
+    while ( it != end )
+    {
+        KPDFBookmark* bm = *it;
+        QRect rect = bm->Rect();
+        int x = point.x(), y = point.y();
+        if( x > rect.left() && x < rect.right() && y > rect.top() && y < rect.bottom() )
+        {
+            it = m_bookmarks.remove(it);
+            break;
+        }
+        ++it;
+    }
+}
+
+void KPDFPage::removeAllBookmarks()
+{
+    QValueList< KPDFBookmark * >::iterator it = m_bookmarks.begin(), end = m_bookmarks.end();
+    while ( it != end )
+    {
+        KPDFBookmark* bm = *it;
+        it = m_bookmarks.remove(it);
+        delete bm;
+    }
+}
+
 
 /** class NormalizedRect **/
 
diff -Nurw kdegraphics-3.5.5/kpdf/core/page.h kdegraphics-3.5.5-bw/kpdf/core/page.h
--- kdegraphics-3.5.5/kpdf/core/page.h	2005-09-10 01:18:42.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/core/page.h	2007-01-15 10:22:22.000000000 -0800
@@ -12,6 +12,9 @@
 
 #include <qmap.h>
 #include <qvaluelist.h>
+#include <qdom.h>
+
+#include "bookmark.h"
 
 class QPixmap;
 class QRect;
@@ -62,8 +65,14 @@
         ~ObjectRect();
 
         // query type and get a const pointer to the stored object
-        inline ObjectType objectType() const { return m_objectType; }
-        inline const void * pointer() const { return m_pointer; }
+	inline ObjectType objectType() const
+	{
+		return m_objectType;
+	}
+	inline const void * pointer() const
+	{
+		return m_pointer;
+	}
 
     private:
         ObjectType m_objectType;
@@ -89,14 +98,30 @@
         ~KPDFPage();
 
         // query properties (const read-only methods)
-        inline int number() const { return m_number; }
-        inline int rotation() const { return m_rotation; }
-        inline float width() const { return m_width; }
-        inline float height() const { return m_height; }
-        inline float ratio() const { return m_height / m_width; }
+	inline int number() const
+	{
+		return m_number;
+	}
+	inline int rotation() const
+	{
+		return m_rotation;
+	}
+	inline float width() const
+	{
+		return m_width;
+	}
+	inline float height() const
+	{
+		return m_height;
+	}
+	inline float ratio() const
+	{
+		return m_height / m_width;
+	}
         bool hasPixmap( int p_id, int width = -1, int height = -1 ) const;
         bool hasSearchPage() const;
-        bool hasBookmark() const;
+	bool hasBookmarks() const;
+	bool hasBookmark(const QPoint &point) const;
         bool hasObjectRect( double x, double y ) const;
         bool hasHighlights( int s_id = -1 ) const;
         //bool hasAnnotation( double x, double y ) const;
@@ -111,7 +136,7 @@
         // operations: set/delete contents (by KPDFDocument)
         void setPixmap( int p_id, QPixmap * pixmap );
         void setSearchPage( TextPage * text );
-        void setBookmark( bool state );
+	void setBookmark( KPDFBookmark *bm );
         void setObjectRects( const QValueList< ObjectRect * > rects );
         void setHighlight( int s_id, NormalizedRect * &r, const QColor & color );
         //void setAnnotation( Annotation * annotation );
@@ -120,16 +145,22 @@
         void deletePixmapsAndRects();
         void deleteHighlights( int s_id = -1 );
 
+	void addBookmarksToXML(QDomDocument &doc, QDomElement &domElement);
+	KPDFBookmark *getBookmark(const QPoint &point);
+	void removeBookmark(const QPoint &point);
+	void removeAllBookmarks();
+
+
     private:
         friend class PagePainter;
         int m_number, m_rotation;
         float m_width, m_height;
-        bool m_bookmarked;
 
         QMap< int, QPixmap * > m_pixmaps;
         TextPage * m_text;
         QValueList< ObjectRect * > m_rects;
         QValueList< HighlightRect * > m_highlights;
+	QValueList< KPDFBookmark *> m_bookmarks;
         //QValueList< Annotation * > m_annotations;
         KPDFPageTransition * m_transition;
 };
@@ -147,3 +178,4 @@
 };
 
 #endif
+
diff -Nurw kdegraphics-3.5.5/kpdf/part.cpp kdegraphics-3.5.5-bw/kpdf/part.cpp
--- kdegraphics-3.5.5/kpdf/part.cpp	2006-10-01 10:25:16.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/part.cpp	2007-01-15 15:34:45.000000000 -0800
@@ -161,7 +161,7 @@
 	m_thumbnailList = new ThumbnailList( thumbsBox, m_document );
 //	ThumbnailController * m_tc = new ThumbnailController( thumbsBox, m_thumbnailList );
 	connect( m_thumbnailList, SIGNAL( urlDropped( const KURL& ) ), SLOT( openURLFromDocument( const KURL & )) );
-	connect( m_thumbnailList, SIGNAL( rightClick(const KPDFPage *, const QPoint &) ), this, SLOT( slotShowMenu(const KPDFPage *, const QPoint &) ) );
+	connect( m_thumbnailList, SIGNAL( rightClick(const KPDFPage *, const QPoint &, const QPoint &) ), this, SLOT( slotShowMenu(const KPDFPage *, const QPoint &, const QPoint &) ) );
 	// shrink the bottom controller toolbar (too hackish..)
 	thumbsBox->setStretchFactor( m_searchWidget, 100 );
 	thumbsBox->setStretchFactor( m_thumbnailList, 100 );
@@ -193,7 +193,7 @@
 	m_pageView = new PageView( m_splitter, m_document );
 	m_pageView->setFocus(); //usability setting
 	connect( m_pageView, SIGNAL( urlDropped( const KURL& ) ), SLOT( openURLFromDocument( const KURL & )));
-	connect( m_pageView, SIGNAL( rightClick(const KPDFPage *, const QPoint &) ), this, SLOT( slotShowMenu(const KPDFPage *, const QPoint &) ) );
+	connect( m_pageView, SIGNAL( rightClick(const KPDFPage *, const QPoint &, const QPoint &) ), this, SLOT( slotShowMenu(const KPDFPage *, const QPoint &, const QPoint &) ) );
 
 	// add document observers
 	m_document->addObserver( this );
@@ -809,7 +809,7 @@
     doPrint(printer);
 }
 
-void Part::slotShowMenu(const KPDFPage *page, const QPoint &point)
+void Part::slotShowMenu(const KPDFPage *page, const QPoint &point, const QPoint &pagePoint)
 {
 	bool reallyShow = false;
 	if (!m_actionsSearched)
@@ -841,17 +841,22 @@
 		m_actionsSearched = true;
 	}
 	
-	
 	KPopupMenu *popup = new KPopupMenu( widget(), "rmb popup" );
+
 	if (page)
 	{
 		popup->insertTitle( i18n( "Page %1" ).arg( page->number() + 1 ) );
-        if ( page->hasBookmark() )
-			popup->insertItem( SmallIcon("bookmark"), i18n("Remove Bookmark"), 1 );
-		else
 			popup->insertItem( SmallIcon("bookmark_add"), i18n("Add Bookmark"), 1 );
+	popup->insertItem( SmallIcon("bookmark_add"), i18n("Edit Selected Bookmark"), 2 );
+	popup->insertItem( SmallIcon("bookmark"), i18n("Remove Selected Bookmark"), 3 );
+	popup->insertItem( SmallIcon("bookmark"), i18n("Remove All Bookmarks from This Page"), 4 );
+
+	popup->setItemEnabled(2, page->hasBookmark(pagePoint));
+	popup->setItemEnabled(3, page->hasBookmark(pagePoint));
+	popup->setItemEnabled(4, page->hasBookmarks());
 		if ( m_pageView->canFitPageWidth() )
-			popup->insertItem( SmallIcon("viewmagfit"), i18n("Fit Width"), 2 );
+		popup->insertItem( SmallIcon("viewmagfit"), i18n("Fit Width"), 5);
+
 		//popup->insertItem( SmallIcon("pencil"), i18n("Edit"), 3 );
 		//popup->setItemEnabled( 3, false );
 		reallyShow = true;
@@ -878,15 +883,44 @@
 		switch ( popup->exec(point) )
 		{
 			case 1:
-				m_document->toggleBookmark( page->number() );
+			{
+				BookmarkDialogImpl bmdlg;
+				bmdlg.exec();
+				if( !bmdlg.wasCanceled() )
+					m_document->addBookmark( page->number(), 
+											 new KPDFBookmark(page->number(), 
+													 pagePoint,
+													 bmdlg.getName().ascii(),
+													 bmdlg.getComment().ascii()));
 				break;
+			}
 			case 2:
+			{
+				KPDFBookmark *bm = m_document->getBookmark(page->number(), pagePoint);
+				if( bm )
+				{
+					BookmarkDialogImpl bmdlg(bm);
+					bmdlg.exec();
+					//BookmarkDialogImpl will update *bm if save is pressed
+					// or leave it alone if canceled
+				}
+				break;
+			}
+			case 3:
+				m_document->removeBookmark(page->number(), pagePoint);
+				break;
+			case 4:
+				m_document->removeBookmarks(page->number());
+				break;
+			case 5:
 				m_pageView->fitPageWidth( page->number() );
 				break;
 	//		case 3: // switch to edit mode
 	//			break;
 		}
 	}
+
+	
 	delete popup;
 }
 
diff -Nurw kdegraphics-3.5.5/kpdf/part.h kdegraphics-3.5.5-bw/kpdf/part.h
--- kdegraphics-3.5.5/kpdf/part.h	2006-10-01 10:25:16.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/part.h	2007-01-15 08:15:30.000000000 -0800
@@ -21,6 +21,7 @@
 #include <qguardedptr.h>
 #include "core/document.h"
 #include "core/observer.h"
+#include "ui/bookmarkdialogimpl.h"
 #include "dcop.h"
 
 class QWidget;
@@ -107,7 +108,7 @@
 	void slotPreferences();
 	void slotNewConfig();
 	void slotPrintPreview();
-	void slotShowMenu(const KPDFPage *page, const QPoint &point);
+	void slotShowMenu(const KPDFPage *page, const QPoint &point, const QPoint &widgetRelativePoint);
 	void slotShowProperties();
 	void slotShowLeftPanel();
 	void slotShowPresentation();
diff -Nurw kdegraphics-3.5.5/kpdf/ui/bookmarkballoon.cpp kdegraphics-3.5.5-bw/kpdf/ui/bookmarkballoon.cpp
--- kdegraphics-3.5.5/kpdf/ui/bookmarkballoon.cpp	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/kpdf/ui/bookmarkballoon.cpp	2007-01-15 11:59:15.000000000 -0800
@@ -0,0 +1,35 @@
+//
+// C++ Implementation: $MODULE$
+//
+// Description: 
+//
+//
+// Author: Albert Astals Cid <tsdgeos@yahoo.es>, (C) 2007
+//
+// Copyright: See COPYING file that comes with this distribution
+//
+//
+#include <qlabel.h>
+#include <qtextedit.h>
+
+#include "bookmarkballoon.h"
+
+BookmarkBalloon::BookmarkBalloon(QWidget *parent, const char *name)
+	:BookmarkBalloon_ui(parent, name, WStyle_Customize|Qt::WStyle_NoBorder)
+{
+	this->setWindowOpacity(0.5);
+	
+}
+
+void BookmarkBalloon::showBalloon(const KPDFBookmark *bm)
+{
+	if( !bm )
+		return;
+		
+	nameLabel->setText(bm->Name());
+	commentEdit->setText(bm->Comment());
+	
+	this->show();
+}
+
+#include "bookmarkballoon.moc"
diff -Nurw kdegraphics-3.5.5/kpdf/ui/bookmarkballoon.h kdegraphics-3.5.5-bw/kpdf/ui/bookmarkballoon.h
--- kdegraphics-3.5.5/kpdf/ui/bookmarkballoon.h	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/kpdf/ui/bookmarkballoon.h	2007-01-15 11:29:09.000000000 -0800
@@ -0,0 +1,32 @@
+//
+// C++ Interface: $MODULE$
+//
+// Description: 
+//
+//
+// Author: Albert Astals Cid <tsdgeos@yahoo.es>, (C) 2007
+//
+// Copyright: See COPYING file that comes with this distribution
+//
+//
+#ifndef BOOKMARKBALLOON_H
+#define BOOKMARKBALLOON_H
+
+#include "bookmarkballoon_ui.h"
+#include "../core/bookmark.h"
+
+class BookmarkBalloon: public BookmarkBalloon_ui {
+Q_OBJECT
+public:
+    BookmarkBalloon(QWidget *parent = 0, const char *name = 0);
+	
+	
+public:
+	void showBalloon(const KPDFBookmark *bm);
+	
+		
+private:
+
+};
+
+#endif
diff -Nurw kdegraphics-3.5.5/kpdf/ui/bookmarkdialogimpl.cpp kdegraphics-3.5.5-bw/kpdf/ui/bookmarkdialogimpl.cpp
--- kdegraphics-3.5.5/kpdf/ui/bookmarkdialogimpl.cpp	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/kpdf/ui/bookmarkdialogimpl.cpp	2007-01-15 12:26:35.000000000 -0800
@@ -0,0 +1,69 @@
+//
+// C++ Implementation: $MODULE$
+//
+// Description: 
+//
+//
+// Author: Albert Astals Cid <tsdgeos@yahoo.es>, (C) 2007
+//
+// Copyright: See COPYING file that comes with this distribution
+//
+//
+#include <qdialog.h>
+#include <qlineedit.h>
+#include <qtextedit.h>
+
+#include "bookmarkdialogimpl.h"
+
+BookmarkDialogImpl::BookmarkDialogImpl(QWidget *parent, const char *name)
+    :BookmarkDialog_ui(parent, name)
+{
+	m_bm = NULL;
+}
+
+BookmarkDialogImpl::BookmarkDialogImpl(KPDFBookmark *bm, QWidget *parent, const char *name)
+    :BookmarkDialog_ui(parent, name)
+{
+	m_bm = bm;
+	if( bm )
+	{
+		name_edit->setText(bm->Name());
+		comment_edit->setText(bm->Comment());
+	}
+}
+
+void BookmarkDialogImpl::save_button_clicked()
+{
+	if( m_bm )	
+	{
+		m_bm->Name(getName());
+		m_bm->Comment(getComment());
+	}
+	m_canceled = false;
+	this->hide();
+}
+
+void BookmarkDialogImpl::cancel_button_clicked()
+{
+	m_canceled = true;
+	this->hide();
+}
+
+QString BookmarkDialogImpl::getName()
+{
+	if( name_edit )
+		return name_edit->text();
+	else
+		return QString("");
+}
+
+QString BookmarkDialogImpl::getComment()
+{
+	if( comment_edit )
+		return comment_edit->text();
+	else
+		return QString("");
+
+}
+
+#include "bookmarkdialogimpl.moc"
diff -Nurw kdegraphics-3.5.5/kpdf/ui/bookmarkdialogimpl.h kdegraphics-3.5.5-bw/kpdf/ui/bookmarkdialogimpl.h
--- kdegraphics-3.5.5/kpdf/ui/bookmarkdialogimpl.h	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/kpdf/ui/bookmarkdialogimpl.h	2007-01-15 09:57:35.000000000 -0800
@@ -0,0 +1,42 @@
+//
+// C++ Interface: $MODULE$
+//
+// Description: 
+//
+//
+// Author: Albert Astals Cid <tsdgeos@yahoo.es>, (C) 2007
+//
+// Copyright: See COPYING file that comes with this distribution
+//
+//
+#ifndef BOOKMARKDIALOGIMPL_H
+#define BOOKMARKDIALOGIMPL_H
+
+#include <qstring.h>
+
+#include "bookmarkdialog.h"
+#include "../core/bookmark.h"
+
+class BookmarkDialogImpl: public BookmarkDialog_ui {
+Q_OBJECT
+public:
+    BookmarkDialogImpl(QWidget *parent = 0, const char *name = 0);
+    BookmarkDialogImpl(KPDFBookmark *bm, QWidget *parent = 0, const char *name = 0);
+
+public slots:
+    virtual void cancel_button_clicked();
+public slots:
+    virtual void save_button_clicked();
+    
+
+public:
+    bool wasCanceled() { return m_canceled; }
+    QString getName();
+    QString getComment();
+
+private:
+    bool m_canceled;
+	KPDFBookmark *m_bm;
+};
+
+#endif
diff -Nurw kdegraphics-3.5.5/kpdf/ui/pagepainter.cpp kdegraphics-3.5.5-bw/kpdf/ui/pagepainter.cpp
--- kdegraphics-3.5.5/kpdf/ui/pagepainter.cpp	2005-09-10 01:18:43.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/pagepainter.cpp	2007-01-15 16:27:10.000000000 -0800
@@ -10,16 +10,20 @@
 // qt / kde includes
 #include <qrect.h>
 #include <qpainter.h>
+#include <qpaintdevice.h>
 #include <qpixmap.h>
 #include <qimage.h>
 #include <qapplication.h>
 #include <kimageeffect.h>
+#include <kiconloader.h>
 
 // local includes
 #include "pagepainter.h"
 #include "core/page.h"
 #include "conf/settings.h"
 
+#include <math.h>
+
 void PagePainter::paintPageOnPainter( const KPDFPage * page, int id, int flags,
     QPainter * destPainter, const QRect & limits, int width, int height )
 {
@@ -69,6 +73,7 @@
     bool paintHighlights = (flags & Highlights) && !page->m_highlights.isEmpty();
     bool enhanceLinks = (flags & EnhanceLinks) && KpdfSettings::highlightLinks();
     bool enhanceImages = (flags & EnhanceImages) && KpdfSettings::highlightImages();
+	bool paintBookmarks = page->hasBookmarks() && (KpdfSettings::renderMode() != KpdfSettings::EnumRenderMode::Paper);
     // check if there are really some highlightRects to paint
     if ( paintHighlights )
     {
@@ -91,7 +96,7 @@
     }
 
     // use backBuffer if 'pixmap direct manipulation' is needed
-    bool backBuffer = paintAccessibility || paintHighlights;
+    bool backBuffer = paintAccessibility || paintHighlights || paintBookmarks;
     QPixmap * backPixmap = 0;
     QPainter * p = destPainter;
     if ( backBuffer )
@@ -194,6 +199,12 @@
                 }
             }
         }
+        // 2.3   bookmark rects in page
+        if ( paintBookmarks )
+			paintBookmarksOnPage(page, p, limits, backImage);
+        
+
+
         backPixmap->convertFromImage( backImage );
     }
 
@@ -237,3 +248,63 @@
         delete backPixmap;
     }
 }
+
+/* static */
+void PagePainter::paintBookmarksOnPage(const KPDFPage * page, QPainter * p, const QRect & limits, QImage &backImage)
+{
+	const int paperClipWidth = 40;
+	const int paperClipHeight = 40;
+	
+	QImage paperClipPix;
+			
+	// draw highlights that are inside the 'limits' paint region
+	QValueList< KPDFBookmark * >::const_iterator hIt = page->m_bookmarks.begin(), hEnd = page->m_bookmarks.end();
+	for ( ; hIt != hEnd; ++hIt )
+	{
+		KPDFBookmark * bm = *hIt;
+		QRect highlightRect = bm->Rect();
+		if( bm->isPageMark() && highlightRect.intersects( limits ) )
+		{
+			paperClipPix = DesktopIcon( "attach", paperClipWidth);
+			
+			// these are page relative coordinates
+			highlightRect.setHeight(paperClipHeight);
+			highlightRect.setWidth(paperClipWidth);
+			QRect absRect = highlightRect.intersect( limits );
+			absRect.moveBy( -limits.left(), -limits.top() );
+			
+			int ox = limits.left() - highlightRect.left() >? 0;
+			int oy = limits.top() - highlightRect.top() >? 0;
+			
+			bitBlt(&backImage, absRect.x(), absRect.y(), 
+					&paperClipPix, ox, oy, paperClipWidth, paperClipHeight);
+			
+		}
+		else if ( highlightRect.isValid() && highlightRect.intersects( limits ) )
+		{
+			// find out the rect to highlight on pixmap
+			highlightRect = highlightRect.intersect( limits );
+			highlightRect.moveBy( -limits.left(), -limits.top() );
+
+			// highlight composition (product: highlight color * destcolor)
+			unsigned int * data = (unsigned int *)backImage.bits();
+			int val, newR, newG, newB,
+			rh = 128,
+			gh = 128,
+			bh = 255,
+			offset = highlightRect.top() * backImage.width();
+			for( int y = highlightRect.top(); y <= highlightRect.bottom(); ++y )
+			{
+				for( int x = highlightRect.left(); x <= highlightRect.right(); ++x )
+				{
+					val = data[ x + offset ];
+					newR = (qRed(val) * rh) / 255;
+					newG = (qGreen(val) * gh) / 255;
+					newB = (qBlue(val) * bh) / 255;
+					data[ x + offset ] = qRgba( newR, newG, newB, 255 );
+				}
+				offset += backImage.width();
+			}
+		}
+	}
+}
diff -Nurw kdegraphics-3.5.5/kpdf/ui/pagepainter.h kdegraphics-3.5.5-bw/kpdf/ui/pagepainter.h
--- kdegraphics-3.5.5/kpdf/ui/pagepainter.h	2005-09-10 01:18:43.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/pagepainter.h	2007-01-15 12:39:03.000000000 -0800
@@ -31,6 +31,9 @@
         // to pick up an alternative pixmap if the pixmap of 'id' is missing.
         static void paintPageOnPainter( const KPDFPage * page, int id, int flags,
             QPainter * p, const QRect & limits, int width = -1, int height = -1 );
+		
+	private:
+		static void paintBookmarksOnPage(const KPDFPage * page, QPainter * p, const QRect & limits, QImage &backImage);
 };
 
 #endif
diff -Nurw kdegraphics-3.5.5/kpdf/ui/pageview.cpp kdegraphics-3.5.5-bw/kpdf/ui/pageview.cpp
--- kdegraphics-3.5.5/kpdf/ui/pageview.cpp	2006-10-01 10:25:16.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/pageview.cpp	2007-01-15 16:57:18.000000000 -0800
@@ -47,6 +47,8 @@
 #include "pageview.h"
 #include "pageviewutils.h"
 #include "pagepainter.h"
+#include "bookmarkdialogimpl.h"
+#include "bookmarkballoon.h"
 #include "core/document.h"
 #include "core/page.h"
 #include "core/link.h"
@@ -96,6 +98,7 @@
     bool blockViewport;                 // prevents changes to viewport
     bool blockPixmapsRequest;           // prevent pixmap requests
     PageViewMessage * messageWindow;    // in pageviewutils.h
+	BookmarkBalloon bmBalloon;
 
     // actions
     KToggleAction * aMouseNormal;
@@ -183,8 +186,10 @@
     d->aZoom = new KSelectAction( i18n( "Zoom" ), "viewmag", 0, this, SLOT( slotZoom() ), ac, "zoom_to" );
     d->aZoom->setEditable( true );
 #if KDE_IS_VERSION(3,4,89)
+
     d->aZoom->setMaxComboViewCount( 13 );
 #endif
+
     updateZoomText();
 
     KStdAction::zoomIn( this, SLOT( slotZoomIn() ), ac, "zoom_in" );
@@ -385,10 +390,6 @@
 
 void PageView::notifyPageChanged( int pageNumber, int changedFlags )
 {
-    // only handle pixmap / highlight changes notifies
-    if ( changedFlags & DocumentObserver::Bookmark )
-        return;
-
     // iterate over visible items: if page(pageNumber) is one of them, repaint it
     QValueList< PageViewItem * >::iterator iIt = d->visibleItems.begin(), iEnd = d->visibleItems.end();
     for ( ; iIt != iEnd; ++iIt )
@@ -802,9 +803,23 @@
             }
             else
             {
+				PageViewItem * pageItem = pickItemOnPoint( e->x(), e->y() );
+				if( pageItem )
+				{
+					QPoint pageRelativePoint = pageItem->getPageRelativePoint(e->pos());
+					KPDFBookmark *bm = d->document->getBookmark(pageItem->pageNumber(), pageRelativePoint);
+					if( bm && (!bm->Comment().isEmpty() || !bm->Name().isEmpty()) )
+						d->bmBalloon.showBalloon(bm);
+					else
+						d->bmBalloon.hide();
+				}
+				else
+				{
+					d->bmBalloon.hide();
                 // only hovering the page, so update the cursor
                 updateCursor( e->pos() );
             }
+			}
             break;
 
         case MouseZoom:
@@ -889,7 +904,7 @@
     {
         // ..except for right Clicks (emitted even it viewport is empty)
         if ( e->button() == RightButton )
-            emit rightClick( 0, e->globalPos() );
+			emit rightClick( 0, e->globalPos(), e->pos() );
         return;
     }
 
@@ -911,7 +926,8 @@
          rightButton = e->button() & RightButton;
     switch ( d->mouseMode )
     {
-        case MouseNormal:{
+		case MouseNormal:
+			{
             // return the cursor to its normal state after dragging
             if ( cursor().shape() == Qt::SizeAllCursor )
                 updateCursor( e->pos() );
@@ -949,11 +965,15 @@
             }
             else if ( rightButton )
             {
+					QPoint pageRelativePoint(0,0);
+					if( pageItem )
+						pageRelativePoint = pageItem->getPageRelativePoint(e->pos());
                 // right click (if not within 5 px of the press point, the mode
                 // had been already changed to 'Selection' instead of 'Normal')
-                emit rightClick( pageItem ? pageItem->page() : 0, e->globalPos() );
+					emit rightClick( pageItem ? pageItem->page() : 0, e->globalPos(), pageRelativePoint );
             }
-            }break;
+			}
+			break;
 
         case MouseZoom:
             // if a selection rect has been defined, zoom into it
@@ -989,12 +1009,13 @@
             }
             break;
 
-        case MouseSelect:{
+		case MouseSelect:
+			{
 
             if (d->mouseSelectionRect.isNull() && rightButton) 
 	    {
 	    	 PageViewItem * pageItem = pickItemOnPoint( e->x(), e->y() );
-	    	 emit rightClick( pageItem ? pageItem->page() : 0, e->globalPos() );
+					emit rightClick( pageItem ? pageItem->page() : 0, e->globalPos(), e->pos() );
 	    }
 	    
             // if a selection is defined, display a popup
@@ -1049,6 +1070,9 @@
             menu.insertTitle( i18n( "Image (%1 by %2 pixels)" ).arg( selectionRect.width() ).arg( selectionRect.height() ) );
             menu.insertItem( SmallIcon("image"), i18n( "Copy to Clipboard" ), 3 );
             menu.insertItem( SmallIcon("filesave"), i18n( "Save to File..." ), 4 );
+				menu.insertTitle( i18n( "Bookmark (%1 by %2 pixels)" ).arg( selectionRect.width() ).arg( selectionRect.height() ) );
+				menu.insertItem( SmallIcon("bookmark_add"), i18n( "Add Bookmark" ), 5 );
+
             int choice = menu.exec( e->globalPos() );
             // IMAGE operation choosen
             if ( choice > 2 )
@@ -1083,6 +1107,33 @@
                         d->messageWindow->display( i18n( "Image [%1x%2] saved to %3 file." ).arg( copyPix.width() ).arg( copyPix.height() ).arg( type ) );
                     }
                 }
+					else if ( choice == 5 )
+					{
+						QValueVector< PageViewItem * >::iterator iIt = d->items.begin(), iEnd = d->items.end();
+						for ( ; iIt != iEnd; ++iIt )
+						{
+							PageViewItem * item = *iIt;
+							const QRect & itemRect = item->geometry();
+							if ( selectionRect.intersects( itemRect ) )
+							{
+								BookmarkDialogImpl bmdlg;
+								bmdlg.exec();
+								if( !bmdlg.wasCanceled() )
+								{
+									// request the textpage if there isn't one
+									KPDFPage * kpdfPage = (KPDFPage *)item->page();
+									// grab text in the rect that intersects itemRect
+									QRect relativeRect = selectionRect.intersect( itemRect );
+									relativeRect.moveBy( -itemRect.left(), -itemRect.top() );
+									d->document->addBookmark(
+											kpdfPage->number(), 
+											new KPDFBookmark(kpdfPage->number(), relativeRect, 
+															 bmdlg.getName().ascii(), bmdlg.getComment().ascii()) );
+								}
+							}
+						}
+					}
+
             }
             // TEXT operation choosen
             else
@@ -1145,7 +1196,8 @@
                 d->aPrevAction->activate();
                 d->aPrevAction = 0;
             }
-            }break;
+			}
+			break;
 
         case MouseEdit:      // ? apply [tool] ?
             break;
@@ -1164,7 +1216,8 @@
     int delta = e->delta(),
         vScroll = verticalScrollBar()->value();
     e->accept();
-    if ( (e->state() & ControlButton) == ControlButton ) {
+	if ( (e->state() & ControlButton) == ControlButton )
+	{
         if ( e->delta() < 0 )
             slotZoomOut();
         else
@@ -1406,10 +1459,12 @@
     KAction * checkedZoomAction = 0;
     switch ( newZoomMode )
     {
-        case ZoomFixed:{ //ZoomFixed case
+		case ZoomFixed:
+			{ //ZoomFixed case
             QString z = d->aZoom->currentText();
             newFactor = KGlobal::locale()->readNumber( z.remove( z.find( '%' ), 1 ) ) / 100.0;
-            }break;
+			}
+			break;
         case ZoomIn:
             newFactor += (newFactor > 0.99) ? ( newFactor > 1.99 ? 0.5 : 0.2 ) : 0.1;
             newZoomMode = ZoomFixed;
@@ -1477,7 +1532,10 @@
 
     // add percent items
     QString double_oh( "00" );
-    const float zoomValue[10] = { 0.125, 0.25, 0.333, 0.5, 0.667, 0.75, 1, 1.25, 1.50, 2 };
+	const float zoomValue[10] =
+	    {
+	        0.125, 0.25, 0.333, 0.5, 0.667, 0.75, 1, 1.25, 1.50, 2
+	    };
     int idx = 0,
         selIdx = 2; // use 3 if "fit text" present
     bool inserted = false; //use: "d->zoomMode != ZoomFixed" to hide Fit/* zoom ratio
@@ -1682,7 +1740,8 @@
                 // advance col index
                 insertX += colWidth[ cIdx ];
                 cIdx++;
-            } else
+			}
+			else
                 item->setGeometry( 0, 0, -1, -1 );
         }
 
@@ -1885,8 +1944,14 @@
 
     // compute delay between timer ticks and scroll amount per tick
     int index = abs( d->scrollIncrement ) - 1;  // 0..9
-    const int scrollDelay[10] =  { 200, 100, 50, 30, 20, 30, 25, 20, 30, 20 };
-    const int scrollOffset[10] = {   1,   1,  1,  1,  1,  2,  2,  2,  4,  4 };
+	const int scrollDelay[10] =
+	    {
+	        200, 100, 50, 30, 20, 30, 25, 20, 30, 20
+	    };
+	const int scrollOffset[10] =
+	    {
+	        1,   1,  1,  1,  1,  2,  2,  2,  4,  4
+	    };
     d->autoScrollTimer->changeInterval( scrollDelay[ index ] );
     scrollBy( 0, d->scrollIncrement > 0 ? scrollOffset[ index ] : -scrollOffset[ index ] );
 }
@@ -1926,17 +1991,20 @@
 
 void PageView::slotFitToWidthToggled( bool on )
 {
-    if ( on ) updateZoom( ZoomFitWidth );
+	if ( on )
+		updateZoom( ZoomFitWidth );
 }
 
 void PageView::slotFitToPageToggled( bool on )
 {
-    if ( on ) updateZoom( ZoomFitPage );
+	if ( on )
+		updateZoom( ZoomFitPage );
 }
 
 void PageView::slotFitToTextToggled( bool on )
 {
-    if ( on ) updateZoom( ZoomFitText );
+	if ( on )
+		updateZoom( ZoomFitText );
 }
 
 void PageView::slotTwoPagesToggled( bool on )
diff -Nurw kdegraphics-3.5.5/kpdf/ui/pageview.h kdegraphics-3.5.5-bw/kpdf/ui/pageview.h
--- kdegraphics-3.5.5/kpdf/ui/pageview.h	2006-10-01 10:25:16.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/pageview.h	2007-01-14 20:30:51.000000000 -0800
@@ -67,7 +67,7 @@
 
     signals:
         void urlDropped( const KURL& );
-        void rightClick( const KPDFPage *, const QPoint & );
+        void rightClick( const KPDFPage *, const QPoint &, const QPoint & );
 
     protected:
         // main draw loop, draws pageViews on viewport
diff -Nurw kdegraphics-3.5.5/kpdf/ui/pageviewutils.cpp kdegraphics-3.5.5-bw/kpdf/ui/pageviewutils.cpp
--- kdegraphics-3.5.5/kpdf/ui/pageviewutils.cpp	2006-10-01 10:25:16.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/pageviewutils.cpp	2007-01-15 10:44:22.000000000 -0800
@@ -197,3 +197,20 @@
     m_geometry.moveLeft( x );
     m_geometry.moveTop( y );
 }
+
+QPoint PageViewItem::getPageRelativePoint(QPoint pos)
+{
+	QPoint pageRelativePoint(0,0);
+	QRect selectionRect(pos.x(), pos.y(), 1, 1);
+	
+	const QRect & itemRect = geometry();
+	if ( selectionRect.intersects( itemRect ) )
+	{
+		QRect relativeRect = selectionRect.intersect( itemRect );
+		relativeRect.moveBy( -itemRect.left(), -itemRect.top() );
+		pageRelativePoint.setX(relativeRect.left());
+		pageRelativePoint.setY(relativeRect.top());
+	}
+	return pageRelativePoint;
+}
+
diff -Nurw kdegraphics-3.5.5/kpdf/ui/pageviewutils.h kdegraphics-3.5.5-bw/kpdf/ui/pageviewutils.h
--- kdegraphics-3.5.5/kpdf/ui/pageviewutils.h	2005-09-10 01:18:43.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/pageviewutils.h	2007-01-15 10:43:42.000000000 -0800
@@ -42,6 +42,8 @@
         void setWHZ( int w, int h, double zoom );
         void moveTo( int x, int y );
 
+	QPoint getPageRelativePoint(QPoint pos);
+
     private:
         const KPDFPage * m_page;
         double m_zoomFactor;
diff -Nurw kdegraphics-3.5.5/kpdf/ui/presentationwidget.cpp kdegraphics-3.5.5-bw/kpdf/ui/presentationwidget.cpp
--- kdegraphics-3.5.5/kpdf/ui/presentationwidget.cpp	2006-10-01 10:25:16.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/presentationwidget.cpp	2007-01-15 10:24:39.000000000 -0800
@@ -919,6 +919,7 @@
             return KPDFPageTransition( KPDFPageTransition::Replace );
             break;
     }
+	return KPDFPageTransition( KPDFPageTransition::Replace );
 }
 
 /** ONLY the TRANSITIONS GENERATION function from here on **/
diff -Nurw kdegraphics-3.5.5/kpdf/ui/thumbnaillist.cpp kdegraphics-3.5.5-bw/kpdf/ui/thumbnaillist.cpp
--- kdegraphics-3.5.5/kpdf/ui/thumbnaillist.cpp	2006-07-22 01:10:27.000000000 -0700
+++ kdegraphics-3.5.5-bw/kpdf/ui/thumbnaillist.cpp	2007-01-14 18:10:25.000000000 -0800
@@ -504,7 +504,7 @@
     if ( clipRect.top() < m_pixmapHeight + m_margin )
     {
         // if page is bookmarked draw a colored border
-        bool isBookmarked = m_page->hasBookmark();
+        bool isBookmarked = m_page->hasBookmarks();
         // draw the inner rect
         p.setPen( isBookmarked ? QColor( 0xFF8000 ) : Qt::black );
         p.drawRect( m_margin/2 - 1, m_margin/2 - 1, m_pixmapWidth + 2, m_pixmapHeight + 2 );
diff -Nurw kdegraphics-3.5.5/ksvg/core/.deps/CanvasFactory.Plo kdegraphics-3.5.5-bw/ksvg/core/.deps/CanvasFactory.Plo
--- kdegraphics-3.5.5/ksvg/core/.deps/CanvasFactory.Plo	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/ksvg/core/.deps/CanvasFactory.Plo	2007-01-13 13:46:25.000000000 -0800
@@ -0,0 +1 @@
+# dummy
diff -Nurw kdegraphics-3.5.5/ksvg/core/.deps/CanvasItems.Plo kdegraphics-3.5.5-bw/ksvg/core/.deps/CanvasItems.Plo
--- kdegraphics-3.5.5/ksvg/core/.deps/CanvasItems.Plo	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/ksvg/core/.deps/CanvasItems.Plo	2007-01-13 13:46:25.000000000 -0800
@@ -0,0 +1 @@
+# dummy
diff -Nurw kdegraphics-3.5.5/ksvg/core/.deps/DocumentFactory.Plo kdegraphics-3.5.5-bw/ksvg/core/.deps/DocumentFactory.Plo
--- kdegraphics-3.5.5/ksvg/core/.deps/DocumentFactory.Plo	1969-12-31 16:00:00.000000000 -0800
+++ kdegraphics-3.5.5-bw/ksvg/core/.deps/DocumentFactory.Plo	2007-01-13 13:46:25.000000000 -0800
@@ -0,0 +1 @@
+# dummy
